<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2025-02-11" />

<title>deep motif score distribution</title>

<script src="site_libs/header-attrs-2.26/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="site_libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.33/datatables.js"></script>
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<script src="site_libs/jszip-1.13.6/jszip.min.js"></script>
<script src="site_libs/pdfmake-1.13.6/pdfmake.js"></script>
<script src="site_libs/pdfmake-1.13.6/vfs_fonts.js"></script>
<link href="site_libs/dt-ext-buttons-1.13.6/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="site_libs/dt-ext-buttons-1.13.6/js/dataTables.buttons.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.13.6/js/buttons.html5.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.13.6/js/buttons.colVis.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.13.6/js/buttons.print.min.js"></script>
<link href="site_libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Autism_Research</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/han16/Autism_Research">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">deep motif score distribution</h1>
<h4 class="date">2025-02-11</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-04-01
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>deep_motifs/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230928code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230928)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230928code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230928)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomhan16deepmotifstree92c7f1d0d6281802939b4a9363bc7def2b77162btargetblank92c7f1da">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/han16/deep_motifs/tree/92c7f1d0d6281802939b4a9363bc7def2b77162b" target="_blank">92c7f1d</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomhan16deepmotifstree92c7f1d0d6281802939b4a9363bc7def2b77162btargetblank92c7f1da"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/han16/deep_motifs/tree/92c7f1d0d6281802939b4a9363bc7def2b77162b" target="_blank">92c7f1d</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Untracked files:
    Untracked:  VennDiagram.2025-04-01_10-42-23.373925.log
    Untracked:  analysis/VennDiagram.2025-04-01_10-32-02.404995.log
    Untracked:  analysis/VennDiagram.2025-04-01_10-32-23.588555.log
    Untracked:  analysis/VennDiagram.2025-04-01_10-38-28.065519.log
    Untracked:  analysis/VennDiagram.2025-04-01_10-41-53.32988.log
    Untracked:  analysis/VennDiagram.2025-04-01_10-42-57.748601.log
    Untracked:  analysis/VennDiagram.2025-04-01_10-43-21.217154.log
    Untracked:  analysis/VennDiagram.2025-04-01_10-43-40.547875.log
    Untracked:  analysis/VennDiagram.2025-04-01_10-43-56.323022.log
    Untracked:  analysis/VennDiagram.2025-04-01_10-44-09.657174.log
    Untracked:  analysis/VennDiagram.2025-04-01_10-44-22.652508.log

Unstaged changes:
    Modified:   analysis/part01_score_dist.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/part01_score_dist.Rmd</code>) and
HTML (<code>docs/part01_score_dist.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/han16/deep_motifs/blob/92c7f1d0d6281802939b4a9363bc7def2b77162b/analysis/part01_score_dist.Rmd" target="_blank">92c7f1d</a>
</td>
<td>
han
</td>
<td>
2025-03-31
</td>
<td>
3/31/2025
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/han16/deep_motifs/92c7f1d0d6281802939b4a9363bc7def2b77162b/docs/part01_score_dist.html" target="_blank">92c7f1d</a>
</td>
<td>
han
</td>
<td>
2025-03-31
</td>
<td>
3/31/2025
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/han16/deep_motifs/blob/889e3786cee3742345dbf9e1e4b0b918d40b4a46/analysis/part01_score_dist.Rmd" target="_blank">889e378</a>
</td>
<td>
han
</td>
<td>
2025-03-13
</td>
<td>
3/13/2025
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/han16/deep_motifs/889e3786cee3742345dbf9e1e4b0b918d40b4a46/docs/part01_score_dist.html" target="_blank">889e378</a>
</td>
<td>
han
</td>
<td>
2025-03-13
</td>
<td>
3/13/2025
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/han16/deep_motifs/blob/6f409b50d474f3abd3de040b695dff679545eeea/analysis/part01_score_dist.Rmd" target="_blank">6f409b5</a>
</td>
<td>
han
</td>
<td>
2025-03-13
</td>
<td>
3/13/2025
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/han16/deep_motifs/6f409b50d474f3abd3de040b695dff679545eeea/docs/part01_score_dist.html" target="_blank">6f409b5</a>
</td>
<td>
han
</td>
<td>
2025-03-13
</td>
<td>
3/13/2025
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/han16/deep_motifs/blob/1115d48291a2e1ad149ed2ed48a790c9ca6bf709/analysis/part01_score_dist.Rmd" target="_blank">1115d48</a>
</td>
<td>
han
</td>
<td>
2025-03-12
</td>
<td>
3/12/2025
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/han16/deep_motifs/1115d48291a2e1ad149ed2ed48a790c9ca6bf709/docs/part01_score_dist.html" target="_blank">1115d48</a>
</td>
<td>
han
</td>
<td>
2025-03-12
</td>
<td>
3/12/2025
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/han16/deep_motifs/blob/a9894a3a61260921d9aa07401ff1a8960b4fb696/analysis/part01_score_dist.Rmd" target="_blank">a9894a3</a>
</td>
<td>
han
</td>
<td>
2025-03-11
</td>
<td>
3/11/2025
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/han16/deep_motifs/a9894a3a61260921d9aa07401ff1a8960b4fb696/docs/part01_score_dist.html" target="_blank">a9894a3</a>
</td>
<td>
han
</td>
<td>
2025-03-11
</td>
<td>
3/11/2025
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/han16/deep_motifs/blob/cc957c3bc6bd0da66aef5d2310dbb0c0022be721/analysis/part01_score_dist.Rmd" target="_blank">cc957c3</a>
</td>
<td>
han
</td>
<td>
2025-02-14
</td>
<td>
2/14/2025
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/han16/deep_motifs/cc957c3bc6bd0da66aef5d2310dbb0c0022be721/docs/part01_score_dist.html" target="_blank">cc957c3</a>
</td>
<td>
han
</td>
<td>
2025-02-14
</td>
<td>
2/14/2025
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/han16/deep_motifs/blob/e6a17fb29def47b59e855f2250d954d552c3a353/analysis/part01_score_dist.Rmd" target="_blank">e6a17fb</a>
</td>
<td>
han
</td>
<td>
2025-02-11
</td>
<td>
2/11/2025
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/han16/deep_motifs/e6a17fb29def47b59e855f2250d954d552c3a353/docs/part01_score_dist.html" target="_blank">e6a17fb</a>
</td>
<td>
han
</td>
<td>
2025-02-11
</td>
<td>
2/11/2025
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>rm(list=ls())
set.seed(123)
library(tidyverse)
library(ggplot2)
library(DT)
library(rprojroot)
root &lt;- rprojroot::find_rstudio_root_file()</code></pre>
<pre class="r"><code>#result_round1=as_tibble(read.csv(file.path(root, &quot;../DiWangResults/ASD_prediction_results/ASD_RISK_PREDICTION_round1.csv&quot;))) # old results
result_round1=as_tibble(read.csv(file.path(root, &quot;../DiWangResults/results updated_20250303/ASD_RISK_PREDICTION round 1.csv&quot;)))
result_round1=result_round1[result_round1[&#39;gene&#39;] != 0,]  # remove rows with gene=0

forecASD_results=as_tibble(read.csv(file.path(root, &quot;../DiWangResults/forecASD_results.csv&quot;)))
forecASD_results=forecASD_results[forecASD_results[&#39;gene&#39;] != 0,]  # remove rows with gene=0</code></pre>
<div id="histogram-of-scores" class="section level2">
<h2>histogram of scores</h2>
<pre class="r"><code>ggplot(result_round1, aes(x = result_round1$RISK_PROBABILITY_FUSION)) +
  geom_histogram(binwidth = 0.1, fill = &quot;blue&quot;, color = &quot;black&quot;, aes(y = ..count..)) +
  stat_bin(binwidth = 0.1, geom = &quot;text&quot;, aes(label = ..count..), vjust = -0.5, color = &quot;red&quot;) +
  labs(title = &quot;Histogram of deep motif scores&quot;, x = &quot;deep motif score&quot;, y = &quot;&quot;)+
  geom_vline(xintercept = 0.5, color = &quot;green&quot;, linetype = &quot;dashed&quot;, size = 1)+
  theme(plot.title = element_text(hjust = 0.5, size=15))  #center the title </code></pre>
<p><img src="figure/part01_score_dist.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/han16/deep_motifs/blob/1115d48291a2e1ad149ed2ed48a790c9ca6bf709/docs/figure/part01_score_dist.Rmd/unnamed-chunk-3-1.png" target="_blank">1115d48</a>
</td>
<td>
han
</td>
<td>
2025-03-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/han16/deep_motifs/blob/a9894a3a61260921d9aa07401ff1a8960b4fb696/docs/figure/part01_score_dist.Rmd/unnamed-chunk-3-1.png" target="_blank">a9894a3</a>
</td>
<td>
han
</td>
<td>
2025-03-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/han16/deep_motifs/blob/cc957c3bc6bd0da66aef5d2310dbb0c0022be721/docs/figure/part01_score_dist.Rmd/unnamed-chunk-3-1.png" target="_blank">cc957c3</a>
</td>
<td>
han
</td>
<td>
2025-02-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/han16/deep_motifs/blob/e6a17fb29def47b59e855f2250d954d552c3a353/docs/figure/part01_score_dist.Rmd/unnamed-chunk-3-1.png" target="_blank">e6a17fb</a>
</td>
<td>
han
</td>
<td>
2025-02-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(forecASD_results, aes(x = forecASD_results$risk.score)) +
  geom_histogram(binwidth = 0.1, fill = &quot;blue&quot;, color = &quot;black&quot;, aes(y = ..count..)) +
  stat_bin(binwidth = 0.1, geom = &quot;text&quot;, aes(label = ..count..), vjust = -0.5, color = &quot;red&quot;) +
  labs(title = &quot;Histogram of forecASD scores&quot;, x = &quot;forecASD score&quot;, y = &quot;&quot;)+
  geom_vline(xintercept = 0.5, color = &quot;green&quot;, linetype = &quot;dashed&quot;, size = 1)+
  theme(plot.title = element_text(hjust = 0.5, size=15))  #center the title </code></pre>
<p><img src="figure/part01_score_dist.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/han16/deep_motifs/blob/92c7f1d0d6281802939b4a9363bc7def2b77162b/docs/figure/part01_score_dist.Rmd/unnamed-chunk-4-1.png" target="_blank">92c7f1d</a>
</td>
<td>
han
</td>
<td>
2025-03-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Combine the data into a data frame
data &lt;- data.frame(
  scores = c(result_round1$RISK_PROBABILITY_FUSION, forecASD_results$risk.score),
  group = factor(c(rep(&quot;deep-motifs&quot;,length(result_round1$RISK_PROBABILITY_FUSION)), rep(&quot;forecASD&quot;, length(forecASD_results$risk.score))))
)

# Load ggplot2 for plotting
library(ggplot2)

# Create the histogram with side-by-side bars
ggplot(data, aes(x = scores, fill = group)) +
  geom_histogram(position = &quot;dodge&quot;, bins = 20, alpha = 0.7) +
  #geom_histogram(position = &quot;dodge&quot;, bins = 20, color = &quot;black&quot;) +
  #stat_bin(aes(label = ..count..), geom = &quot;text&quot;, position = position_dodge(width = 0.9), vjust = -0.5) +
  labs(title = &quot;Histogram of scores by forecASD and deep-motifs&quot;, x = &quot;scores&quot;, y = &quot;&quot;)+
  geom_vline(xintercept = 0.5, color = &quot;green&quot;, linetype = &quot;dashed&quot;, size = 1)+
  theme(plot.title = element_text(hjust = 0.5, size=15))  #center the title </code></pre>
<p><img src="figure/part01_score_dist.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/han16/deep_motifs/blob/92c7f1d0d6281802939b4a9363bc7def2b77162b/docs/figure/part01_score_dist.Rmd/unnamed-chunk-5-1.png" target="_blank">92c7f1d</a>
</td>
<td>
han
</td>
<td>
2025-03-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="implicated-risk-genes" class="section level2">
<h2>implicated risk genes</h2>
<pre class="r"><code>score_cutoff=0.5


SFARI_genes=as_tibble(read.csv(file.path(root, &quot;../../../Dataset/SFARI_base/SFARI-Gene_genes_08-19-2024release_09-16-2024export.csv&quot;)))
positive_genes=SFARI_genes %&gt;% filter (gene.score==&quot;1&quot;) %&gt;% dplyr::select(gene.symbol) %&gt;% pull()


motifs_genes=result_round1 %&gt;% filter(RISK_PROBABILITY_FUSION&gt;score_cutoff) %&gt;% dplyr::select(gene) %&gt;% pull()



SFARI_score_for_motif_genes &lt;- ifelse(
  motifs_genes %in% SFARI_genes$gene.symbol,
  SFARI_genes$gene.score[match(motifs_genes, SFARI_genes$gene.symbol)],
  NA
)

result_round1 %&gt;% filter(RISK_PROBABILITY_FUSION&gt;score_cutoff) %&gt;% dplyr::select(gene, gene_gencodeV33, chromosome, RISK_PROBABILITY_FUSION)%&gt;% mutate(SFARI_score=SFARI_score_for_motif_genes) %&gt;% arrange(desc(RISK_PROBABILITY_FUSION)) %&gt;% 
datatable(extensions = &#39;Buttons&#39;,
            options = list(dom = &#39;Blfrtip&#39;,
                           buttons = c(&#39;copy&#39;, &#39;csv&#39;, &#39;excel&#39;, &#39;pdf&#39;, &#39;print&#39;),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,&quot;All&quot;))))</code></pre>
<div class="datatables html-widget html-fill-item" id="htmlwidget-9eceb20dc2a931752ae4" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-9eceb20dc2a931752ae4">{"x":{"filter":"none","vertical":false,"extensions":["Buttons"],"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140","141","142","143","144","145","146","147","148","149","150","151","152","153","154","155","156","157","158","159","160","161","162","163","164","165","166","167","168","169","170","171","172","173","174","175","176","177","178","179","180","181","182","183","184","185","186","187","188","189","190","191","192","193","194","195","196","197","198","199","200","201","202","203","204","205","206","207","208","209","210","211","212","213","214","215","216","217","218","219","220","221","222","223","224","225","226","227","228","229","230","231","232","233","234","235","236","237","238","239","240","241","242","243","244","245","246","247","248","249","250","251","252","253","254","255","256","257","258","259","260","261","262","263","264","265","266","267","268","269","270","271","272","273","274","275","276","277","278","279","280","281","282","283","284","285","286","287","288","289","290","291","292","293","294","295","296","297","298","299","300","301","302","303","304","305","306","307","308","309","310","311","312","313","314","315","316","317","318","319","320","321","322","323","324","325","326","327","328","329","330","331","332","333","334","335","336","337","338","339","340","341","342","343","344","345","346","347","348","349","350","351","352","353","354","355","356","357","358","359","360","361","362","363","364","365","366","367","368","369","370","371","372","373","374","375","376","377","378","379","380","381","382","383","384","385","386","387","388","389","390","391","392","393","394","395","396","397","398","399","400","401","402","403","404","405","406","407","408","409","410","411","412","413","414","415","416","417","418","419","420","421","422","423","424","425","426","427","428","429","430","431","432","433","434","435","436","437","438","439","440","441","442","443","444","445","446","447"],["YY1","MEF2C","TBL1XR1","ANK2","SETD1A","TRIP12","CACNA1E","SMARCA4","ANKRD11","AP2S1","DYNC1H1","ATP2B2","PPP1R9B","PHF12","CTNNB1","DLG4","ANP32A","RORB","KDM6B","PHF21A","EIF3G","ASXL3","GRIN1","SMARCA2","CTCF","SUV420H1","BRSK2","POGZ","GABRB3","PTPN11","SHANK3","GNAI1","LRRC4C","MYT1L","ZMYND8","DMPK","SATB1","RELN","KDM5B","PHF3","ZMYND11","ELAVL3","QRICH1","HNRNPUL2","KMT2E","GRIA2","TANC2","RFX3","TSHZ1","NRXN3","IRF2BPL","GIGYF1","ADNP","MBD5","TCF4","DMWD","DSCAM","PBX1","DYRK1A","AGAP2","PSMD11","SIN3A","NRXN1","TRIO","RERE","LDB1","CAMTA2","SCN1A","CYB5R3","VEZF1","EHMT1","SHANK2","STXBP1","CADM2","NF1","NACC1","SPAST","CHD2","TLE3","APBB1","PCDH10","KMT2C","WDFY3","PEG10","PACS1","CREBBP","CELF4","ARID1A","KDM3B","KMT2A","AUTS2","SPEN","TBR1","KIF5C","HNRNPD","RALGAPB","WAC","SRPR","UBR1","MED13","PPP2R5D","SOX5","RAI1","SATB2","PRR12","SETD5","MEIS2","CHD8","CAMSAP1","MED13L","NAA15","SLC6A1","GRIN2B","SMARCC2","ASH1L","FOXP1","ARID1B","CUL3","PSMD6","CNTN1","SYNGAP1","GATAD2B","SKI","MAP1A","TSC1","CHD3","HECTD4","TCF20","KCNB1","AHDC1","CSDE1","PAX5","SYNCRIP","BCL11A","BRD4","CAPRIN1","GSK3B","SCN2A","PHF2","SYT1","TAOK1","SPRY2","ZNF462","SRGAP3","EBF3","SRCAP","FUBP3","TLK2","MARK2","EP300","ANK3","KIAA0232","SCAF1","LRRC4","RUNX1T1","KIF3C","CELF2","UBR5","GGNBP2","SPTBN1","HECTD1","CPSF7","TCF7L2","HDAC4","ZBTB21","PSD","ZNF292","CSNK1E","NIPBL","SETBP1","SETD1B","DEDD","NR4A2","NFIX","CYFIP2","NCOA1","PTEN","CSNK2A1","TRAF7","PSMD12","PTPRF","LMTK3","EXT1","TERF2","RUSC1","CBX1","C16orf72","KCNQ2","PCLO","NAV3","CIC","NSD1","PLEKHO1","DPYSL2","CNOT3","CASZ1","LZTS3","EIF3D","EHBP1","CLPTM1","ABCE1","TNRC6C","CLASP1","DEAF1","TNPO3","PTPRD","MIB1","MLXIP","ATRN","TBC1D14","TBCEL","HNRNPU","NBEA","GRIK2","TSC22D1","ZBTB20","RTF1","PUM1","PNN","SSR2","RAB11FIP4","PLXNA2","AP2M1","ARF3","KLHDC10","DNAJC5","TARDBP","PLXNB1","PSMA3","NUP155","DNMT3A","ZMYM2","TM9SF4","CGREF1","HDAC5","TAF4","INO80D","MBD6","NPTN","CORO1A","THRB","PUM2","CASKIN1","MACF1","DIP2C","EFTUD2","NCKAP1","G3BP1","GFAP","DIP2A","PGM2L1","SMG7","BAZ2B","CAMK2B","GABRB2","DDB1","ATP1A3","AHNAK","NR3C2","SMARCAD1","CUX2","TNRC6B","PCDHAC2","CPD","ELAVL2","GRN","KAT6A","NOVA2","PRR14L","ZEB2","KCNQ3","SMARCE1","KLC2","MRFAP1","BAZ1B","CHD6","FEZF2","GIGYF2","EIF3B","PRH1","CACNA1C","TESK1","RAB2A","TCERG1","PAPOLG","CACNA1A","KIDINS220","GRIN2A","ARHGEF18","MTOR","SETD2","VGF","NLGN2","SON","MKX","SCN8A","KDM2B","SOS2","PPP5C","KIF1B","UBAP2L","HDLBP","TSC2","BRAF","PLXNA1","RIMS1","TRIM23","NRXN2","MAGEL2","ADSL","MAZ","HIVEP2","CHD7","MYCBP2","TSHZ3","PTK7","TEK","HRAS","UBE3A","CHAMP1","KLHL20","PHIP","CACNA2D3","LZTR1","KANSL1","TBCK","SLIT1","FOXP2","KATNAL2","FOXG1","ADRM1","POMGNT1","EIF4G3","ZBTB18","NDST1","NEO1","RBM12","ZC3H14","FLRT2","DOCK3","ZNF608","DGKD","TWF2","DLL1","BRPF3","NFIL3","PAH","GAPVD1","SEC24B","BCKDK","DPP6","ITSN1","FAM120A","MTMR3","MYH10","WHSC1","ARHGEF12","RBBP5","KIAA0368","KLF7","MSL2","ITGB4","YLPM1","PPP1CC","CTNND1","ADCY1","PLEKHM3","DLGAP4","ILF3","PPM1E","CXXC5","TNIK","KDM4A","BPTF","TSPAN14","KDM4B","ZNF865","SP3","CBX4","FBXO11","MBOAT7","PPIG","SLC17A7","RPL15","AKAP9","DVL3","USP5","EDF1","PAK7","HCN1","DYRK2","SCAF4","SFPQ","HUNK","CAMK2A","RNF220","ZNF423","ARID5B","AGO4","VPS13B","ARIH1","DHCR7","PRDM15","DNAH1","BRINP1","FUS","UNC13A","DDX6","SMG1","C11orf30","CECR2","ZMYM4","ATP1B1","MAN1A1","MYO18A","SEC24D","ATXN2L","SYT13","MGEA5","ZNF536","MYO5A","PIK3CA","ASAP1","APC2","NDRG4","EIF4E","LINGO1","YWHAG","ALDH5A1","GRAMD1B","COBL","SCAPER","ARHGAP32","CLTC","PLXNB2","KCNH7","SNRNP200","LEO1","TRIM24","GRIK3","MTCH1","APLP2","BIRC6","SRCIN1","ZMIZ1","CDK13","RNF146"],["YY1","MEF2C","TBL1XR1","ANK2","SETD1A","TRIP12","CACNA1E","SMARCA4","ANKRD11","AP2S1","DYNC1H1","ATP2B2","PPP1R9B","PHF12","CTNNB1","DLG4","ANP32A","RORB","KDM6B","PHF21A","EIF3G","ASXL3","GRIN1","SMARCA2","CTCF","KMT5B","BRSK2","POGZ","GABRB3","PTPN11","SHANK3","GNAI1","LRRC4C","MYT1L","ZMYND8","DMPK","SATB1","RELN","KDM5B","PHF3","ZMYND11","ELAVL3","QRICH1","HNRNPUL2","KMT2E","GRIA2","TANC2","RFX3","TSHZ1","NRXN3","IRF2BPL","GIGYF1","ADNP","MBD5","TCF4","DMWD","DSCAM","PBX1","DYRK1A","AGAP2","PSMD11","SIN3A","NRXN1","TRIO","RERE","LDB1","CAMTA2","SCN1A","CYB5R3","VEZF1","EHMT1","SHANK2","STXBP1","CADM2","NF1","NACC1","SPAST","CHD2","TLE3","APBB1","PCDH10","KMT2C","WDFY3","PEG10","PACS1","CREBBP","CELF4","ARID1A","KDM3B","KMT2A","AUTS2","SPEN","TBR1","KIF5C","HNRNPD","RALGAPB","WAC","SRPRA","UBR1","MED13","PPP2R5D","SOX5","RAI1","SATB2","PRR12","SETD5","MEIS2","CHD8","CAMSAP1","MED13L","NAA15","SLC6A1","GRIN2B","SMARCC2","ASH1L","FOXP1","ARID1B","CUL3","PSMD6","CNTN1","SYNGAP1","GATAD2B","SKI","MAP1A","TSC1","CHD3","HECTD4","TCF20","KCNB1","AHDC1","CSDE1","PAX5","SYNCRIP","BCL11A","BRD4","CAPRIN1","GSK3B","SCN2A","PHF2","SYT1","TAOK1","SPRY2","ZNF462","SRGAP3","EBF3","SRCAP","FUBP3","TLK2","MARK2","EP300","ANK3","KIAA0232","SCAF1","LRRC4","RUNX1T1","KIF3C","CELF2","UBR5","GGNBP2","SPTBN1","HECTD1","CPSF7","TCF7L2","HDAC4","ZBTB21","PSD","ZNF292","CSNK1E","NIPBL","SETBP1","SETD1B","DEDD","NR4A2","NFIX","CYFIP2","NCOA1","PTEN","CSNK2A1","TRAF7","PSMD12","PTPRF","LMTK3","EXT1","TERF2","RUSC1","CBX1","C16orf72","KCNQ2","PCLO","NAV3","CIC","NSD1","PLEKHO1","DPYSL2","CNOT3","CASZ1","LZTS3","EIF3D","EHBP1","CLPTM1","ABCE1","TNRC6C","CLASP1","DEAF1","TNPO3","PTPRD","MIB1","MLXIP","ATRN","TBC1D14","TBCEL","HNRNPU","NBEA","GRIK2","TSC22D1","ZBTB20","RTF1","PUM1","PNN","SSR2","RAB11FIP4","PLXNA2","AP2M1","ARF3","KLHDC10","DNAJC5","TARDBP","PLXNB1","PSMA3","NUP155","DNMT3A","ZMYM2","TM9SF4","CGREF1","HDAC5","TAF4","INO80D","MBD6","NPTN","CORO1A","THRB","PUM2","CASKIN1","MACF1","DIP2C","EFTUD2","NCKAP1","G3BP1","GFAP","DIP2A","PGM2L1","SMG7","BAZ2B","CAMK2B","GABRB2","DDB1","ATP1A3","AHNAK","NR3C2","SMARCAD1","CUX2","TNRC6B","PCDHAC2","CPD","ELAVL2","GRN","KAT6A","NOVA2","PRR14L","ZEB2","KCNQ3","SMARCE1","KLC2","MRFAP1","BAZ1B","CHD6","FEZF2","GIGYF2","EIF3B","PRH1","CACNA1C","TESK1","RAB2A","TCERG1","PAPOLG","CACNA1A","KIDINS220","GRIN2A","ARHGEF18","MTOR","SETD2","VGF","NLGN2","SON","MKX","SCN8A","KDM2B","SOS2","PPP5C","KIF1B","UBAP2L","HDLBP","TSC2","BRAF","PLXNA1","RIMS1","TRIM23","NRXN2","MAGEL2","ADSL","MAZ","HIVEP2","CHD7","MYCBP2","TSHZ3","PTK7","TEK","HRAS","UBE3A","CHAMP1","KLHL20","PHIP","CACNA2D3","LZTR1","KANSL1","TBCK","SLIT1","FOXP2","KATNAL2","FOXG1","ADRM1","POMGNT1","EIF4G3","ZBTB18","NDST1","NEO1","RBM12","ZC3H14","FLRT2","DOCK3","ZNF608","DGKD","TWF2","DLL1","BRPF3","NFIL3","PAH","GAPVD1","SEC24B","BCKDK","DPP6","ITSN1","FAM120A","MTMR3","MYH10","NSD2","ARHGEF12","RBBP5","ECPAS","KLF7","MSL2","ITGB4","YLPM1","PPP1CC","CTNND1","ADCY1","PLEKHM3","DLGAP4","ILF3","PPM1E","CXXC5","TNIK","KDM4A","BPTF","TSPAN14","KDM4B","ZNF865","SP3","CBX4","FBXO11","MBOAT7","PPIG","SLC17A7","RPL15","AKAP9","DVL3","USP5","EDF1","PAK5","HCN1","DYRK2","SCAF4","SFPQ","HUNK","CAMK2A","RNF220","ZNF423","ARID5B","AGO4","VPS13B","ARIH1","DHCR7","PRDM15","DNAH1","BRINP1","FUS","UNC13A","DDX6","SMG1","EMSY","CECR2","ZMYM4","ATP1B1","MAN1A1","MYO18A","SEC24D","ATXN2L","SYT13","OGA","ZNF536","MYO5A","PIK3CA","ASAP1","APC2","NDRG4","EIF4E","LINGO1","YWHAG","ALDH5A1","GRAMD1B","COBL","SCAPER","ARHGAP32","CLTC","PLXNB2","KCNH7","SNRNP200","LEO1","TRIM24","GRIK3","MTCH1","APLP2","BIRC6","SRCIN1","ZMIZ1","CDK13","RNF146"],[14,5,3,4,16,2,1,19,16,19,14,3,17,17,3,17,15,9,17,11,19,18,9,9,16,11,11,1,15,12,22,7,11,2,20,19,3,7,1,6,10,19,3,11,7,4,17,9,18,14,14,7,20,2,18,19,21,1,21,12,17,15,2,5,1,10,17,2,22,17,9,11,9,3,17,19,2,15,15,11,4,7,4,7,11,16,18,1,5,11,7,1,2,2,4,20,10,11,15,17,6,12,17,2,19,3,15,14,9,12,4,3,12,12,1,3,6,2,3,12,6,1,1,15,9,17,12,22,20,1,1,9,6,2,19,11,3,2,9,12,17,13,9,3,10,16,9,17,11,22,10,4,19,7,8,2,10,8,17,2,14,11,10,2,21,10,6,22,5,18,12,1,2,19,5,2,10,20,16,17,1,19,8,16,1,17,16,20,7,12,19,5,1,8,19,1,20,22,2,19,4,17,2,11,7,9,18,12,20,4,11,1,13,6,13,3,15,1,14,1,17,1,3,12,7,20,1,3,14,5,2,13,20,2,17,20,2,12,15,16,3,2,16,1,10,17,2,5,17,21,11,1,2,7,5,11,19,11,4,4,12,22,5,17,9,17,8,19,22,2,8,17,11,4,7,20,3,2,7,12,12,9,8,5,2,19,2,16,19,1,3,7,17,21,10,12,12,14,19,1,1,2,16,7,3,6,5,11,15,22,16,6,8,13,19,6,9,11,15,13,1,6,3,22,17,4,10,7,18,14,20,1,1,1,5,15,20,14,14,3,5,2,3,6,6,9,12,9,4,16,7,21,9,22,17,4,11,1,9,2,3,17,14,12,11,7,2,20,19,17,5,3,1,17,10,19,19,2,17,2,19,2,19,3,7,3,12,9,20,5,12,21,1,21,5,1,16,10,1,8,15,11,21,3,9,16,19,11,16,11,22,1,1,6,17,4,16,11,10,19,15,3,8,19,16,4,15,7,6,11,7,15,11,17,22,2,2,15,7,1,6,11,2,17,10,7,6],[0.9977296,0.99769735,0.9976969,0.9976768499999999,0.997671,0.99766797,0.9976632,0.99766093,0.9976548,0.9976533,0.9976509,0.9976425,0.99764097,0.99763834,0.99763787,0.9976357,0.9976325,0.9976322,0.997632,0.9976287,0.99762803,0.99762505,0.99762505,0.997625,0.9976237999999999,0.9976225,0.9976195,0.9976155,0.99761546,0.9976151599999999,0.9976142,0.99761325,0.9976115,0.9976102999999999,0.9976102999999999,0.9976096,0.99760896,0.9976078,0.9976064,0.9976053,0.997605,0.99760437,0.9976036,0.99760264,0.9976009,0.9976007,0.9976004000000001,0.9975994,0.99759424,0.9975918,0.99758893,0.997588,0.9975878,0.9975856,0.9975855,0.9975853,0.9975830999999999,0.99758136,0.9975805,0.9975797,0.997568,0.9975645,0.9975635,0.9975607,0.9975604,0.9975589500000001,0.9975581,0.99755496,0.9975534700000001,0.9975518,0.99755067,0.99754953,0.9975494,0.9975489400000001,0.9975402,0.99753857,0.99753857,0.9975368,0.9975344,0.99753255,0.9975317,0.99753106,0.9975303,0.9975246,0.99752367,0.99752027,0.99751806,0.99751645,0.9975157,0.9975155,0.99751407,0.9975128,0.9975114,0.99751073,0.9975106,0.99751014,0.9975095,0.9975088,0.9975074,0.99750483,0.99750125,0.9974985,0.9974942,0.9974927,0.99749243,0.99749184,0.9974915,0.9974906,0.9974885999999999,0.9974883,0.99748737,0.997486,0.9974831,0.9974828,0.99748266,0.9974823,0.9974806000000001,0.99747694,0.997476,0.9974737,0.9974736,0.9974734,0.99747217,0.9974717,0.99747145,0.9974712,0.99747056,0.9974704,0.9974704,0.9974688,0.99746877,0.9974674,0.99746716,0.9974662,0.9974626,0.9974623,0.9974615,0.9974599,0.99745893,0.9974487399999999,0.9974468399999999,0.99744207,0.99744165,0.9974412,0.9974376,0.9974326999999999,0.99742126,0.9974209000000001,0.9974209000000001,0.99741554,0.9974154,0.99740756,0.99740756,0.9973963,0.9973919999999999,0.99738514,0.9973749,0.99736965,0.9973661,0.9973641600000001,0.9973611999999999,0.9973446,0.9973389,0.99732953,0.99732846,0.9973213,0.9973195,0.99731904,0.9972429,0.9972345,0.9972288,0.9972251,0.9972026000000001,0.99719965,0.9971922,0.9971746,0.9971725299999999,0.9971674699999999,0.99711293,0.9971119000000001,0.9971021,0.9970583,0.99704504,0.9969848,0.9969562,0.9969473,0.99693805,0.99691933,0.9968859,0.9968812,0.9968606,0.9967897999999999,0.9967739,0.9966429,0.99660796,0.99657553,0.99653655,0.9963582,0.99623567,0.996078,0.99591905,0.99575585,0.99573725,0.9956700000000001,0.99551964,0.9955079,0.99542284,0.99540436,0.9951371,0.99512947,0.99488497,0.9944725,0.9943989,0.9943360999999999,0.9941993,0.9941287,0.9938939,0.99364316,0.9936242,0.9934367,0.9923889,0.99202365,0.9914587,0.9909114999999999,0.99006927,0.9898326,0.9894129,0.98937434,0.9884526,0.9882069999999999,0.98777854,0.98747504,0.98587495,0.98522633,0.98061836,0.9803851,0.9763632,0.9757284000000001,0.9751538,0.9725911,0.97240716,0.9681858,0.9632267,0.9621851,0.9564083,0.93725985,0.9370044,0.93393743,0.93347013,0.9268567,0.9247719599999999,0.9152954,0.9102050699999999,0.9096673,0.9089210599999999,0.90850526,0.90525407,0.9036007,0.9017289000000001,0.90066594,0.88389426,0.883185,0.8825943000000001,0.8750067,0.87495214,0.8722224,0.86903125,0.8623269,0.8622527,0.8595975,0.8586176599999999,0.84811753,0.8476205,0.83982515,0.8385941,0.8374852,0.831995,0.8277898,0.82576513,0.8217886,0.81707877,0.8154192,0.8105699,0.8077156,0.806291,0.8037445,0.8025869,0.80232114,0.8022561,0.8007228,0.80049026,0.7999085,0.7998365,0.7997198,0.7991969,0.7989043,0.7987668,0.79866064,0.79855764,0.7985517,0.7985404,0.7984516,0.7984514,0.7983717,0.7983545,0.7983521,0.7982787,0.79821885,0.7980424,0.79801625,0.79773915,0.7977221,0.79766846,0.79764616,0.7976278999999999,0.79761237,0.79748315,0.79746425,0.79738307,0.7971792,0.7969567,0.79686654,0.7967194,0.79669,0.7965783,0.7962912,0.7962233,0.7959732,0.7957319,0.7950865,0.793747,0.7935025999999999,0.7913416599999999,0.7900816000000001,0.789238,0.78752154,0.7833338,0.77928776,0.7775445,0.77367485,0.77015924,0.76484644,0.7631656999999999,0.7587752,0.7572113,0.75307906,0.7510255,0.75044113,0.7484987400000001,0.7438315,0.738256,0.73808897,0.7380601,0.73768866,0.7338079,0.73214364,0.72194403,0.7201621499999999,0.7182785,0.717596,0.7090737,0.7085911,0.6906365,0.6903677,0.68569696,0.67742205,0.6760627,0.6758707,0.6632365,0.6623924,0.6615807,0.66002524,0.65415174,0.6540898000000001,0.6503447999999999,0.64766806,0.6402124,0.6281816,0.6253172,0.6253072,0.62444586,0.6243320999999999,0.6228286,0.6214806,0.6195731,0.6174718,0.6169258399999999,0.61662644,0.6143726,0.6129464999999999,0.6126493,0.6117099499999999,0.6075275999999999,0.6065609,0.60643893,0.6028396499999999,0.6020068,0.6013747,0.60107076,0.5996456999999999,0.5994815,0.5993199,0.5989662,0.5985886,0.5977902,0.59751487,0.5972332,0.5958844,0.5943432,0.59384537,0.5921861,0.590606,0.58835715,0.5882396,0.5877068,0.58762306,0.5872287,0.5870297,0.5870274,0.5865000500000001,0.5855987,0.58554536,0.5831045500000001,0.5823537,0.5810206999999999,0.57459724,0.57278717,0.5725951,0.5663399,0.5631597,0.5599023,0.5554427,0.54965913,0.5461494,0.5454005,0.5435750499999999,0.5382606,0.5285813,0.52761304,0.52512825,0.5229856000000001,0.51935685,0.51373035,0.5079747,0.5041013,0.50351256,0.5007688],[1,1,1,1,1,1,1,1,1,1,1,2,1,1,1,1,1,1,1,1,1,1,1,1,1,null,1,1,1,1,1,1,1,1,1,1,1,1,1,1,2,1,2,2,1,1,1,1,1,1,1,1,1,1,1,2,1,2,1,2,1,1,1,1,1,1,1,1,null,1,1,1,1,2,1,1,1,1,1,2,2,1,1,null,1,1,1,null,1,1,1,2,1,2,2,1,1,null,1,1,1,1,1,2,1,1,1,1,null,1,1,1,1,1,1,1,1,1,1,null,1,null,1,1,1,1,1,1,1,1,1,1,2,1,2,1,null,1,1,null,1,2,1,2,1,1,null,1,2,1,1,1,2,2,1,null,2,2,2,2,null,1,1,2,1,null,1,2,1,1,2,null,1,null,null,1,1,1,1,1,null,3,2,2,null,3,null,2,2,2,1,1,null,1,1,1,null,null,null,null,1,2,2,1,3,2,2,null,null,null,1,1,1,2,null,1,null,null,null,null,null,null,2,1,null,null,null,2,null,1,1,2,1,null,null,2,null,2,3,1,null,null,2,3,2,null,1,null,1,1,null,null,1,null,1,null,2,2,1,null,2,2,2,null,2,null,2,null,1,null,1,null,null,null,null,null,2,1,null,null,1,null,2,null,2,1,null,1,null,1,1,null,1,1,1,1,1,1,1,null,1,1,1,1,null,1,1,1,1,1,null,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,null,1,1,1,null,1,null,null,null,2,null,null,null,null,null,null,null,2,null,null,1,null,null,1,2,2,null,null,2,null,null,2,null,3,3,null,null,null,null,null,null,null,null,null,null,null,null,null,null,2,null,null,null,2,1,null,null,null,2,2,null,null,null,3,null,2,null,null,2,null,null,null,2,1,null,1,null,null,null,null,2,null,null,null,2,null,null,null,null,null,null,null,null,null,2,3,null,null,null,2,null,3,1,null,null,null,2,null,null,null,null,2,null,2,null,null,2,null,2,null,null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>gene<\/th>\n      <th>gene_gencodeV33<\/th>\n      <th>chromosome<\/th>\n      <th>RISK_PROBABILITY_FUSION<\/th>\n      <th>SFARI_score<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["copy","csv","excel","pdf","print"],"lengthMenu":[[10,25,50,-1],["10","25","50","All"]],"columnDefs":[{"className":"dt-right","targets":[3,4,5]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"gene","targets":1},{"name":"gene_gencodeV33","targets":2},{"name":"chromosome","targets":3},{"name":"RISK_PROBABILITY_FUSION","targets":4},{"name":"SFARI_score","targets":5}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>SFARI_table=table(SFARI_score_for_motif_genes, useNA = &quot;ifany&quot;)


# Define labels, replacing NA with a readable string
labels &lt;- names(SFARI_table)
labels[is.na(labels)] &lt;- &quot;NA&quot;

# Append counts to labels
labels &lt;- paste(labels, &quot;(&quot;, SFARI_table, &quot;)&quot;, sep = &quot;&quot;)

# Create a pie chart
pie(SFARI_table, labels = labels, main = &quot;break-down of motifs genes by SFARI categories&quot;, col = rainbow(length(SFARI_table)))</code></pre>
<p><img src="figure/part01_score_dist.Rmd/unnamed-chunk-6-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-2">
Past versions of unnamed-chunk-6-2.png
</button>
</p>
<div id="fig-unnamed-chunk-6-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/han16/deep_motifs/blob/92c7f1d0d6281802939b4a9363bc7def2b77162b/docs/figure/part01_score_dist.Rmd/unnamed-chunk-6-2.png" target="_blank">92c7f1d</a>
</td>
<td>
han
</td>
<td>
2025-03-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<ul>
<li>with score cutoff 0.5, 447 genes are selected.</li>
</ul>
<pre class="r"><code>score_cutoff=0.5


SFARI_genes=as_tibble(read.csv(file.path(root, &quot;../../../Dataset/SFARI_base/SFARI-Gene_genes_08-19-2024release_09-16-2024export.csv&quot;)))
positive_genes=SFARI_genes %&gt;% filter (gene.score==&quot;1&quot;) %&gt;% dplyr::select(gene.symbol) %&gt;% pull()


forecASD_genes= forecASD_results%&gt;% filter(risk.score&gt;score_cutoff) %&gt;% dplyr::select(gene) %&gt;% pull()



SFARI_score_for_forecASD_genes &lt;- ifelse(
  forecASD_genes %in% SFARI_genes$gene.symbol,
  SFARI_genes$gene.score[match(forecASD_genes, SFARI_genes$gene.symbol)],
  NA
)

forecASD_results %&gt;% filter(risk.score&gt;score_cutoff) %&gt;% dplyr::select(gene, gene_gencodeV33, chromosome, risk.score)%&gt;% mutate(SFARI_score=SFARI_score_for_forecASD_genes) %&gt;% arrange(desc(risk.score)) %&gt;% 
datatable(extensions = &#39;Buttons&#39;,
            options = list(dom = &#39;Blfrtip&#39;,
                           buttons = c(&#39;copy&#39;, &#39;csv&#39;, &#39;excel&#39;, &#39;pdf&#39;, &#39;print&#39;),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,&quot;All&quot;))))</code></pre>
<div class="datatables html-widget html-fill-item" id="htmlwidget-f30d7598595affc4f25a" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-f30d7598595affc4f25a">{"x":{"filter":"none","vertical":false,"extensions":["Buttons"],"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140","141","142","143","144","145","146","147","148","149","150","151","152","153","154","155","156","157","158","159","160","161","162","163","164","165","166","167","168","169","170","171","172","173","174","175","176","177","178","179","180","181","182","183","184","185","186","187","188","189","190","191","192","193","194","195","196","197","198","199","200","201","202","203","204","205","206","207","208","209","210","211","212","213","214","215","216","217","218","219","220","221","222","223","224","225","226","227","228","229","230","231","232","233","234","235","236","237","238","239","240","241","242","243","244","245","246","247","248","249","250","251","252","253","254","255","256","257","258","259","260","261","262","263","264","265","266","267","268","269","270","271","272","273","274","275","276","277","278","279","280","281","282","283","284","285","286","287","288","289","290","291","292","293","294","295","296","297","298","299","300","301","302","303","304","305","306","307","308","309","310","311","312","313","314","315","316","317","318","319","320","321","322","323","324","325","326","327","328","329","330","331","332","333","334","335","336","337","338","339","340","341","342","343","344","345","346","347","348","349","350","351","352","353","354","355","356","357","358","359","360","361","362","363","364","365","366","367","368","369","370","371","372","373","374","375","376","377","378","379","380","381","382","383","384","385","386","387","388","389","390","391","392","393","394","395","396","397","398","399","400","401","402","403","404","405","406","407","408","409","410","411","412","413","414","415","416","417","418","419","420","421","422","423","424","425","426","427","428","429","430","431","432","433","434","435","436","437","438","439","440","441","442","443","444","445","446","447","448","449","450","451","452","453","454","455","456","457","458","459","460","461","462","463","464","465","466","467","468","469","470","471"],["MED13","NIPBL","NCKAP1","BRD4","PRR12","SYNGAP1","ZC3H4","TRIP12","KMT2E","KDM3B","ASH1L","MED13L","RPRD2","KMT2A","KAT6A","SIN3A","ARID1A","SCN1A","WDFY3","EHMT1","TLE3","NBEA","TSC2","CNOT3","AHDC1","GIGYF2","SUV420H1","PHF21A","NAV1","DYNC1H1","CHD8","WAC","ZNF462","STXBP1","MYT1L","SPTBN1","ANK3","SATB2","TCF20","SRCAP","GATAD2B","BPTF","EP300","ARID1B","ANK2","RALGAPB","TBL1XR1","ANKRD11","ZEB2","BAZ1B","SETBP1","RORB","NRXN3","PTPRD","PHIP","HNRNPU","TLK2","MYCBP2","RAI1","SMARCC2","SON","WHSC1","POGZ","RERE","KMT2C","ADNP","CACNA1E","KCNB1","HECTD4","TNRC6C","SCN2A","CTNNB1","SCN8A","ZMYND8","NR4A2","TULP4","PHF12","PTPN11","SETD1A","MBD5","KDM6B","ZNF292","CSDE1","SRRM1","CHD3","SLC6A1","EBF3","CASZ1","ZNF609","UNC13A","TANC2","PSMD11","PSMD12","PAX5","PAK7","NACC1","DDX6","EPHA4","PPP2R5D","CLASP1","PHF2","KDM2B","TEK","ARHGEF12","CIC","MEIS2","MTOR","ZC3H11A","SOX5","DSCAM","SKI","TBR1","KIAA0232","YY1","CPSF7","NFIA","ATP9A","UBE3A","SETD2","BRAF","DYRK1A","TCF4","ANP32A","FOXP2","SMG7","FEZF2","EFNB2","SPAST","NFIB","HNRNPUL2","SETD5","CUL3","AGAP1","CELF4","ACTB","PSMD6","VEZF1","BRD1","GRIA2","NCOA1","ZBTB21","TM9SF4","ZMIZ1","AUTS2","DLGAP1","PTPRS","NRXN1","PBX1","SOS2","GRIA1","PLXNA1","DLG5","NCOA6","CHAMP1","ADCY1","TAOK1","EIF3G","PTK7","PRR14L","NAA15","BCL11B","PUM2","PLXNB2","TSHZ3","PHF3","TNRC18","ZBTB10","NF1","LDB1","SATB1","MYO5A","CSNK2A1","GAB2","LRRC4C","CAPRIN1","RUNX1T1","TBCEL","ABCE1","BRD3","ZBTB20","BCL11A","BRSK2","SPECC1L","GSK3B","CACNA2D3","GABRB3","UBR1","KIDINS220","NR3C2","DIP2A","FOXN3","LPPR4","SRPR","MKX","IRF2BPL","PEG10","PLXNA2","SOX11","IRF2BP2","SHANK2","ELAVL3","DNM3","GIGYF1","PTEN","KLHL20","FEM1B","ERC1","BICD1","KDM5B","TRIM23","ISLR2","SEZ6","CTTNBP2","MEF2C","VPS13B","GAPDH","KATNAL2","CCSAP","BCKDK","TBCK","HRAS","BCLAF1","ALDH5A1","GFAP","POMGNT1","PAH","DNMT3A","MACF1","NSD1","CACNA1C","DPYSL5","KAT6B","GSE1","PUM1","TRIO","WHSC1L1","NFIX","MARCH6","PPP1R9B","BSN","AKT3","ELAVL2","BMPR2","MAP1A","SMARCA2","CAMSAP2","WNK2","FAM193A","YTHDF3","APC2","KANSL1","TSHZ1","YLPM1","RIMS1","PPM1E","IGF1R","CORO1A","NUP155","FOXG1","ELAVL4","FLRT2","NEUROD6","PRDM8","EPM2AIP1","AP2S1","CERS6","PPM1L","CHL1","IRGQ","PCCB","GNG2","MT3","LZTR1","LRP1","TOP1","CREBBP","SPEN","JMJD1C","HIVEP2","HECTD1","ROBO2","MYH10","KIF1B","PRRC2B","GRIK3","SOGA1","NEDD4L","PRKCE","CELF5","ATF7IP","PRDM2","RAPGEF1","DCLK1","CCND2","CAMSAP1","KHDRBS1","CELSR3","MTMR4","CELSR2","NMNAT2","CPEB4","ZCCHC14","SRGAP2","LIMCH1","KBTBD6","NEUROD2","RBFOX1","SLC22A23","TMCC1","USP22","STOX2","CSRNP3","CCP110","SOX12","PGM2L1","PTGFRN","LMO3","CNR1","ADRA2A","NRGN","TNIK","ELFN2","USP7","CTCF","KDM4A","CELF2","ZKSCAN1","NEO1","FAM115A","SBK1","CHERP","MN1","LMNB2","SOBP","RNF38","ZNF445","LRP8","LATS1","LMTK2","MEGF9","RND3","SCD","SOGA3","PPP1R15B","GNB4","RGMB","MPPED1","RPRM","PTPRF","C20orf112","PLEKHM3","KDM4B","SLIT1","ZRANB1","MSL2","GRIN2B","CADM2","DDX17","PPP1R10","DENND5B","SAP130","LMTK3","CTNND1","SRGAP3","TP53BP1","ADD2","UNC5D","ATP8A1","SEZ6L","NOLC1","LHFPL4","KBTBD11","KDM2A","RAPGEF2","LPGAT1","ARHGAP33","SYT13","NUAK1","MCM4","C5orf24","SCAF8","SIN3B","TOX","EPHB1","ZNF770","PDZD2","EML1","AMOTL2","KIF21A","SCN3B","UBXN7","POU3F2","KIF21B","PSD3","ROBO1","RYBP","RSF1","MLLT3","XPR1","APC","CKAP5","ABL1","JAKMIP2","ARHGEF7","PLXNA4","ARRDC3","BICD2","SLC38A1","SOX4","SEMA7A","SPIRE1","ALDOC","DACT1","SUPT16H","MKL2","CAMKV","ZNF532","APBA1","SH3GL2","MBP","VCAN","TTYH3","HNRNPH3","ELMOD1","POU3F3","NEUROD1","PCM1","CDH13","UBXN2B","KCNG1","TTC28","PLK2","SRRM4","DCAF16","IGFBPL1","SHANK3","ATXN2L","TET3","VCPIP1","FBXO11","GABRB2","NCK2","PIP5K1C","GTF3C4","PEG3","B4GALT5","DOCK3","HERC1","STT3B","ZYG11B","TRIM2","CDH2","NGEF","MEX3C","KIF3B","RCOR2","NES","RIMS4","NKAIN2","SMC3","CNOT6","TMSB10","EPC1","ZNF512B","FBXW7","KLF6","SLC35F1","GNA13","OSBPL8","KMT2D","NCOR1"],["MED13","NIPBL","NCKAP1","BRD4","PRR12","SYNGAP1","ZC3H4","TRIP12","KMT2E","KDM3B","ASH1L","MED13L","RPRD2","KMT2A","KAT6A","SIN3A","ARID1A","SCN1A","WDFY3","EHMT1","TLE3","NBEA","TSC2","CNOT3","AHDC1","GIGYF2","KMT5B","PHF21A","NAV1","DYNC1H1","CHD8","WAC","ZNF462","STXBP1","MYT1L","SPTBN1","ANK3","SATB2","TCF20","SRCAP","GATAD2B","BPTF","EP300","ARID1B","ANK2","RALGAPB","TBL1XR1","ANKRD11","ZEB2","BAZ1B","SETBP1","RORB","NRXN3","PTPRD","PHIP","HNRNPU","TLK2","MYCBP2","RAI1","SMARCC2","SON","NSD2","POGZ","RERE","KMT2C","ADNP","CACNA1E","KCNB1","HECTD4","TNRC6C","SCN2A","CTNNB1","SCN8A","ZMYND8","NR4A2","TULP4","PHF12","PTPN11","SETD1A","MBD5","KDM6B","ZNF292","CSDE1","SRRM1","CHD3","SLC6A1","EBF3","CASZ1","ZNF609","UNC13A","TANC2","PSMD11","PSMD12","PAX5","PAK5","NACC1","DDX6","EPHA4","PPP2R5D","CLASP1","PHF2","KDM2B","TEK","ARHGEF12","CIC","MEIS2","MTOR","ZC3H11A","SOX5","DSCAM","SKI","TBR1","KIAA0232","YY1","CPSF7","NFIA","ATP9A","UBE3A","SETD2","BRAF","DYRK1A","TCF4","ANP32A","FOXP2","SMG7","FEZF2","EFNB2","SPAST","NFIB","HNRNPUL2","SETD5","CUL3","AGAP1","CELF4","ACTB","PSMD6","VEZF1","BRD1","GRIA2","NCOA1","ZBTB21","TM9SF4","ZMIZ1","AUTS2","DLGAP1","PTPRS","NRXN1","PBX1","SOS2","GRIA1","PLXNA1","DLG5","NCOA6","CHAMP1","ADCY1","TAOK1","EIF3G","PTK7","PRR14L","NAA15","BCL11B","PUM2","PLXNB2","TSHZ3","PHF3","TNRC18","ZBTB10","NF1","LDB1","SATB1","MYO5A","CSNK2A1","GAB2","LRRC4C","CAPRIN1","RUNX1T1","TBCEL","ABCE1","BRD3","ZBTB20","BCL11A","BRSK2","SPECC1L","GSK3B","CACNA2D3","GABRB3","UBR1","KIDINS220","NR3C2","DIP2A","FOXN3","PLPPR4","SRPRA","MKX","IRF2BPL","PEG10","PLXNA2","SOX11","IRF2BP2","SHANK2","ELAVL3","DNM3","GIGYF1","PTEN","KLHL20","FEM1B","ERC1","BICD1","KDM5B","TRIM23","ISLR2","SEZ6","CTTNBP2","MEF2C","VPS13B","GAPDH","KATNAL2","CCSAP","BCKDK","TBCK","HRAS","BCLAF1","ALDH5A1","GFAP","POMGNT1","PAH","DNMT3A","MACF1","NSD1","CACNA1C","DPYSL5","KAT6B","GSE1","PUM1","TRIO","NSD3","NFIX","MARCHF6","PPP1R9B","BSN","AKT3","ELAVL2","BMPR2","MAP1A","SMARCA2","CAMSAP2","WNK2","FAM193A","YTHDF3","APC2","KANSL1","TSHZ1","YLPM1","RIMS1","PPM1E","IGF1R","CORO1A","NUP155","FOXG1","ELAVL4","FLRT2","NEUROD6","PRDM8","EPM2AIP1","AP2S1","CERS6","PPM1L","CHL1","IRGQ","PCCB","GNG2","MT3","LZTR1","LRP1","TOP1","CREBBP","SPEN","JMJD1C","HIVEP2","HECTD1","ROBO2","MYH10","KIF1B","PRRC2B","GRIK3","SOGA1","NEDD4L","PRKCE","CELF5","ATF7IP","PRDM2","RAPGEF1","DCLK1","CCND2","CAMSAP1","KHDRBS1","CELSR3","MTMR4","CELSR2","NMNAT2","CPEB4","ZCCHC14","SRGAP2","LIMCH1","KBTBD6","NEUROD2","RBFOX1","SLC22A23","TMCC1","USP22","STOX2","CSRNP3","CCP110","SOX12","PGM2L1","PTGFRN","LMO3","CNR1","ADRA2A","NRGN","TNIK","ELFN2","USP7","CTCF","KDM4A","CELF2","ZKSCAN1","NEO1","TCAF1","SBK1","CHERP","MN1","LMNB2","SOBP","RNF38","ZNF445","LRP8","LATS1","LMTK2","MEGF9","RND3","SCD","SOGA3","PPP1R15B","GNB4","RGMB","MPPED1","RPRM","PTPRF","NOL4L","PLEKHM3","KDM4B","SLIT1","ZRANB1","MSL2","GRIN2B","CADM2","DDX17","PPP1R10","DENND5B","SAP130","LMTK3","CTNND1","SRGAP3","TP53BP1","ADD2","UNC5D","ATP8A1","SEZ6L","NOLC1","LHFPL4","KBTBD11","KDM2A","RAPGEF2","LPGAT1","ARHGAP33","SYT13","NUAK1","MCM4","C5orf24","SCAF8","SIN3B","TOX","EPHB1","ZNF770","PDZD2","EML1","AMOTL2","KIF21A","SCN3B","UBXN7","POU3F2","KIF21B","PSD3","ROBO1","RYBP","RSF1","MLLT3","XPR1","APC","CKAP5","ABL1","JAKMIP2","ARHGEF7","PLXNA4","ARRDC3","BICD2","SLC38A1","SOX4","SEMA7A","SPIRE1","ALDOC","DACT1","SUPT16H","MRTFB","CAMKV","ZNF532","APBA1","SH3GL2","MBP","VCAN","TTYH3","HNRNPH3","ELMOD1","POU3F3","NEUROD1","PCM1","CDH13","UBXN2B","KCNG1","TTC28","PLK2","SRRM4","DCAF16","IGFBPL1","SHANK3","ATXN2L","TET3","VCPIP1","FBXO11","GABRB2","NCK2","PIP5K1C","GTF3C4","PEG3","B4GALT5","DOCK3","HERC1","STT3B","ZYG11B","TRIM2","CDH2","NGEF","MEX3C","KIF3B","RCOR2","NES","RIMS4","NKAIN2","SMC3","CNOT6","TMSB10","EPC1","ZNF512B","FBXW7","KLF6","SLC35F1","GNA13","OSBPL8","KMT2D","NCOR1"],[17,5,2,19,19,6,19,2,7,5,1,12,1,11,8,15,1,2,4,9,15,13,16,19,1,2,11,11,1,14,14,10,9,9,2,2,10,2,22,16,1,17,22,6,4,20,3,16,2,7,18,9,14,9,6,1,17,13,17,12,21,4,1,1,7,20,1,20,12,17,2,3,12,20,2,6,17,12,16,2,17,6,1,1,17,3,10,1,15,19,17,17,17,9,20,19,11,2,6,2,9,12,9,11,19,15,1,1,12,21,1,2,4,14,11,1,20,15,3,7,21,18,15,7,1,3,13,2,9,11,3,2,2,18,7,3,17,22,4,2,21,20,10,7,18,19,2,1,14,5,3,10,20,13,7,17,19,6,22,4,14,2,22,19,6,7,8,17,10,3,15,20,11,11,11,8,11,4,9,3,2,11,22,3,3,15,15,2,4,21,14,1,11,10,14,7,1,2,1,11,19,1,7,10,1,15,12,12,1,5,15,17,7,5,8,12,18,1,16,4,11,6,6,17,1,12,2,1,5,12,2,10,16,1,5,8,19,5,17,3,1,9,2,15,9,1,9,4,8,19,17,18,14,6,17,15,16,5,14,1,14,7,4,3,19,2,3,3,19,3,14,16,22,12,20,16,1,10,6,14,3,17,1,9,1,20,18,2,19,12,1,9,13,12,9,1,3,17,1,1,5,16,1,4,13,17,16,6,3,17,4,2,16,20,11,1,12,6,10,11,3,22,16,16,1,10,7,15,7,16,19,22,19,6,9,3,1,6,7,9,2,10,6,1,3,5,22,2,1,20,2,19,10,10,3,12,3,22,6,12,2,19,11,3,15,2,8,4,22,10,3,8,11,4,1,19,11,12,8,5,6,19,8,3,15,5,14,3,12,11,3,6,1,8,3,3,11,9,1,5,11,9,5,13,7,5,9,12,6,15,18,17,14,14,16,3,18,9,9,18,5,7,10,11,2,2,8,16,8,20,22,5,12,4,9,22,16,2,8,2,5,2,19,9,19,20,3,15,3,1,4,18,2,18,20,11,1,20,6,10,5,2,10,20,4,10,6,17,12,12,17],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0.99,0.99,0.99,0.99,0.99,0.99,0.99,0.99,0.99,0.99,0.99,0.99,0.99,0.99,0.99,0.99,0.99,0.99,0.99,0.99,0.99,0.99,0.99,0.99,0.99,0.99,0.99,0.99,0.99,0.99,0.99,0.99,0.99,0.99,0.99,0.99,0.99,0.99,0.99,0.99,0.99,0.99,0.99,0.99,0.99,0.99,0.98,0.98,0.98,0.98,0.98,0.98,0.98,0.98,0.98,0.98,0.98,0.98,0.98,0.98,0.98,0.98,0.98,0.98,0.98,0.98,0.98,0.98,0.98,0.98,0.98,0.98,0.98,0.98,0.98,0.98,0.98,0.98,0.98,0.98,0.98,0.98,0.98,0.98,0.98,0.98,0.98,0.98,0.98,0.98,0.98,0.98,0.98,0.97,0.97,0.96,0.96,0.96,0.96,0.96,0.96,0.96,0.96,0.95,0.95,0.95,0.95,0.95,0.95,0.95,0.95,0.95,0.95,0.95,0.95,0.95,0.95,0.95,0.95,0.95,0.95,0.93,0.93,0.89,0.87,0.87,0.87,0.87,0.86,0.86,0.85,0.85,0.85,0.85,0.85,0.85,0.85,0.85,0.85,0.85,0.85,0.85,0.85,0.85,0.85,0.84,0.84,0.84,0.84,0.84,0.84,0.84,0.84,0.83,0.83,0.83,0.83,0.83,0.83,0.83,0.83,0.82,0.82,0.8100000000000001,0.8100000000000001,0.8100000000000001,0.8100000000000001,0.8100000000000001,0.8100000000000001,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.79,0.78,0.78,0.78,0.78,0.78,0.78,0.78,0.77,0.77,0.77,0.77,0.77,0.77,0.77,0.77,0.77,0.77,0.76,0.76,0.76,0.76,0.76,0.6899999999999999,0.6899999999999999,0.68,0.68,0.67,0.67,0.64,0.63,0.63,0.63,0.62,0.62,0.62,0.62,0.62,0.62,0.62,0.62,0.62,0.62,0.62,0.62,0.62,0.62,0.61,0.61,0.61,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.57,0.51],[1,1,1,2,1,1,2,1,1,1,1,1,null,1,2,1,null,1,1,1,1,1,1,1,1,1,null,1,null,1,1,1,1,1,1,2,1,2,1,1,null,null,1,1,1,1,1,1,null,null,1,1,1,2,1,1,1,1,1,1,1,null,1,1,1,1,1,1,1,2,1,1,1,1,1,null,1,1,1,1,1,1,1,null,1,1,1,1,null,2,1,1,1,1,null,1,null,null,1,2,1,1,1,null,1,1,1,2,1,1,1,1,1,1,1,2,null,1,1,1,1,1,1,1,null,2,null,1,2,2,1,1,2,1,1,1,1,null,1,1,1,1,2,1,2,null,1,2,1,2,null,null,null,1,null,1,1,1,1,1,3,null,null,1,1,null,null,1,1,1,2,1,null,1,1,1,1,1,null,1,1,1,null,null,1,1,1,null,1,1,null,null,null,1,1,null,null,null,null,1,1,null,1,1,1,null,null,null,1,1,null,null,2,1,1,null,1,null,1,1,1,null,1,1,1,1,1,3,1,1,null,3,null,null,1,null,null,null,1,null,null,2,null,1,1,null,null,null,null,null,1,1,null,1,null,null,1,1,1,null,null,null,null,null,1,null,null,null,null,1,null,null,1,2,null,1,2,2,1,null,2,2,null,null,2,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,2,null,null,null,null,null,2,null,null,null,null,null,null,null,null,null,null,2,null,null,null,null,2,1,null,2,null,2,null,null,null,null,null,null,2,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,2,null,null,3,1,2,null,null,null,null,3,null,2,null,null,3,null,null,null,null,null,3,null,null,null,null,2,2,null,null,2,null,3,null,null,null,null,null,null,null,null,null,2,null,null,null,null,null,null,null,null,null,null,2,null,null,null,null,null,null,null,null,2,null,null,null,null,null,null,null,null,null,null,null,null,2,2,null,null,null,null,null,null,null,1,null,null,null,2,1,null,null,null,null,null,null,2,null,null,null,3,null,null,null,null,null,null,null,2,null,null,null,null,null,null,null,null,null,null,2]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>gene<\/th>\n      <th>gene_gencodeV33<\/th>\n      <th>chromosome<\/th>\n      <th>risk.score<\/th>\n      <th>SFARI_score<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["copy","csv","excel","pdf","print"],"lengthMenu":[[10,25,50,-1],["10","25","50","All"]],"columnDefs":[{"className":"dt-right","targets":[3,4,5]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"gene","targets":1},{"name":"gene_gencodeV33","targets":2},{"name":"chromosome","targets":3},{"name":"risk.score","targets":4},{"name":"SFARI_score","targets":5}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>SFARI_table=table(SFARI_score_for_forecASD_genes, useNA = &quot;ifany&quot;)


# Define labels, replacing NA with a readable string
labels &lt;- names(SFARI_table)
labels[is.na(labels)] &lt;- &quot;NA&quot;

# Append counts to labels
labels &lt;- paste(labels, &quot;(&quot;, SFARI_table, &quot;)&quot;, sep = &quot;&quot;)

# Create a pie chart
pie(SFARI_table, labels = labels, main = &quot;break-down of forecASD genes by SFARI categories&quot;, col = rainbow(length(SFARI_table)))</code></pre>
<p><img src="figure/part01_score_dist.Rmd/unnamed-chunk-7-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-2">
Past versions of unnamed-chunk-7-2.png
</button>
</p>
<div id="fig-unnamed-chunk-7-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/han16/deep_motifs/blob/92c7f1d0d6281802939b4a9363bc7def2b77162b/docs/figure/part01_score_dist.Rmd/unnamed-chunk-7-2.png" target="_blank">92c7f1d</a>
</td>
<td>
han
</td>
<td>
2025-03-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<ul>
<li>with cutoff 0.5, forecASD identified 471 risk genes.</li>
</ul>
</div>
<div id="deep-motifs-new-genes-not-in-sfari-cate-1"
class="section level2">
<h2>deep-motifs new genes not in SFARI cate 1</h2>
<pre class="r"><code>SFARI_genes=as_tibble(read.csv(file.path(root, &quot;../../../Dataset/SFARI_base/SFARI-Gene_genes_08-19-2024release_09-16-2024export.csv&quot;)))

positive_genes=SFARI_genes %&gt;% filter (gene.score==&quot;1&quot;) %&gt;% dplyr::select(gene.symbol) %&gt;% pull()
positive_genes=intersect(positive_genes, result_round1$gene)

score_cutoff=0.5

motifs_new_genes=result_round1 %&gt;% filter(RISK_PROBABILITY_FUSION&gt;score_cutoff) %&gt;% dplyr::select(gene)%&gt;% filter (! gene %in% positive_genes) %&gt;% pull()

length(result_round1 %&gt;% filter(RISK_PROBABILITY_FUSION&gt;score_cutoff) %&gt;% dplyr::select(gene)%&gt;% filter (gene %in% positive_genes) %&gt;% pull())</code></pre>
<pre><code>[1] 206</code></pre>
<pre class="r"><code>SFARI_score_for_motif_new_genes &lt;- ifelse(
  motifs_new_genes %in% SFARI_genes$gene.symbol,
  SFARI_genes$gene.score[match(motifs_new_genes, SFARI_genes$gene.symbol)],
  NA
)

sum(is.na(SFARI_score_for_motif_new_genes))</code></pre>
<pre><code>[1] 154</code></pre>
<pre class="r"><code>result_round1 %&gt;% filter(RISK_PROBABILITY_FUSION&gt;score_cutoff) %&gt;% dplyr::select(gene, gene_gencodeV33, chromosome, RISK_PROBABILITY_FUSION)%&gt;% filter (! gene %in% positive_genes)%&gt;% mutate (SFARI_score=SFARI_score_for_motif_new_genes) %&gt;%arrange(desc(RISK_PROBABILITY_FUSION)) %&gt;% 
datatable(extensions = &#39;Buttons&#39;,
          caption = &quot;new ASD genes (not in SFARI cate 1)&quot;, 
            options = list(dom = &#39;Blfrtip&#39;,
                           buttons = c(&#39;copy&#39;, &#39;csv&#39;, &#39;excel&#39;, &#39;pdf&#39;, &#39;print&#39;),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,&quot;All&quot;))))</code></pre>
<div class="datatables html-widget html-fill-item" id="htmlwidget-b85bf888dc62471906a9" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-b85bf888dc62471906a9">{"x":{"filter":"none","vertical":false,"extensions":["Buttons"],"caption":"<caption>new ASD genes (not in SFARI cate 1)<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140","141","142","143","144","145","146","147","148","149","150","151","152","153","154","155","156","157","158","159","160","161","162","163","164","165","166","167","168","169","170","171","172","173","174","175","176","177","178","179","180","181","182","183","184","185","186","187","188","189","190","191","192","193","194","195","196","197","198","199","200","201","202","203","204","205","206","207","208","209","210","211","212","213","214","215","216","217","218","219","220","221","222","223","224","225","226","227","228","229","230","231","232","233","234","235","236","237","238","239","240","241"],["ATP2B2","SUV420H1","ZMYND11","QRICH1","HNRNPUL2","DMWD","PBX1","AGAP2","CYB5R3","CADM2","APBB1","PCDH10","PEG10","ARID1A","SPEN","KIF5C","HNRNPD","SRPR","SATB2","CAMSAP1","CNTN1","GATAD2B","SYNCRIP","BRD4","GSK3B","SYT1","SPRY2","SRGAP3","FUBP3","MARK2","SCAF1","LRRC4","KIF3C","CELF2","UBR5","GGNBP2","SPTBN1","HECTD1","HDAC4","PSD","CSNK1E","SETD1B","DEDD","NFIX","CYFIP2","PTPRF","LMTK3","EXT1","TERF2","RUSC1","CBX1","C16orf72","KCNQ2","PCLO","NAV3","PLEKHO1","LZTS3","EIF3D","EHBP1","CLPTM1","TNRC6C","CLASP1","TNPO3","PTPRD","MIB1","MLXIP","ATRN","TBC1D14","GRIK2","TSC22D1","RTF1","PUM1","PNN","SSR2","RAB11FIP4","PLXNA2","AP2M1","KLHDC10","DNAJC5","TARDBP","PLXNB1","PSMA3","ZMYM2","CGREF1","HDAC5","TAF4","INO80D","MBD6","NPTN","THRB","PUM2","CASKIN1","MACF1","DIP2C","EFTUD2","G3BP1","PGM2L1","SMG7","CAMK2B","DDB1","ATP1A3","AHNAK","SMARCAD1","CUX2","TNRC6B","PCDHAC2","CPD","ELAVL2","GRN","KAT6A","NOVA2","ZEB2","SMARCE1","KLC2","MRFAP1","BAZ1B","CHD6","FEZF2","EIF3B","PRH1","TESK1","RAB2A","TCERG1","PAPOLG","KIDINS220","ARHGEF18","VGF","KIF1B","PLXNA1","MAZ","SLIT1","ADRM1","EIF4G3","ZBTB18","NDST1","NEO1","RBM12","ZC3H14","FLRT2","DOCK3","ZNF608","DGKD","TWF2","DLL1","BRPF3","NFIL3","GAPVD1","SEC24B","DPP6","ITSN1","FAM120A","MTMR3","MYH10","WHSC1","ARHGEF12","RBBP5","KIAA0368","KLF7","MSL2","ITGB4","YLPM1","PPP1CC","CTNND1","ADCY1","PLEKHM3","DLGAP4","ILF3","PPM1E","CXXC5","TNIK","KDM4A","BPTF","TSPAN14","KDM4B","ZNF865","SP3","CBX4","FBXO11","PPIG","SLC17A7","RPL15","AKAP9","DVL3","USP5","EDF1","PAK7","HCN1","DYRK2","SCAF4","SFPQ","HUNK","CAMK2A","RNF220","ZNF423","ARID5B","AGO4","ARIH1","PRDM15","DNAH1","BRINP1","FUS","UNC13A","DDX6","SMG1","C11orf30","CECR2","ZMYM4","ATP1B1","MAN1A1","MYO18A","SEC24D","ATXN2L","SYT13","MGEA5","ZNF536","MYO5A","PIK3CA","ASAP1","APC2","NDRG4","EIF4E","LINGO1","YWHAG","GRAMD1B","COBL","SCAPER","ARHGAP32","CLTC","PLXNB2","KCNH7","SNRNP200","LEO1","TRIM24","GRIK3","MTCH1","APLP2","BIRC6","SRCIN1","ZMIZ1","CDK13","RNF146"],["ATP2B2","KMT5B","ZMYND11","QRICH1","HNRNPUL2","DMWD","PBX1","AGAP2","CYB5R3","CADM2","APBB1","PCDH10","PEG10","ARID1A","SPEN","KIF5C","HNRNPD","SRPRA","SATB2","CAMSAP1","CNTN1","GATAD2B","SYNCRIP","BRD4","GSK3B","SYT1","SPRY2","SRGAP3","FUBP3","MARK2","SCAF1","LRRC4","KIF3C","CELF2","UBR5","GGNBP2","SPTBN1","HECTD1","HDAC4","PSD","CSNK1E","SETD1B","DEDD","NFIX","CYFIP2","PTPRF","LMTK3","EXT1","TERF2","RUSC1","CBX1","C16orf72","KCNQ2","PCLO","NAV3","PLEKHO1","LZTS3","EIF3D","EHBP1","CLPTM1","TNRC6C","CLASP1","TNPO3","PTPRD","MIB1","MLXIP","ATRN","TBC1D14","GRIK2","TSC22D1","RTF1","PUM1","PNN","SSR2","RAB11FIP4","PLXNA2","AP2M1","KLHDC10","DNAJC5","TARDBP","PLXNB1","PSMA3","ZMYM2","CGREF1","HDAC5","TAF4","INO80D","MBD6","NPTN","THRB","PUM2","CASKIN1","MACF1","DIP2C","EFTUD2","G3BP1","PGM2L1","SMG7","CAMK2B","DDB1","ATP1A3","AHNAK","SMARCAD1","CUX2","TNRC6B","PCDHAC2","CPD","ELAVL2","GRN","KAT6A","NOVA2","ZEB2","SMARCE1","KLC2","MRFAP1","BAZ1B","CHD6","FEZF2","EIF3B","PRH1","TESK1","RAB2A","TCERG1","PAPOLG","KIDINS220","ARHGEF18","VGF","KIF1B","PLXNA1","MAZ","SLIT1","ADRM1","EIF4G3","ZBTB18","NDST1","NEO1","RBM12","ZC3H14","FLRT2","DOCK3","ZNF608","DGKD","TWF2","DLL1","BRPF3","NFIL3","GAPVD1","SEC24B","DPP6","ITSN1","FAM120A","MTMR3","MYH10","NSD2","ARHGEF12","RBBP5","ECPAS","KLF7","MSL2","ITGB4","YLPM1","PPP1CC","CTNND1","ADCY1","PLEKHM3","DLGAP4","ILF3","PPM1E","CXXC5","TNIK","KDM4A","BPTF","TSPAN14","KDM4B","ZNF865","SP3","CBX4","FBXO11","PPIG","SLC17A7","RPL15","AKAP9","DVL3","USP5","EDF1","PAK5","HCN1","DYRK2","SCAF4","SFPQ","HUNK","CAMK2A","RNF220","ZNF423","ARID5B","AGO4","ARIH1","PRDM15","DNAH1","BRINP1","FUS","UNC13A","DDX6","SMG1","EMSY","CECR2","ZMYM4","ATP1B1","MAN1A1","MYO18A","SEC24D","ATXN2L","SYT13","OGA","ZNF536","MYO5A","PIK3CA","ASAP1","APC2","NDRG4","EIF4E","LINGO1","YWHAG","GRAMD1B","COBL","SCAPER","ARHGAP32","CLTC","PLXNB2","KCNH7","SNRNP200","LEO1","TRIM24","GRIK3","MTCH1","APLP2","BIRC6","SRCIN1","ZMIZ1","CDK13","RNF146"],[3,11,10,3,11,19,1,12,22,3,11,4,7,1,1,2,4,11,2,9,12,1,6,19,3,12,13,3,9,11,19,7,2,10,8,17,2,14,2,10,22,12,1,19,5,1,19,8,16,1,17,16,20,7,12,1,20,22,2,19,17,2,7,9,18,12,20,4,6,13,15,1,14,1,17,1,3,7,20,1,3,14,13,2,17,20,2,12,15,3,2,16,1,10,17,5,11,1,7,11,19,11,4,12,22,5,17,9,17,8,19,2,17,11,4,7,20,3,7,12,9,8,5,2,2,19,7,1,3,16,10,20,1,1,5,15,20,14,14,3,5,2,3,6,6,9,9,4,7,21,9,22,17,4,11,1,9,2,3,17,14,12,11,7,2,20,19,17,5,3,1,17,10,19,19,2,17,2,2,19,3,7,3,12,9,20,5,12,21,1,21,5,1,16,10,1,15,21,3,9,16,19,11,16,11,22,1,1,6,17,4,16,11,10,19,15,3,8,19,16,4,15,7,11,7,15,11,17,22,2,2,15,7,1,6,11,2,17,10,7,6],[0.9976425,0.9976225,0.997605,0.9976036,0.99760264,0.9975853,0.99758136,0.9975797,0.9975534700000001,0.9975489400000001,0.99753255,0.9975317,0.9975246,0.99751645,0.9975128,0.99751073,0.9975106,0.9975088,0.9974927,0.9974885999999999,0.9974737,0.9974734,0.99746716,0.9974626,0.9974615,0.9974487399999999,0.99744207,0.9974412,0.99742126,0.9974209000000001,0.99740756,0.9973963,0.99738514,0.9973749,0.99736965,0.9973661,0.9973641600000001,0.9973611999999999,0.99732953,0.9973213,0.99731904,0.9972288,0.9972251,0.99719965,0.9971922,0.9971021,0.9970583,0.99704504,0.9969848,0.9969562,0.9969473,0.99693805,0.99691933,0.9968859,0.9968812,0.9967739,0.99653655,0.9963582,0.99623567,0.996078,0.99575585,0.99573725,0.99551964,0.9955079,0.99542284,0.99540436,0.9951371,0.99512947,0.9943360999999999,0.9941993,0.9938939,0.99364316,0.9936242,0.9934367,0.9923889,0.99202365,0.9914587,0.99006927,0.9898326,0.9894129,0.98937434,0.9884526,0.98747504,0.98522633,0.98061836,0.9803851,0.9763632,0.9757284000000001,0.9751538,0.97240716,0.9681858,0.9632267,0.9621851,0.9564083,0.93725985,0.93393743,0.9247719599999999,0.9152954,0.9096673,0.90850526,0.90525407,0.9036007,0.90066594,0.88389426,0.883185,0.8825943000000001,0.8750067,0.87495214,0.8722224,0.86903125,0.8623269,0.8595975,0.84811753,0.8476205,0.83982515,0.8385941,0.8374852,0.831995,0.82576513,0.8217886,0.8154192,0.8105699,0.8077156,0.806291,0.8025869,0.8022561,0.7999085,0.7985517,0.7983545,0.79773915,0.7962233,0.793747,0.7913416599999999,0.7900816000000001,0.789238,0.78752154,0.7833338,0.77928776,0.7775445,0.77367485,0.77015924,0.76484644,0.7631656999999999,0.7587752,0.7572113,0.75307906,0.75044113,0.7484987400000001,0.738256,0.73808897,0.7380601,0.73768866,0.7338079,0.73214364,0.72194403,0.7201621499999999,0.7182785,0.717596,0.7090737,0.7085911,0.6906365,0.6903677,0.68569696,0.67742205,0.6760627,0.6758707,0.6632365,0.6623924,0.6615807,0.66002524,0.65415174,0.6540898000000001,0.6503447999999999,0.64766806,0.6402124,0.6281816,0.6253172,0.6253072,0.6243320999999999,0.6228286,0.6214806,0.6195731,0.6174718,0.6169258399999999,0.61662644,0.6143726,0.6129464999999999,0.6126493,0.6117099499999999,0.6075275999999999,0.6065609,0.60643893,0.6028396499999999,0.6020068,0.6013747,0.60107076,0.5994815,0.5989662,0.5985886,0.5977902,0.59751487,0.5972332,0.5958844,0.5943432,0.59384537,0.5921861,0.590606,0.58835715,0.5882396,0.5877068,0.58762306,0.5872287,0.5870297,0.5870274,0.5865000500000001,0.5855987,0.58554536,0.5831045500000001,0.5823537,0.5810206999999999,0.57459724,0.57278717,0.5725951,0.5631597,0.5599023,0.5554427,0.54965913,0.5461494,0.5454005,0.5435750499999999,0.5382606,0.5285813,0.52761304,0.52512825,0.5229856000000001,0.51935685,0.51373035,0.5079747,0.5041013,0.50351256,0.5007688],[2,null,2,2,2,2,2,2,null,2,2,2,null,null,2,2,2,null,2,null,null,null,2,2,null,null,2,2,null,2,2,2,null,2,2,2,2,null,2,null,2,2,null,null,null,null,3,2,2,null,3,null,2,2,2,null,null,null,null,null,2,2,3,2,2,null,null,null,2,null,null,null,null,null,null,null,2,null,null,null,2,null,2,null,null,2,null,2,3,null,null,2,3,2,null,null,null,null,null,null,2,2,null,2,2,2,null,2,null,2,null,null,null,null,null,null,null,2,null,null,null,2,null,2,null,null,null,null,null,null,null,null,null,null,null,2,null,null,null,null,null,null,null,2,null,null,null,null,2,2,null,null,2,null,null,2,null,3,3,null,null,null,null,null,null,null,null,null,null,null,null,null,null,2,null,null,null,2,null,null,null,2,2,null,null,null,3,null,2,null,null,2,null,null,null,2,null,null,null,null,null,2,null,null,null,2,null,null,null,null,null,null,null,null,null,2,3,null,null,null,2,null,3,null,null,null,2,null,null,null,null,2,null,2,null,null,2,null,2,null,null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>gene<\/th>\n      <th>gene_gencodeV33<\/th>\n      <th>chromosome<\/th>\n      <th>RISK_PROBABILITY_FUSION<\/th>\n      <th>SFARI_score<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["copy","csv","excel","pdf","print"],"lengthMenu":[[10,25,50,-1],["10","25","50","All"]],"columnDefs":[{"className":"dt-right","targets":[3,4,5]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"gene","targets":1},{"name":"gene_gencodeV33","targets":2},{"name":"chromosome","targets":3},{"name":"RISK_PROBABILITY_FUSION","targets":4},{"name":"SFARI_score","targets":5}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<ul>
<li><p>233 genes are in SFARI category 1, but 208 are used and appear in
final result. Among 208 genes, 206 are captured by motifs, being among
447 genes, and 2 genes (“ACTB”, “PCCB”) are missed.</p></li>
<li><p>among 241 new ASD genes (not in cate 1), 154 are not in SFARI
list at all.</p></li>
</ul>
</div>
<div id="ranking-of-deep-motifs-genes-vs-forecasd-genes"
class="section level2">
<h2>ranking of deep-motifs genes vs forecASD genes</h2>
<pre class="r"><code>total_risk_genes=unique(c(forecASD_genes, motifs_genes))
result_round1_sorted=result_round1 %&gt;% arrange(desc(RISK_PROBABILITY_FUSION))
forecASD_results_sorted=forecASD_results %&gt;% arrange(desc(risk.score))

total_risk_genes_rank1 &lt;- match(total_risk_genes, result_round1_sorted$gene)
total_risk_genes_rank2=match(total_risk_genes, forecASD_results_sorted$gene)


# Combine into a data frame
df &lt;- data.frame(
  Gene = total_risk_genes,
  Rank1_forecASD = total_risk_genes_rank2,
  Rank2_deep_motifs = total_risk_genes_rank1
)

# Reshape data for plotting
df_long &lt;- df %&gt;%
  tidyr::pivot_longer(cols = c(Rank1_forecASD, Rank2_deep_motifs), names_to = &quot;Ranking&quot;, values_to = &quot;Rank&quot;)

# Plot the slope graph with gene labels on both sides
ggplot(df_long, aes(x = Ranking, y = Rank, group = Gene)) +
  geom_line(aes(color = Gene), size = 1) +  # Connecting lines
  geom_point(size = 1) +                    # Points at each ranking
  # Add gene labels on the left side for Rank1
  geom_text(data = df_long %&gt;% filter(Ranking == &quot;Rank1_forecASD&quot;), 
            aes(label = Gene), 
            hjust = 1.2, size = 2) +
  # Add gene labels on the right side for Rank2
  geom_text(data = df_long %&gt;% filter(Ranking == &quot;Rank2_deep_motifs&quot;), 
            aes(label = Gene), 
            hjust = -0.2, size = 2) +
  scale_y_reverse() +                       # Reverse y-axis for ranking
  labs(title = &quot;&quot;, x = &quot;&quot;, y = &quot;Rank&quot;) +
  theme_minimal() +
  theme(legend.position = &quot;none&quot;)            # Remove legend for clarity</code></pre>
<p><img src="figure/part01_score_dist.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/han16/deep_motifs/blob/92c7f1d0d6281802939b4a9363bc7def2b77162b/docs/figure/part01_score_dist.Rmd/unnamed-chunk-9-1.png" target="_blank">92c7f1d</a>
</td>
<td>
han
</td>
<td>
2025-03-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Perform the test
result &lt;- wilcox.test(total_risk_genes_rank1, total_risk_genes_rank2, paired = TRUE, alternative = &quot;two.sided&quot;)

# Display the results
print(result)</code></pre>
<pre><code>
    Wilcoxon signed rank test with continuity correction

data:  total_risk_genes_rank1 and total_risk_genes_rank2
V = 131438, p-value = 0.001744
alternative hypothesis: true location shift is not equal to 0</code></pre>
</div>
<div id="genes-overlapping-among-deep-motifs-forecasd-and-sfari"
class="section level2">
<h2>genes overlapping among deep motifs, forecASD, and SFARI</h2>
<pre class="r"><code>x=list(
  MOTIFs_gene=motifs_genes,
  #Autism_gene=as_tibble(neuro_disorers) %&gt;% filter(Category==&quot;ASD genes&quot;) %&gt;% dplyr::select(`Gene name`) %&gt;% pull(),
  ForecASD_gene=forecASD_genes,
  SFARI_genes=positive_genes

)
library(VennDiagram)

display_venn &lt;- function(x, ...){
  library(VennDiagram)
  grid.newpage()
  venn_object &lt;- venn.diagram(x, filename = NULL, ...)
  grid.draw(venn_object)
}

#png(&quot;C:\\Users\\hans\\OneDrive - Marquette University\\SCZ-DAWN\\Figures\\Fig_overlapping_genes.png&quot;)

display_venn(
        x,
    #    category.names = c(&quot;MIRAGE FDR genes&quot; , &quot;Autism genes&quot; , &quot;DD/ID genes&quot;, &quot;SCZ GWAS genes&quot;),
    category.names = c(&quot;MOTIFs genes&quot; , &quot;ForecASD genes&quot; , &quot;SFARI genes&quot;),
        # Circles
        lwd = 2,
        lty = &#39;blank&#39;,
     #   fill = c(&quot;#999999&quot;, &quot;#E69F00&quot;, &quot;#56B4E9&quot;, &quot;#009E73&quot;),
    fill = c(&quot;red&quot;, &quot;blue&quot;, &quot;green&quot;),
        # Numbers
        cex = 1.5,
        fontface = &quot;italic&quot;,
        # Set names
        cat.cex = 1.5,
        cat.fontface = &quot;bold&quot;,
        cat.default.pos = &quot;outer&quot;,
       # cat.dist = c(0.055, 0.055, 0.1, 0.1, 0.3 )
    cat.pos = c(-20, 20, 180),  # Adjust angles as needed
         cat.dist = c(0.055, 0.055, -0.15) # Adjust distances from the circles
)</code></pre>
<p><img src="figure/part01_score_dist.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/han16/deep_motifs/blob/92c7f1d0d6281802939b4a9363bc7def2b77162b/docs/figure/part01_score_dist.Rmd/unnamed-chunk-10-1.png" target="_blank">92c7f1d</a>
</td>
<td>
han
</td>
<td>
2025-03-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.0 (2024-04-24 ucrt)
Platform: x86_64-w64-mingw32/x64
Running under: Windows 10 x64 (build 19045)

Matrix products: default


locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] VennDiagram_1.7.3   futile.logger_1.4.3 rprojroot_2.0.4    
 [4] DT_0.33             lubridate_1.9.3     forcats_1.0.0      
 [7] stringr_1.5.1       dplyr_1.1.4         purrr_1.0.2        
[10] readr_2.1.5         tidyr_1.3.1         tibble_3.2.1       
[13] ggplot2_3.5.1       tidyverse_2.0.0    

loaded via a namespace (and not attached):
 [1] sass_0.4.9           utf8_1.2.4           generics_0.1.3      
 [4] futile.options_1.0.1 stringi_1.8.4        hms_1.1.3           
 [7] digest_0.6.35        magrittr_2.0.3       timechange_0.3.0    
[10] evaluate_0.23        fastmap_1.1.1        workflowr_1.7.1     
[13] jsonlite_1.8.8       whisker_0.4.1        formatR_1.14        
[16] promises_1.3.0       fansi_1.0.6          crosstalk_1.2.1     
[19] scales_1.3.0         jquerylib_0.1.4      cli_3.6.2           
[22] rlang_1.1.3          munsell_0.5.1        withr_3.0.0         
[25] cachem_1.0.8         yaml_2.3.8           tools_4.4.0         
[28] tzdb_0.4.0           colorspace_2.1-0     httpuv_1.6.15       
[31] lambda.r_1.2.4       vctrs_0.6.5          R6_2.5.1            
[34] lifecycle_1.0.4      git2r_0.33.0         htmlwidgets_1.6.4   
[37] fs_1.6.4             pkgconfig_2.0.3      pillar_1.9.0        
[40] bslib_0.7.0          later_1.3.2          gtable_0.3.5        
[43] glue_1.7.0           Rcpp_1.0.12          highr_0.10          
[46] xfun_0.43            tidyselect_1.2.1     rstudioapi_0.16.0   
[49] knitr_1.46           farver_2.1.2         htmltools_0.5.8.1   
[52] labeling_0.4.3       rmarkdown_2.26       compiler_4.4.0      </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
