<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2025-03-19" />

<title>3/19 2025</title>

<script src="site_libs/header-attrs-2.26/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Autism_Research</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/han16/Autism_Research">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">3/19 2025</h1>
<h4 class="date">2025-03-19</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-04-03
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>deep_motifs/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230928code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230928)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230928code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230928)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomhan16deepmotifstree9081803e44ae8c2885e0c2a878aaa6a67f893444targetblank9081803a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/han16/deep_motifs/tree/9081803e44ae8c2885e0c2a878aaa6a67f893444" target="_blank">9081803</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomhan16deepmotifstree9081803e44ae8c2885e0c2a878aaa6a67f893444targetblank9081803a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/han16/deep_motifs/tree/9081803e44ae8c2885e0c2a878aaa6a67f893444" target="_blank">9081803</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Untracked files:
    Untracked:  VennDiagram.2025-04-03_10-07-40.627823.log
    Untracked:  VennDiagram.2025-04-03_10-11-47.561581.log

Unstaged changes:
    Modified:   analysis/part01_score_dist.Rmd
    Modified:   analysis/part04_clustering_analysis.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/part04_clustering_analysis.Rmd</code>) and HTML
(<code>docs/part04_clustering_analysis.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/han16/deep_motifs/blob/92c7f1d0d6281802939b4a9363bc7def2b77162b/analysis/part04_clustering_analysis.Rmd" target="_blank">92c7f1d</a>
</td>
<td>
han
</td>
<td>
2025-03-31
</td>
<td>
3/31/2025
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/han16/deep_motifs/92c7f1d0d6281802939b4a9363bc7def2b77162b/docs/part04_clustering_analysis.html" target="_blank">92c7f1d</a>
</td>
<td>
han
</td>
<td>
2025-03-31
</td>
<td>
3/31/2025
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>rm(list=ls())
set.seed(123)
library(tidyverse)
library(ggplot2)
library(DT)
library(VennDiagram)
library(rprojroot)
library(biomaRt)
library(igraph)
root &lt;- rprojroot::find_rstudio_root_file()</code></pre>
<pre class="r"><code>#result_round1=as_tibble(read.csv(file.path(root, &quot;../DiWangResults/ASD_prediction_results/ASD_RISK_PREDICTION_round1.csv&quot;))) # old results 

result_round1=as_tibble(read.csv(file.path(root, &quot;../DiWangResults/results updated_20250303/ASD_RISK_PREDICTION round 1.csv&quot;)))
result_round1=result_round1[result_round1[&#39;gene&#39;] != 0,]  # remove rows with gene=0</code></pre>
<pre class="r"><code>score_cutoff=0.5
top_genes=result_round1 %&gt;% filter(RISK_PROBABILITY_FUSION&gt;score_cutoff) %&gt;% dplyr::select(gene_gencodeV33) %&gt;% pull

#library(biomaRt)
ensembl &lt;- useMart(&quot;ensembl&quot;, dataset = &quot;hsapiens_gene_ensembl&quot;, host = &quot;https://grch37.ensembl.org&quot;)
# Define function to split queries
query_genes_in_batches &lt;- function(genes, batch_size = 50) {
  results &lt;- list()
  for (i in seq(1, length(genes), by = batch_size)) {
    batch &lt;- genes[i:min(i + batch_size - 1, length(genes))]
    cat(&quot;Querying batch:&quot;, i, &quot;to&quot;, min(i + batch_size - 1, length(genes)), &quot;\n&quot;)
    
    batch_result &lt;- tryCatch({
      getBM(attributes = c(&quot;hgnc_symbol&quot;, &quot;ensembl_gene_id&quot;, &quot;ensembl_peptide_id&quot;),
            filters = &quot;hgnc_symbol&quot;,
            values = batch,
            mart = ensembl)
    }, error = function(e) {
      cat(&quot;Error in batch query:&quot;, e$message, &quot;\n&quot;)
      return(NULL)
    })
    
    results &lt;- append(results, list(batch_result))
  }
  return(do.call(rbind, results))
}

# Run the batch query
top_genes_info &lt;- query_genes_in_batches(top_genes, batch_size = 1000)</code></pre>
<pre><code>Querying batch: 1 to 447 </code></pre>
<pre class="r"><code>#write.csv(jack_fu_gene_info, file=&quot;C:\\Users\\hans\\OneDrive - Marquette University\\AI_for_Autism\\jack_fu_gene_info.csv&quot;)


############# load string protein interaction data 
load(&quot;C:\\han\\Projects\\IntegratePPI\\tmp01_STRING_graph.Rdata&quot;)  # this is the protein interaction with score&gt;400


protein1_gene &lt;- top_genes_info$hgnc_symbol[match(string.keep$protein1, top_genes_info$ensembl_peptide_id)]
protein2_gene &lt;- top_genes_info$hgnc_symbol[match(string.keep$protein2, top_genes_info$ensembl_peptide_id)]

top_genes_interaction=data.frame(protein1_gene, protein2_gene) %&gt;% drop_na()  # string interactions for top genes 

#library(igraph)
# Create the graph from the edge list (gene interactions)
g &lt;- graph_from_data_frame(top_genes_interaction, directed = FALSE)

# Simplify the graph by removing multi-edges and loops (if any)
g &lt;- simplify(g, remove.multiple = TRUE, remove.loops = TRUE)

# Compute community structure
fg_community &lt;- fastgreedy.community(g)

# Print community structure
print(fg_community)</code></pre>
<pre><code>IGRAPH clustering fast greedy, groups: 12, mod: 0.43
+ groups:
  $`1`
   [1] &quot;SYT13&quot;    &quot;SLC17A7&quot;  &quot;LRRC4&quot;    &quot;GFAP&quot;     &quot;SYT1&quot;     &quot;CUX2&quot;    
   [7] &quot;NDST1&quot;    &quot;NOVA2&quot;    &quot;DGKD&quot;     &quot;PCDH10&quot;   &quot;NRXN2&quot;    &quot;DOCK3&quot;   
  [13] &quot;SLIT1&quot;    &quot;CACNA1C&quot;  &quot;GABRB2&quot;   &quot;LRRC4C&quot;   &quot;ANK3&quot;     &quot;FEZF2&quot;   
  [19] &quot;NPTN&quot;     &quot;GRIA2&quot;    &quot;ADCY1&quot;    &quot;SCN1A&quot;    &quot;TWF2&quot;     &quot;NLGN2&quot;   
  [25] &quot;ZNF608&quot;   &quot;GABRB3&quot;   &quot;ALDH5A1&quot;  &quot;KCNH7&quot;    &quot;GRIN2A&quot;   &quot;PCLO&quot;    
  [31] &quot;FOXG1&quot;    &quot;NEO1&quot;     &quot;PPP1CC&quot;   &quot;AUTS2&quot;    &quot;SHANK2&quot;   &quot;CASKIN1&quot; 
  [37] &quot;SCN8A&quot;    &quot;LINGO1&quot;   &quot;ZNF536&quot;   &quot;AKAP9&quot;    &quot;SPTBN1&quot;   &quot;ANK2&quot;    
  [43] &quot;KCNQ2&quot;    &quot;ELAVL3&quot;   &quot;ATP2B2&quot;   &quot;DNAJC5&quot;   &quot;CYB5R3&quot;   &quot;CACNA1E&quot; 
  [49] &quot;ATP1B1&quot;   &quot;GRIN1&quot;    &quot;KCNB1&quot;    &quot;POMGNT1&quot;  &quot;GRIK3&quot;    &quot;STXBP1&quot;  
  + ... omitted several groups/vertices</code></pre>
<pre class="r"><code># Get membership (which community each gene belongs to)
#membership(fg_community)

# Plot the network with communities
plot(fg_community, g, vertex.size = 5, vertex.label = NA,  main = &quot;Gene module communities&quot;)</code></pre>
<p><img src="figure/part04_clustering_analysis.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/han16/deep_motifs/blob/92c7f1d0d6281802939b4a9363bc7def2b77162b/docs/figure/part04_clustering_analysis.Rmd/unnamed-chunk-3-1.png" target="_blank">92c7f1d</a>
</td>
<td>
han
</td>
<td>
2025-03-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Extract the membership of each vertex
membership_vector &lt;- membership(fg_community)

# Print the membership of each vertex
print(membership_vector)</code></pre>
<pre><code>    PPP5C     SYT13       GRN     KDM4B     CDK13     ITGB4    KLHL20     EIF3D 
        2         1         2         4         4         6         2         2 
     SOS2     PSMA3       PNN   CSNK2A1      TSC2       MAZ   SLC17A7      EDF1 
        2         2         2         4         2         3         1         2 
      CPD     HDAC5      USP5    NUP155    TRIM23    PAPOLG    TARDBP     TSHZ3 
       10         4         2         5         2         2         2         4 
  KATNAL2    MBOAT7       VGF     LRRC4    ZC3H14      TAF4     EIF3G      GFAP 
        2         2         4         1         2         4         2         1 
      FUS     KDM6B     TERF2      ARF3 KIDINS220      PPIG      SYT1      MIB1 
        2         4         2         2         2         2         1         3 
   PSMD11     TAOK1      CUX2     NDST1      PHF3     KMT2C       YY1    CREBBP 
        2         2         1         1         4         4         4         4 
    SRCAP    SETD1A     RAB2A   RALGAPB      ATRN     EP300     NOVA2      BRD4 
        4         4         2         4         4         4         1         4 
   CLASP1     KIF1B    PIK3CA     EHBP1      DGKD    PCDH10      CUL3     RBBP5 
        6         2         2         7         1         1         2         4 
    HDAC4    DNMT3A     KIF3C     TNPO3     NRXN2     KAT6A    BRINP1     DOCK3 
        4         4         2         5         1         4         5         1 
    SLIT1   CACNA1C     ASXL3      CBX4    PRDM15      HUNK      DMWD     LMTK3 
        1         1         4         4         3         3        11         3 
     POGZ    GABRB2      PHIP    GIGYF1   FAM120A    ZNF462    LRRC4C    ARID5B 
        4         1         3         2         2         4         1         4 
   SEC24D      ANK3     DIP2C    MED13L     NIPBL     FEZF2     SCAF4      NPTN 
        2         1         5         4         4         1         3         1 
     UBR1     NACC1     WDFY3     GRIA2     NAA15     ABCE1    TCERG1     ADCY1 
        3         3         4         1         5         2         2         1 
    NFIL3      TSC1      LEO1     MAP1A   ANKRD11      DDB1  HNRNPUL2     CXXC5 
        8         2         4         2         4         2         2         4 
    SCN1A      TWF2  KIAA0232     NLGN2     YWHAG      BPTF    ZNF608    GABRB3 
        1         1         2         1         2         4         1         1 
   ZBTB21     RPL15      KLF7       SP3  ARHGAP32      MSL2     KMT2E     MLXIP 
        3         2         4         2         7         4         4         2 
   HNRNPD   ALDH5A1      DVL3     PACS1      TLK2  SNRNP200     FUBP3    MRFAP1 
        2         1         2        10         2         2         2         4 
    TRAF7    ARID1A     SPAST     ZMYM4      RAI1   SMARCE1     YLPM1     GSK3B 
        3         4         4         3         4         4         2         2 
    KDM3B     PHF12  C16orf72     KCNH7    GRIN2A   KLHDC10      PCLO     ZMIZ1 
        4         4         2         1         1         2         1         4 
   TNRC6C     TESK1      PUM2      RERE     FOXG1      TRIO   CAPRIN1    DYRK1A 
        4         2         2         4         1         2         2         2 
   TRIM24     MEF2C      NEO1    PPP1CC     DYRK2     BAZ1B   SMARCA4     AUTS2 
        2         4         1         1         2         4         4         1 
    NR4A2    SHANK2     CPSF7   CASKIN1      DMPK     SCN8A       WAC    LINGO1 
        8         1         2         1        11         1         5         1 
    DHCR7    ZNF536      MBD6    PSMD12     AKAP9   DYNC1H1       SON    SPTBN1 
       12         1         9         2         1         2         2         1 
     ANK2      SFPQ     BRPF3     NR3C2      PAX5       NF1     CNOT3    PLXNB1 
        1         2         4         2         4         2         2         2 
   VPS13B    ZBTB18  SMARCAD1     KCNQ2    ELAVL3      PHF2    PLXNB2     TCF20 
        4         4         4         1         1         4         2         4 
    PTPRF    ATP2B2     MYH10     SCAF1    DNAJC5     EIF3B    NCKAP1    CYB5R3 
        2         1         2         3         1         2         7         1 
     MTOR     MACF1      DLL1    PLXNA2     KDM5B   CACNA1E    HIVEP2    ATP1B1 
        2         6         3         2         4         1         4         1 
     DEDD    RNF146     RUSC1      EBF3   GATAD2B    ZNF292   SYNCRIP     GRIN1 
        2         2        12         4         4         4         2         1 
     ADNP     KCNB1      PTEN   POMGNT1   TSPAN14     KDM4A     GRIK3      AGO4 
        4         1         2         1         2         4         1         4 
     CHD6    STXBP1    DLGAP4     AHDC1     RBM12      SPEN      RORB     KDM2B 
        4         1         1         4         5         4         1         4 
    SPRY2    HECTD4      DPP6     AHNAK      EXT1       SKI     ARIH1       TEK 
        2         1         1         1         1         4         2         2 
   ELAVL2      CHD3     PTPRD     ITSN1   ZMYND11      RFX3     BRSK2   SMARCA2 
        2         4         1         2         4         1         2         4 
    DEAF1    SRGAP3     KCNQ3    TRIP12   CAMSAP1      TBR1      RTF1     HDLBP 
        2         1         1         4         2         1         4         2 
    ASH1L     BAZ2B      CBX1    PLXNA1    GAPVD1     G3BP1      CHD2     SIN3A 
        4         4         4         2         2         2         4         4 
    BCKDK    QRICH1    ATXN2L    CAMK2B      THRB    CSNK1E      NFIX     MED13 
        2         2         2         1         4         2         1         4 
 ARHGEF12      NAV3      TCF4    CTNND1     MYO5A    HECTD1    EIF4G3     DSCAM 
        2         1         4         2         2         3         2         1 
    CECR2     CADM2    INO80D     MTMR3    FBXO11       PSD     NRXN1     NCOA1 
        4         1         4         2         2         6         1         4 
    MARK2     SETD5      MBD5     FOXP2     SETD2    GIGYF2    KANSL1     TANC2 
        1         4         9         1         4         2         4         1 
   UBAP2L      PUM1      CHD7     DIP2A    EFTUD2      RELN     KIF5C     BIRC6 
        2         2         4         5         2         1         2         3 
    PRR12      NSD1     GRIK2      SOX5    PHF21A      KLC2      TNIK     SATB1 
        3         4         1         1         4         2         6         1 
    SATB2    TNRC6B   RUNX1T1      SMG1     AP2M1      ILF3   TBL1XR1      PBX1 
        1         4         4         2         1         2         4         4 
     HRAS     CELF4    CAMTA2    ANP32A   PPP2R5D     EHMT1     PSMD6      PTK7 
        2         1         1         1         2         4         2         2 
     BRAF  CACNA2D3    ZBTB20      SMG7     EIF4E    DPYSL2     RIMS1    SEC24B 
        2         1         4         2         2         1         1         2 
     UBR5    UNC13A     ASAP1     KMT2A    MYO18A    ZMYND8      APC2    MYCBP2 
        2         1         6         4         2         4         2         3 
   ATP1A3     CNTN1     AGAP2   SMARCC2      TLE3     MEIS2    SCAPER    ZNF423 
        1         1         1         4         4         4         7         4 
   ZNF865     VEZF1     TSHZ1     AP2S1    SETD1B     APBB1    GRIN2B 
        3         2         4         1         4         4         1 </code></pre>
<pre class="r"><code># Group the vertices by community
communities &lt;- split(V(g)$name, membership_vector)

# Print each community and its members
for (i in seq_along(communities)) {
  cat(&quot;Community&quot;, i, &quot;:&quot;, communities[[i]], &quot;\n&quot;)
}</code></pre>
<pre><code>Community 1 : SYT13 SLC17A7 LRRC4 GFAP SYT1 CUX2 NDST1 NOVA2 DGKD PCDH10 NRXN2 DOCK3 SLIT1 CACNA1C GABRB2 LRRC4C ANK3 FEZF2 NPTN GRIA2 ADCY1 SCN1A TWF2 NLGN2 ZNF608 GABRB3 ALDH5A1 KCNH7 GRIN2A PCLO FOXG1 NEO1 PPP1CC AUTS2 SHANK2 CASKIN1 SCN8A LINGO1 ZNF536 AKAP9 SPTBN1 ANK2 KCNQ2 ELAVL3 ATP2B2 DNAJC5 CYB5R3 CACNA1E ATP1B1 GRIN1 KCNB1 POMGNT1 GRIK3 STXBP1 DLGAP4 RORB HECTD4 DPP6 AHNAK EXT1 PTPRD RFX3 SRGAP3 KCNQ3 TBR1 CAMK2B NFIX NAV3 DSCAM CADM2 NRXN1 MARK2 FOXP2 TANC2 RELN GRIK2 SOX5 SATB1 SATB2 AP2M1 CELF4 CAMTA2 ANP32A CACNA2D3 DPYSL2 RIMS1 UNC13A ATP1A3 CNTN1 AGAP2 AP2S1 GRIN2B 
Community 2 : PPP5C GRN KLHL20 EIF3D SOS2 PSMA3 PNN TSC2 EDF1 USP5 TRIM23 PAPOLG TARDBP KATNAL2 MBOAT7 ZC3H14 EIF3G FUS TERF2 ARF3 KIDINS220 PPIG PSMD11 TAOK1 RAB2A KIF1B PIK3CA CUL3 KIF3C GIGYF1 FAM120A SEC24D ABCE1 TCERG1 TSC1 MAP1A DDB1 HNRNPUL2 KIAA0232 YWHAG RPL15 SP3 MLXIP HNRNPD DVL3 TLK2 SNRNP200 FUBP3 YLPM1 GSK3B C16orf72 KLHDC10 TESK1 PUM2 TRIO CAPRIN1 DYRK1A TRIM24 DYRK2 CPSF7 PSMD12 DYNC1H1 SON SFPQ NR3C2 NF1 CNOT3 PLXNB1 PLXNB2 PTPRF MYH10 EIF3B MTOR PLXNA2 DEDD RNF146 SYNCRIP PTEN TSPAN14 SPRY2 ARIH1 TEK ELAVL2 ITSN1 BRSK2 DEAF1 CAMSAP1 HDLBP PLXNA1 GAPVD1 G3BP1 BCKDK QRICH1 ATXN2L CSNK1E ARHGEF12 CTNND1 MYO5A EIF4G3 MTMR3 FBXO11 GIGYF2 UBAP2L PUM1 EFTUD2 KIF5C KLC2 SMG1 ILF3 HRAS PPP2R5D PSMD6 PTK7 BRAF SMG7 EIF4E SEC24B UBR5 MYO18A APC2 VEZF1 
Community 3 : MAZ MIB1 PRDM15 HUNK LMTK3 PHIP SCAF4 UBR1 NACC1 ZBTB21 TRAF7 ZMYM4 SCAF1 DLL1 HECTD1 BIRC6 PRR12 MYCBP2 ZNF865 
Community 4 : KDM4B CDK13 CSNK2A1 HDAC5 TSHZ3 VGF TAF4 KDM6B PHF3 KMT2C YY1 CREBBP SRCAP SETD1A RALGAPB ATRN EP300 BRD4 RBBP5 HDAC4 DNMT3A KAT6A ASXL3 CBX4 POGZ ZNF462 ARID5B MED13L NIPBL WDFY3 LEO1 ANKRD11 CXXC5 BPTF KLF7 MSL2 KMT2E MRFAP1 ARID1A SPAST RAI1 SMARCE1 KDM3B PHF12 ZMIZ1 TNRC6C RERE MEF2C BAZ1B SMARCA4 BRPF3 PAX5 VPS13B ZBTB18 SMARCAD1 PHF2 TCF20 KDM5B HIVEP2 EBF3 GATAD2B ZNF292 ADNP KDM4A AGO4 CHD6 AHDC1 SPEN KDM2B SKI CHD3 ZMYND11 SMARCA2 TRIP12 RTF1 ASH1L BAZ2B CBX1 CHD2 SIN3A THRB MED13 TCF4 CECR2 INO80D NCOA1 SETD5 SETD2 KANSL1 CHD7 NSD1 PHF21A TNRC6B RUNX1T1 TBL1XR1 PBX1 EHMT1 ZBTB20 KMT2A ZMYND8 SMARCC2 TLE3 MEIS2 ZNF423 TSHZ1 SETD1B APBB1 
Community 5 : NUP155 TNPO3 BRINP1 DIP2C NAA15 WAC RBM12 DIP2A 
Community 6 : ITGB4 CLASP1 MACF1 PSD TNIK ASAP1 
Community 7 : EHBP1 ARHGAP32 NCKAP1 SCAPER 
Community 8 : NFIL3 NR4A2 
Community 9 : MBD6 MBD5 
Community 10 : CPD PACS1 
Community 11 : DMWD DMPK 
Community 12 : DHCR7 RUSC1 </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.0 (2024-04-24 ucrt)
Platform: x86_64-w64-mingw32/x64
Running under: Windows 10 x64 (build 19045)

Matrix products: default


locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] igraph_2.0.3        biomaRt_2.60.1      rprojroot_2.0.4    
 [4] VennDiagram_1.7.3   futile.logger_1.4.3 DT_0.33            
 [7] lubridate_1.9.3     forcats_1.0.0       stringr_1.5.1      
[10] dplyr_1.1.4         purrr_1.0.2         readr_2.1.5        
[13] tidyr_1.3.1         tibble_3.2.1        ggplot2_3.5.1      
[16] tidyverse_2.0.0    

loaded via a namespace (and not attached):
 [1] tidyselect_1.2.1        blob_1.2.4              filelock_1.0.3         
 [4] Biostrings_2.72.0       fastmap_1.1.1           BiocFileCache_2.12.0   
 [7] promises_1.3.0          digest_0.6.35           timechange_0.3.0       
[10] lifecycle_1.0.4         KEGGREST_1.44.0         RSQLite_2.3.6          
[13] magrittr_2.0.3          compiler_4.4.0          rlang_1.1.3            
[16] sass_0.4.9              progress_1.2.3          tools_4.4.0            
[19] utf8_1.2.4              yaml_2.3.8              knitr_1.46             
[22] lambda.r_1.2.4          prettyunits_1.2.0       htmlwidgets_1.6.4      
[25] curl_5.2.1              bit_4.0.5               xml2_1.3.6             
[28] workflowr_1.7.1         withr_3.0.0             BiocGenerics_0.50.0    
[31] stats4_4.4.0            fansi_1.0.6             git2r_0.33.0           
[34] colorspace_2.1-0        scales_1.3.0            cli_3.6.2              
[37] rmarkdown_2.26          crayon_1.5.2            generics_0.1.3         
[40] rstudioapi_0.16.0       httr_1.4.7              tzdb_0.4.0             
[43] DBI_1.2.3               cachem_1.0.8            zlibbioc_1.50.0        
[46] AnnotationDbi_1.66.0    formatR_1.14            XVector_0.44.0         
[49] vctrs_0.6.5             jsonlite_1.8.8          IRanges_2.38.0         
[52] hms_1.1.3               S4Vectors_0.42.0        bit64_4.0.5            
[55] jquerylib_0.1.4         glue_1.7.0              stringi_1.8.4          
[58] gtable_0.3.5            later_1.3.2             GenomeInfoDb_1.40.0    
[61] UCSC.utils_1.0.0        munsell_0.5.1           pillar_1.9.0           
[64] rappdirs_0.3.3          htmltools_0.5.8.1       GenomeInfoDbData_1.2.12
[67] dbplyr_2.5.0            R6_2.5.1                httr2_1.0.1            
[70] evaluate_0.23           Biobase_2.64.0          highr_0.10             
[73] futile.options_1.0.1    png_0.1-8               memoise_2.0.1          
[76] httpuv_1.6.15           bslib_0.7.0             Rcpp_1.0.12            
[79] whisker_0.4.1           xfun_0.43               fs_1.6.4               
[82] pkgconfig_2.0.3        </code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
