<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2025-07-25" />

<title>07/25 2025</title>

<script src="site_libs/header-attrs-2.26/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="site_libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.33/datatables.js"></script>
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<script src="site_libs/jszip-1.13.6/jszip.min.js"></script>
<script src="site_libs/pdfmake-1.13.6/pdfmake.js"></script>
<script src="site_libs/pdfmake-1.13.6/vfs_fonts.js"></script>
<link href="site_libs/dt-ext-buttons-1.13.6/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="site_libs/dt-ext-buttons-1.13.6/js/dataTables.buttons.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.13.6/js/buttons.html5.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.13.6/js/buttons.colVis.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.13.6/js/buttons.print.min.js"></script>
<link href="site_libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Autism_Research</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/han16/Autism_Research">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">07/25 2025</h1>
<h4 class="date">2025-07-25</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-07-25
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>deep_motifs/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230928code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230928)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230928code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230928)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomhan16deepmotifstreeb72befa9aeb95c38d6f968e0676e04599eae8721targetblankb72befaa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/han16/deep_motifs/tree/b72befa9aeb95c38d6f968e0676e04599eae8721" target="_blank">b72befa</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomhan16deepmotifstreeb72befa9aeb95c38d6f968e0676e04599eae8721targetblankb72befaa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/han16/deep_motifs/tree/b72befa9aeb95c38d6f968e0676e04599eae8721" target="_blank">b72befa</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory

Untracked files:
    Untracked:  analysis/part03_func_enrichment_clusterprofiler.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<pre class="r"><code>rm(list=ls())
set.seed(123)
library(tidyverse)
library(ggplot2)
library(DT)
library(VennDiagram)
library(rprojroot)
root &lt;- rprojroot::find_rstudio_root_file()</code></pre>
<pre class="r"><code>#result_round1=as_tibble(read.csv(file.path(root, &quot;../DiWangResults/ASD_prediction_results/ASD_RISK_PREDICTION_round1.csv&quot;))) # old results 

result_round1=as_tibble(read.csv(file.path(root, &quot;../DiWangResults/results updated_20250303/ASD_RISK_PREDICTION round 1.csv&quot;)))
result_round1=result_round1[result_round1[&#39;gene&#39;] != 0,]  # remove rows with gene=0


SFARI_genes=as_tibble(read.csv(file.path(root, &quot;../../../Dataset/SFARI_base/SFARI-Gene_genes_08-19-2024release_09-16-2024export.csv&quot;)))
positive_genes=SFARI_genes %&gt;% filter (gene.score==&quot;1&quot;) %&gt;% dplyr::select(gene.symbol) %&gt;% pull()
positive_genes=intersect(positive_genes, result_round1$gene)


score_cutoff=0.5
top_genes=result_round1 %&gt;% filter(RISK_PROBABILITY_FUSION&gt;score_cutoff) %&gt;% dplyr::select(gene_gencodeV33) %&gt;% pull</code></pre>
<div id="clusterprofiler" class="section level2">
<h2>clusterprofiler</h2>
<pre class="r"><code>#if (!requireNamespace(&quot;BiocManager&quot;, quietly = TRUE))
#  install.packages(&quot;BiocManager&quot;)

#BiocManager::install(c(&quot;clusterProfiler&quot;, &quot;org.Hs.eg.db&quot;, &quot;enrichplot&quot;))


# Load the libraries
library(clusterProfiler)
library(org.Hs.eg.db)  # For human gene annotations
library(enrichplot)    # For visualization</code></pre>
<pre class="r"><code>ego &lt;- enrichGO(gene          = top_genes, 
               OrgDb         = org.Hs.eg.db, 
               keyType       = &quot;SYMBOL&quot;, 
               ont           = &quot;All&quot;,        # BP: Biological Process, CC: Cellular Component, MF: Molecular Function
               pAdjustMethod = &quot;BH&quot;, 
               pvalueCutoff  = 0.05, 
               qvalueCutoff  = 0.01)

# View the result
#head(ego)

# Dotplot for GO enrichment
dotplot(ego, showCategory=nrow(ego), title=&quot;&quot;)+
  theme(axis.text.y = element_text(size = 6))  # Adjust the size as needed</code></pre>
<p><img src="figure/part03_func_enrichment_clusterprofiler.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Extract significant terms while preserving the enrichResult object
#significant_ego &lt;- ego
#significant_ego@result &lt;- significant_ego@result[significant_ego@result$p.adjust &lt; 0.05, ]

# Create the dot plot with significant terms only
#dotplot(significant_ego, showCategory = nrow(significant_ego), title = &quot;GO Enrichment - Biological Process&quot;)</code></pre>
<pre class="r"><code>gsego &lt;- gseGO(gene          = sort(top_genes, decreasing = T),  
               OrgDb         = org.Hs.eg.db,
               keyType       = &quot;SYMBOL&quot;, 
               ont           = &quot;All&quot;,        # BP: Biological Process, CC: Cellular Component, MF: Molecular Function
               pAdjustMethod = &quot;BH&quot;, 
               pvalueCutoff  = 0.05, 
               qvalueCutoff  = 0.01)

# View the result
#head(ego)

# Dotplot for GO enrichment
dotplot(ego, showCategory=nrow(ego), title=&quot;&quot;)+
  theme(axis.text.y = element_text(size = 6))  # Adjust the size as needed

# Extract significant terms while preserving the enrichResult object
#significant_ego &lt;- ego
#significant_ego@result &lt;- significant_ego@result[significant_ego@result$p.adjust &lt; 0.05, ]

# Create the dot plot with significant terms only
#dotplot(significant_ego, showCategory = nrow(significant_ego), title = &quot;GO Enrichment - Biological Process&quot;)</code></pre>
<pre class="r"><code># Compute pairwise term similarity
ego_sim &lt;- pairwise_termsim(ego)

# Visualize the results using the enrichment map plot
emapplot(ego_sim, label=&quot;ID&quot;)+
 # theme(legend.position = &quot;none&quot;)
  theme(legend.text = element_text(size = 10))


emap &lt;- emapplot(ego_sim, showCategory = 30, label =F)  # or any number of categories
emap + theme(legend.text = element_text(size = 10))</code></pre>
<div id="gene-concept-network" class="section level3">
<h3>Gene-Concept Network</h3>
<pre class="r"><code># Convert gene symbols to Entrez IDs
gene_df &lt;- bitr(top_genes, fromType = &quot;SYMBOL&quot;, toType = &quot;ENTREZID&quot;, OrgDb = org.Hs.eg.db)

# Extract Entrez IDs
entrez_ids &lt;- gene_df$ENTREZID
#print(entrez_ids)

ekegg &lt;- enrichKEGG(gene         = entrez_ids, 
                   organism     = &quot;hsa&quot;,      # Human
                   pAdjustMethod = &quot;BH&quot;, 
                   pvalueCutoff  = 0.05)

# View the result
#head(ekegg)

# Barplot for KEGG enrichment
barplot(ekegg, showCategory= nrow(ekegg), title=&quot;KEGG Pathway Enrichment&quot;)+
  theme(axis.text.y = element_text(size = 5))</code></pre>
<p><img src="figure/part03_func_enrichment_clusterprofiler.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#Sort ekegg results by Count in descending order
ekegg_sorted &lt;- ekegg
ekegg_sorted@result &lt;- ekegg@result[order(ekegg@result$Count, decreasing = TRUE), ]

# Plot with sorted categories
barplot(ekegg_sorted,
        showCategory = nrow(ekegg_sorted),  # show all
        title = &quot;KEGG Pathway Enrichment&quot;) +
  theme(axis.text.y = element_text(size = 5), 
        plot.title = element_text(hjust = 0.5))  # Center title</code></pre>
<p><img src="figure/part03_func_enrichment_clusterprofiler.Rmd/unnamed-chunk-7-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Extract and filter
df &lt;- as.data.frame(ekegg@result)
df_filtered &lt;- df[df$p.adjust &lt; 0.05, ]

# Sort by Fold Enrichment
df_filtered &lt;- df_filtered[order(df_filtered$FoldEnrichment, decreasing = TRUE), ]

# Plot
kegg_pathway=ggplot(df_filtered, aes(x = reorder(Description, FoldEnrichment), 
                        y = FoldEnrichment,
                        fill = p.adjust)) +
  geom_col() +
  coord_flip() +
  scale_fill_gradient(low = &quot;red&quot;, high = &quot;blue&quot;, name = &quot;Adj. p-value&quot;) +
  labs(title = &quot;(A) Significant KEGG Pathways&quot;,
       x = &quot;&quot;,
       y = &quot;Fold Enrichment&quot;) +
  theme_minimal() +
  theme(
    plot.title = element_text(hjust = 0.5, size = 25),
    axis.text.y = element_text(size = 10), 
    axis.text.x = element_text(size = 14), 
    legend.title = element_text(size = 14),      # increase legend title
    legend.text = element_text(size = 12),        # increase legend label text, 
     axis.title.x = element_text(size = 18)       # y-axis title (this line was added)
  )
kegg_pathway</code></pre>
<p><img src="figure/part03_func_enrichment_clusterprofiler.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="sfari-genes" class="section level4">
<h4>SFARI genes</h4>
<pre class="r"><code># Convert gene symbols to Entrez IDs
gene_df &lt;- bitr(positive_genes, fromType = &quot;SYMBOL&quot;, toType = &quot;ENTREZID&quot;, OrgDb = org.Hs.eg.db)

# Extract Entrez IDs
entrez_ids &lt;- gene_df$ENTREZID
#print(entrez_ids)

sfari_ekegg &lt;- enrichKEGG(gene         = entrez_ids, 
                   organism     = &quot;hsa&quot;,      # Human
                   pAdjustMethod = &quot;BH&quot;, 
                   pvalueCutoff  = 0.05)


# Extract and filter
sfari_df &lt;- as.data.frame(sfari_ekegg@result)
sfari_df_filtered &lt;- sfari_df[sfari_df$p.adjust &lt; 0.05, ]

# Sort by Fold Enrichment
sfari_df_filtered &lt;- sfari_df_filtered[order(sfari_df_filtered$FoldEnrichment, decreasing = TRUE), ]

# Plot
sfari_kegg_pathway=ggplot(sfari_df_filtered, aes(x = reorder(Description, FoldEnrichment), 
                        y = FoldEnrichment,
                        fill = p.adjust)) +
  geom_col() +
  coord_flip() +
  scale_fill_gradient(low = &quot;red&quot;, high = &quot;blue&quot;, name = &quot;Adj. p-value&quot;) +
  labs(title = &quot;SFARI&quot;,
       x = &quot;&quot;,
       y = &quot;Fold Enrichment&quot;) +
  theme_minimal() +
  theme(
    plot.title = element_text(hjust = 0.5, size = 25),
    axis.text.y = element_text(size = 10), 
    axis.text.x = element_text(size = 14), 
    legend.title = element_text(size = 14),      # increase legend title
    legend.text = element_text(size = 12),        # increase legend label text, 
     axis.title.x = element_text(size = 18)       # y-axis title (this line was added)
  )
sfari_kegg_pathway</code></pre>
<p><img src="figure/part03_func_enrichment_clusterprofiler.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#kegg_pathway</code></pre>
</div>
<div id="compare-sfari-vs-motifs" class="section level4">
<h4>compare SFARI vs MOTIFs</h4>
<pre class="r"><code># Add a source column to both data frames
df_filtered$Source &lt;- &quot;Deep-MOTIFs&quot;
sfari_df_filtered$Source &lt;- &quot;SFARI&quot;

# Combine both data frames
combined_df &lt;- rbind(df_filtered, sfari_df_filtered)

# Make sure &#39;Description&#39; is treated as a factor and keeps order
combined_df$Description &lt;- factor(combined_df$Description, 
                                  levels = unique(combined_df$Description[order(combined_df$FoldEnrichment)]))


# Order by Count within Deep-MOTIFs
ordered_levels &lt;- combined_df %&gt;%
  filter(Source == &quot;Deep-MOTIFs&quot;) %&gt;%
  arrange(Count) %&gt;%
  pull(Description)

# Apply factor levels to Description in combined_df
combined_df$Description &lt;- factor(combined_df$Description, levels = ordered_levels)

combined_df$Count[is.na(combined_df$Count)] &lt;- 0  # replace NA with 0 
combined_df &lt;- combined_df[!is.na(combined_df$Description), ]  # remove NA Description 
# Plot
kegg_comparison_plot &lt;- ggplot(combined_df, aes(x = Description, 
                                                 y = Count,
                                                 fill = Source)) +
  geom_col(position = position_dodge(width = 0.7)) +
  coord_flip() +
  labs(title = &quot;(A) KEGG Pathway Comparison&quot;,
       x = &quot;&quot;,
       y = &quot;Gene Count&quot;) +
  theme_minimal() +
  theme(
    plot.title = element_text(hjust = 0.5, size = 25),
    axis.text.y = element_text(size = 10), 
    axis.text.x = element_text(size = 14), 
    legend.title = element_text(size = 14),      
    legend.text = element_text(size = 12),        
    axis.title.x = element_text(size = 18)
  ) +
 guides(fill = guide_legend(title = NULL))+
  scale_fill_manual(values = c(&quot;SFARI&quot; = &quot;steelblue&quot;, &quot;Deep-MOTIFs&quot; = &quot;tomato&quot;))

kegg_comparison_plot</code></pre>
<p><img src="figure/part03_func_enrichment_clusterprofiler.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code> ggplot(combined_df, aes(x = Description, 
                                                 y = -log(p.adjust),
                                                 fill = Source)) +
  geom_col(position = position_dodge(width = 0.7)) +
  coord_flip() +
  labs(title = &quot;(A) KEGG Pathway Comparison&quot;,
       x = &quot;&quot;,
       y = &quot;Gene Count&quot;) +
  theme_minimal() +
  theme(
    plot.title = element_text(hjust = 0.5, size = 25),
    axis.text.y = element_text(size = 10), 
    axis.text.x = element_text(size = 14), 
    legend.title = element_text(size = 14),      
    legend.text = element_text(size = 12),        
    axis.title.x = element_text(size = 18)
  ) +
  scale_fill_manual(values = c(&quot;SFARI&quot; = &quot;steelblue&quot;, &quot;Deep-MOTIFs&quot; = &quot;tomato&quot;))</code></pre>
<p><img src="figure/part03_func_enrichment_clusterprofiler.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>kegg_comparison_plot=ggplot(combined_df, aes(x = Description, y = Count, fill = p.adjust)) +
  geom_col() +
  facet_wrap(~Source) +
  scale_fill_gradient(low = &quot;red&quot;, high = &quot;blue&quot;, name = &quot;Adj. p-value&quot;) +
  coord_flip() +
  labs(title = &quot;(C) KEGG pathway comparisons&quot;,
       x = &quot;&quot;,
       y = &quot;Gene Count&quot;) +
  theme_minimal() +
  theme(
    plot.title = element_text(hjust = 0.5, size = 25),
    strip.text = element_text(size = 18, face = &quot;bold&quot;),  # facet subtitle size
    axis.text.y = element_text(size = 10), 
    axis.text.x = element_text(size = 14), 
    legend.title = element_text(size = 14),      
    legend.text = element_text(size = 12),        
    axis.title.x = element_text(size = 18),
    axis.title.y = element_blank()                # remove y-axis label
  )
kegg_comparison_plot</code></pre>
<p><img src="figure/part03_func_enrichment_clusterprofiler.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>combined_df %&gt;% filter(Source==&quot;Deep-MOTIFs&quot;)%&gt;%
datatable(extensions = &#39;Buttons&#39;,
            options = list(dom = &#39;Blfrtip&#39;,
                           buttons = c(&#39;copy&#39;, &#39;csv&#39;, &#39;excel&#39;, &#39;pdf&#39;, &#39;print&#39;),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,&quot;All&quot;))))</code></pre>
<div class="datatables html-widget html-fill-item" id="htmlwidget-ceb20dc2a931752ae4f3" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-ceb20dc2a931752ae4f3">{"x":{"filter":"none","vertical":false,"extensions":["Buttons"],"data":[["hsa05033","hsa04720","hsa00310","hsa05213","hsa04724","hsa03082","hsa04721","hsa04919","hsa05030","hsa05017","hsa05031","hsa05211","hsa04961","hsa01521","hsa04916","hsa03050","hsa03083","hsa04930","hsa05214","hsa04012","hsa05215","hsa05210","hsa04360","hsa05221","hsa05225","hsa04728","hsa04330","hsa04713","hsa04934","hsa04211","hsa05224","hsa03015","hsa04725","hsa04931","hsa04935","hsa04925","hsa04911","hsa04310","hsa04910","hsa04971","hsa04727","hsa04520","hsa04024","hsa04068","hsa04722","hsa04066","hsa04261","hsa01522","hsa05231","hsa04150","hsa05226","hsa05016","hsa05205","hsa04514","hsa04072","hsa04723","hsa04082","hsa05022","hsa05010","hsa04714","hsa05165","hsa05020","hsa05014"],["Human Diseases","Organismal Systems","Metabolism","Human Diseases","Organismal Systems","Genetic Information Processing","Organismal Systems","Organismal Systems","Human Diseases","Human Diseases","Human Diseases","Human Diseases","Organismal Systems","Human Diseases","Organismal Systems","Genetic Information Processing","Genetic Information Processing","Human Diseases","Human Diseases","Environmental Information Processing","Human Diseases","Human Diseases","Organismal Systems","Human Diseases","Human Diseases","Organismal Systems","Environmental Information Processing","Organismal Systems","Human Diseases","Organismal Systems","Human Diseases","Genetic Information Processing","Organismal Systems","Human Diseases","Organismal Systems","Organismal Systems","Organismal Systems","Environmental Information Processing","Organismal Systems","Organismal Systems","Organismal Systems","Cellular Processes","Environmental Information Processing","Environmental Information Processing","Organismal Systems","Environmental Information Processing","Organismal Systems","Human Diseases","Human Diseases","Environmental Information Processing","Human Diseases","Human Diseases","Human Diseases","Environmental Information Processing","Environmental Information Processing","Organismal Systems",null,"Human Diseases","Human Diseases","Organismal Systems","Human Diseases","Human Diseases","Human Diseases"],["Substance dependence","Nervous system","Amino acid metabolism","Cancer: specific types","Nervous system","Chromosome","Nervous system","Endocrine system","Substance dependence","Neurodegenerative disease","Substance dependence","Cancer: specific types","Excretory system","Drug resistance: antineoplastic","Endocrine system","Folding, sorting and degradation","Chromosome","Endocrine and metabolic disease","Cancer: specific types","Signal transduction","Cancer: specific types","Cancer: specific types","Development and regeneration","Cancer: specific types","Cancer: specific types","Nervous system","Signal transduction","Environmental adaptation","Endocrine and metabolic disease","Aging","Cancer: specific types","Translation","Nervous system","Endocrine and metabolic disease","Endocrine system","Endocrine system","Endocrine system","Signal transduction","Endocrine system","Digestive system","Nervous system","Cellular community - eukaryotes","Signal transduction","Signal transduction","Nervous system","Signal transduction","Circulatory system","Drug resistance: antineoplastic","Cancer: overview","Signal transduction","Cancer: specific types","Neurodegenerative disease","Cancer: overview","Signaling molecules and interaction","Signal transduction","Nervous system",null,"Neurodegenerative disease","Neurodegenerative disease","Environmental adaptation","Infectious disease: viral","Neurodegenerative disease","Neurodegenerative disease"],["hsa05033","hsa04720","hsa00310","hsa05213","hsa04724","hsa03082","hsa04721","hsa04919","hsa05030","hsa05017","hsa05031","hsa05211","hsa04961","hsa01521","hsa04916","hsa03050","hsa03083","hsa04930","hsa05214","hsa04012","hsa05215","hsa05210","hsa04360","hsa05221","hsa05225","hsa04728","hsa04330","hsa04713","hsa04934","hsa04211","hsa05224","hsa03015","hsa04725","hsa04931","hsa04935","hsa04925","hsa04911","hsa04310","hsa04910","hsa04971","hsa04727","hsa04520","hsa04024","hsa04068","hsa04722","hsa04066","hsa04261","hsa01522","hsa05231","hsa04150","hsa05226","hsa05016","hsa05205","hsa04514","hsa04072","hsa04723","hsa04082","hsa05022","hsa05010","hsa04714","hsa05165","hsa05020","hsa05014"],["Nicotine addiction","Long-term potentiation","Lysine degradation","Endometrial cancer","Glutamatergic synapse","ATP-dependent chromatin remodeling","Synaptic vesicle cycle","Thyroid hormone signaling pathway","Cocaine addiction","Spinocerebellar ataxia","Amphetamine addiction","Renal cell carcinoma","Endocrine and other factor-regulated calcium reabsorption","EGFR tyrosine kinase inhibitor resistance","Melanogenesis","Proteasome","Polycomb repressive complex","Type II diabetes mellitus","Glioma","ErbB signaling pathway","Prostate cancer","Colorectal cancer","Axon guidance","Acute myeloid leukemia","Hepatocellular carcinoma","Dopaminergic synapse","Notch signaling pathway","Circadian entrainment","Cushing syndrome","Longevity regulating pathway","Breast cancer","mRNA surveillance pathway","Cholinergic synapse","Insulin resistance","Growth hormone synthesis, secretion and action","Aldosterone synthesis and secretion","Insulin secretion","Wnt signaling pathway","Insulin signaling pathway","Gastric acid secretion","GABAergic synapse","Adherens junction","cAMP signaling pathway","FoxO signaling pathway","Neurotrophin signaling pathway","HIF-1 signaling pathway","Adrenergic signaling in cardiomyocytes","Endocrine resistance","Choline metabolism in cancer","mTOR signaling pathway","Gastric cancer","Huntington disease","Proteoglycans in cancer","Cell adhesion molecules","Phospholipase D signaling pathway","Retrograde endocannabinoid signaling","Neuroactive ligand signaling","Pathways of neurodegeneration - multiple diseases","Alzheimer disease","Thermogenesis","Human papillomavirus infection","Prion disease","Amyotrophic lateral sclerosis"],["8/238","13/238","11/238","9/238","15/238","15/238","10/238","15/238","6/238","17/238","8/238","8/238","6/238","9/238","11/238","5/238","9/238","5/238","8/238","9/238","11/238","9/238","19/238","7/238","17/238","13/238","6/238","9/238","14/238","8/238","13/238","9/238","10/238","9/238","10/238","8/238","7/238","14/238","11/238","6/238","7/238","7/238","17/238","10/238","9/238","8/238","11/238","7/238","7/238","11/238","10/238","20/238","13/238","10/238","9/238","9/238","12/238","28/238","22/238","13/238","18/238","14/238","17/238"],["41/9434","67/9434","63/9434","59/9434","116/9434","117/9434","79/9434","122/9434","49/9434","144/9434","69/9434","70/9434","53/9434","80/9434","101/9434","46/9434","83/9434","47/9434","76/9434","86/9434","106/9434","87/9434","184/9434","68/9434","170/9434","132/9434","62/9434","97/9434","155/9434","90/9434","148/9434","103/9434","116/9434","109/9434","122/9434","98/9434","86/9434","174/9434","138/9434","76/9434","89/9434","93/9434","226/9434","133/9434","120/9434","110/9434","154/9434","99/9434","99/9434","158/9434","150/9434","311/9434","204/9434","160/9434","149/9434","149/9434","199/9434","483/9434","391/9434","235/9434","333/9434","278/9434","371/9434"],[0.1951219512195122,0.1940298507462687,0.1746031746031746,0.1525423728813559,0.1293103448275862,0.1282051282051282,0.1265822784810127,0.1229508196721311,0.1224489795918367,0.1180555555555556,0.1159420289855072,0.1142857142857143,0.1132075471698113,0.1125,0.1089108910891089,0.108695652173913,0.108433734939759,0.1063829787234043,0.1052631578947368,0.1046511627906977,0.1037735849056604,0.103448275862069,0.1032608695652174,0.1029411764705882,0.1,0.09848484848484848,0.09677419354838709,0.09278350515463918,0.09032258064516129,0.08888888888888889,0.08783783783783784,0.08737864077669903,0.08620689655172414,0.08256880733944955,0.08196721311475409,0.08163265306122448,0.08139534883720931,0.08045977011494253,0.07971014492753623,0.07894736842105263,0.07865168539325842,0.07526881720430108,0.0752212389380531,0.07518796992481203,0.075,0.07272727272727272,0.07142857142857142,0.0707070707070707,0.0707070707070707,0.06962025316455696,0.06666666666666667,0.06430868167202572,0.06372549019607843,0.0625,0.06040268456375839,0.06040268456375839,0.06030150753768844,0.05797101449275362,0.05626598465473145,0.05531914893617021,0.05405405405405406,0.05035971223021583,0.04582210242587601],[7.734371797499487,7.691082403110499,6.921035080698946,6.046574562028201,5.12568820631701,5.081878905408317,5.0175513243272,4.873605179776829,4.853712913736923,4.679563492063492,4.595786140543174,4.530132052821129,4.487394957983193,4.459348739495798,4.317081287960729,4.308549506759225,4.298167459754986,4.216878240657965,4.172490048651039,4.148231385577487,4.11344537815126,4.100550565053608,4.093122031421264,4.08044982698962,3.963865546218488,3.903806977336389,3.835998915695311,3.677813393398596,3.580265654648956,3.5234360410831,3.48177379059732,3.463571836501591,3.417125470878007,3.272916506051962,3.249070119851219,3.235808609157949,3.22640218878249,3.189317106152806,3.159602971623432,3.129367536488279,3.117647058823529,2.983554712207464,2.981668773704172,2.980350034750742,2.972899159663865,2.882811306340718,2.831332533013205,2.802733214497921,2.802733214497921,2.759653228379959,2.642577030812325,2.549109676024751,2.525992750041193,2.477415966386554,2.394281202413852,2.394281202413852,2.390270681136776,2.297893070271587,2.230307979969481,2.192776685142142,2.142630024982966,1.996191282268303,1.816326530612245],[6.951856059006343,8.84193736478287,7.585581074044794,6.255340809328867,7.192468352394656,7.147084457294909,5.768554469039255,6.927678089233194,4.350850636107423,7.157866673596686,4.822568700077099,4.768954549734371,4.095702626098214,4.99867120036603,5.391630021102571,3.618627900017522,4.855082730674302,3.556592084424917,4.46713892196358,4.718135751986058,5.18577690561036,4.67386579358099,6.816304983511362,4.101147919877164,6.273322295282707,5.404810317140911,3.604123628386663,4.264581431247511,5.210670339775492,3.869555285017607,4.895450079173287,4.044209664500157,4.213866289371566,3.839602145711134,4.022308177350183,3.579160343607528,3.33662506933804,4.689123980794736,4.111301403683857,2.998332683646125,3.229035683114512,3.092451425975735,4.85083547205744,3.700129065162667,3.498978778743373,3.195331510595272,3.686103262075363,2.900720011536271,2.900720011536271,3.588303581524106,3.262253634157252,4.468969988615371,3.544709245010916,3.032097400679705,2.759729188177012,2.759729188177012,3.188756573952487,4.710756053796485,3.99723672283318,2.978767985964253,3.41505030889891,2.712227763549802,2.580648116930799],[6.786398007401921e-06,9.160031792850275e-09,4.049530010518083e-07,1.488962780770686e-05,1.961324284884202e-07,2.199833678339175e-07,2.705313439374984e-05,3.834178594886361e-07,0.00136510561140648,1.149292945434391e-07,0.0003233514861818734,0.0003572612905002869,0.002061114262951927,0.0001734900538373006,4.49591566709028e-05,0.00578179510931449,0.0002303029080336736,0.00633997509462918,0.0006267956444957247,0.0003017748959953849,7.039249693468173e-05,0.0003293284642094463,1.760483224475392e-07,0.00155862657755524,1.258265769693928e-06,2.719191368526488e-05,0.004574137085368276,0.0007378800430036281,3.548787274868528e-05,0.001905560811092427,9.014297164113033e-05,0.001137184145417923,0.000678740469537361,0.001695672608133858,0.001005462094515785,0.003257654755382549,0.005897948104143153,0.0001253693685463776,0.0007182785117374682,0.01216612105441402,0.007098918467757399,0.008971029466276375,5.585841854293925e-05,0.001935067645576821,0.003276258600151151,0.006559230328614051,0.001763016993272879,0.01242524186036942,0.01242524186036942,0.002161139818305123,0.004628494830141512,0.0001132920754586718,0.001959395303189103,0.007239989566081531,0.01306903367583603,0.01306903367583603,0.004524115904905004,3.143065205718895e-05,0.000351145544579536,0.006542709125116076,0.001898779904107869,0.01075524214621994,0.01269562880697354],[0.0001787084808615839,2.170927534905515e-06,1.371055160703979e-05,0.0003528841790426525,1.042721163532769e-05,1.042721163532769e-05,0.0005370402952839813,1.371055160703979e-05,0.009515589114803994,1.042721163532769e-05,0.003122033840705551,0.00313596021661363,0.01163057334094302,0.001957959179020964,0.0007103546754002643,0.02795627401363855,0.002480990418362757,0.02933089788455717,0.005305377419481669,0.003109593493517662,0.0009813542219717394,0.003122033840705551,1.042721163532769e-05,0.01055412853944548,3.727612342718262e-05,0.0005370402952839813,0.02285319322382372,0.005641211941672899,0.0006007589886741722,0.01132626065501994,0.001186882459941549,0.008167049771637808,0.00554694797518464,0.0111631780035479,0.007446703637507535,0.01725496196079606,0.02795627401363855,0.001485627017274574,0.005641211941672899,0.04907970534845922,0.03115636438626859,0.03796667827691966,0.0008274028246672877,0.01132626065501994,0.01725496196079606,0.02933089788455717,0.01129283857853168,0.04907970534845922,0.04907970534845922,0.01191139853344917,0.02285319322382372,0.00141316957282659,0.01132626065501994,0.03119777322111496,0.04916446001862124,0.04916446001862124,0.02285319322382372,0.0005730049644272139,0.00313596021661363,0.02933089788455717,0.01132626065501994,0.04471916471323029,0.04916446001862124],[0.0001206470756871453,1.465605086856044e-06,9.256068595469903e-06,0.0002382340449233097,7.03946777068536e-06,7.03946777068536e-06,0.0003625588491368651,9.256068595469903e-06,0.006424026406618731,7.03946777068536e-06,0.002107702170940457,0.002117103943705404,0.007851863858864484,0.001321828981617529,0.0004795643378229633,0.01887343393325809,0.001674930240244899,0.01980145004864619,0.003581689397118427,0.002099303624315721,0.0006625176182087693,0.002107702170940457,7.03946777068536e-06,0.007125150068823956,2.516531539387856e-05,0.0003625588491368651,0.01542831610047171,0.003808413125180017,0.0004055756885564032,0.007646420695372109,0.0008012708590322698,0.005513620098995991,0.003744775004344061,0.007536322702817148,0.005027310472578927,0.01164891946720409,0.01887343393325809,0.001002954948371021,0.003808413125180017,0.03313397829431847,0.02103383249705896,0.02563151276078965,0.0005585841854293925,0.007646420695372109,0.01164891946720409,0.01980145004864619,0.007623857268207044,0.03313397829431847,0.03313397829431847,0.008041450486716735,0.01542831610047171,0.0009540385301782889,0.007646420695372109,0.02106178782860082,0.03319119663704388,0.03319119663704388,0.01542831610047171,0.0003868387945500179,0.002117103943705404,0.01980145004864619,0.007646420695372109,0.03019015339289809,0.03319119663704388],["2904/57030/773/2903/2891/2902/2562/2561","2904/1387/2033/775/2903/673/2891/815/107/2902/5501/816/3265","55904/55870/4297/79813/51111/64324/7468/58508/9739/23067/29072","5290/1499/673/10297/6655/6934/2932/5728/3265","2904/775/57030/85358/773/2903/2899/9762/1742/2891/2898/107/2902/2770/22941","6597/8289/10847/57459/2186/57492/9031/6601/1107/27443/6605/7528/6595/54891/29994","6812/1213/57030/773/6529/23025/1173/22999/6857/1175","9969/478/23389/1387/25942/7249/2033/5290/1499/481/2475/7068/8648/2932/3265","2904/2903/1742/2891/2902/2770","2904/5290/9698/773/11273/5717/5718/11047/5649/23152/2475/2903/9861/2891/5684/23369/2902","2904/775/2903/2891/815/2902/5501/816","1387/2033/5290/5781/57144/673/6655/3265","478/1213/491/481/1173/1175","5290/2475/673/6655/4763/2932/1977/5728/3265","1387/2033/1499/815/107/6934/2932/816/2770/1857/3265","5717/5718/11047/9861/5684","1387/2033/55777/84678/7528/80816/114785/26053/8535","775/5290/777/773/2475","5290/2475/673/815/6655/816/5728/3265","5290/57144/2475/673/815/6655/2932/816/3265","1387/2033/5290/1499/2475/673/6655/6934/2932/5728/3265","5290/1499/2475/673/10297/6655/6934/2932/3265","5290/5781/64101/57144/23365/6585/9901/815/5361/1808/23654/57689/2932/4756/816/2770/5364/5362/3265","5290/2475/673/6655/6934/862/3265","6597/8289/57492/5290/6601/1499/2475/6605/6595/673/10297/6655/6934/2932/1857/5728/3265","2904/6323/775/3800/773/5528/2903/2891/815/2932/5501/816/2770","1387/23013/7090/28514/2033/1857","2904/775/2903/2891/815/107/2902/816/2770","4297/775/1499/5929/673/10297/815/5087/107/6934/2932/816/2770/1857","79813/7249/5290/7248/2475/107/1977/3265","28514/5290/1499/2475/673/10297/8648/6655/6934/2932/1857/5728/3265","23049/5528/79869/9887/2521/64895/23435/5501/5411","775/5290/773/3785/815/107/3786/816/2770/3265","5290/5781/2475/5792/22877/10724/2932/5501/5728","1387/2033/775/5290/2475/6655/107/2932/2770/3265","478/775/4929/491/481/815/107/816","478/775/27445/481/815/107/816","1387/7090/57680/2033/79718/1499/1454/10297/815/6934/1457/2932/816/1857","7249/5290/7248/2475/5792/673/6655/2932/5501/1977/3265","478/481/815/107/816/2770","775/773/6529/107/2562/2770/2561","1387/2033/1499/1500/5792/6934/1457","2904/478/1387/2033/775/5290/491/481/2903/673/2891/815/107/2902/5501/816/2770","1387/2033/5290/1454/673/6655/116986/2290/5728/3265","5290/5781/673/815/6655/2932/57498/816/3265","1387/2033/5290/7010/2475/815/1977/816","478/775/491/481/5528/815/107/5501/55799/816/2770","28514/5290/2475/673/6655/107/3265","7249/5290/7248/2475/6655/8527/3265","7249/5290/7248/2475/673/6655/2932/1977/1857/5728/3265","5290/1499/2475/673/10297/6655/6934/2932/1857/3265","2904/1387/25942/6874/2033/1213/3800/5717/5718/11047/2475/1173/9861/1742/2891/5684/2902/64837/25981/1175","288/287/5290/1499/5781/23365/2475/673/815/6655/5501/816/3265","9369/5789/57555/64101/5792/9378/9379/57689/4756/1272","7249/5290/7248/5781/2475/6655/107/8527/3265","775/57030/773/2891/107/22999/2562/2770/2561","2904/57030/6529/2903/2899/2891/2898/107/2902/2562/2770/2561","2904/775/3800/1499/1454/11273/5717/5718/11047/2475/2903/673/10297/9861/2521/1742/2891/815/23435/5684/1457/2902/64837/2932/816/1857/25981/3265","2904/775/5290/3800/1499/1454/5717/5718/11047/2475/2903/673/10297/9861/5684/1457/2902/64837/322/2932/1857/3265","6597/51780/8289/7249/57492/6601/7248/2475/6605/6595/6655/107/3265","1387/7249/2033/5290/7248/1499/5528/5649/2475/7337/10297/6655/6934/2932/1857/5728/3691/3265","2904/775/5290/3800/5717/5718/11047/2903/9861/5684/1457/2902/64837/2932","2904/3800/11273/5717/5718/11047/2475/2903/9861/2521/2891/23435/5684/2902/64837/9631/25981"],[8,13,11,9,15,15,10,15,6,17,8,8,6,9,11,5,9,5,8,9,11,9,19,7,17,13,6,9,14,8,13,9,10,9,10,8,7,14,11,6,7,7,17,10,9,8,11,7,7,11,10,20,13,10,9,9,12,28,22,13,18,14,17],["Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs","Deep-MOTIFs"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>category<\/th>\n      <th>subcategory<\/th>\n      <th>ID<\/th>\n      <th>Description<\/th>\n      <th>GeneRatio<\/th>\n      <th>BgRatio<\/th>\n      <th>RichFactor<\/th>\n      <th>FoldEnrichment<\/th>\n      <th>zScore<\/th>\n      <th>pvalue<\/th>\n      <th>p.adjust<\/th>\n      <th>qvalue<\/th>\n      <th>geneID<\/th>\n      <th>Count<\/th>\n      <th>Source<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["copy","csv","excel","pdf","print"],"lengthMenu":[[10,25,50,-1],["10","25","50","All"]],"columnDefs":[{"className":"dt-right","targets":[7,8,9,10,11,12,14]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"category","targets":1},{"name":"subcategory","targets":2},{"name":"ID","targets":3},{"name":"Description","targets":4},{"name":"GeneRatio","targets":5},{"name":"BgRatio","targets":6},{"name":"RichFactor","targets":7},{"name":"FoldEnrichment","targets":8},{"name":"zScore","targets":9},{"name":"pvalue","targets":10},{"name":"p.adjust","targets":11},{"name":"qvalue","targets":12},{"name":"geneID","targets":13},{"name":"Count","targets":14},{"name":"Source","targets":15}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>length(unique(combined_df %&gt;% filter(Source==&quot;Deep-MOTIFs&quot;)%&gt;% dplyr::select(Description) %&gt;% pull()))</code></pre>
<pre><code>[1] 63</code></pre>
<pre class="r"><code># Convert gene symbols to Entrez IDs
gene_df &lt;- bitr(top_genes, fromType = &quot;SYMBOL&quot;, toType = &quot;ENTREZID&quot;, OrgDb = org.Hs.eg.db)

# Extract Entrez IDs
entrez_ids &lt;- gene_df$ENTREZID
#print(entrez_ids)

ekegg &lt;- enrichKEGG(gene         = entrez_ids, 
                   organism     = &quot;hsa&quot;,      # Human
                   pAdjustMethod = &quot;BH&quot;, 
                   pvalueCutoff  = 0.05)

# Network plot for KEGG pathways
cnetplot(ekegg, circular = TRUE, colorEdge = TRUE, cex_label_gene=0.5, cex_label_category=5)</code></pre>
<p><img src="figure/part03_func_enrichment_clusterprofiler.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>length(unique(ekegg@gene))</code></pre>
<pre><code>[1] 446</code></pre>
<pre class="r"><code># Convert Entrez IDs to Gene Symbols
gene_symbols &lt;- bitr(entrez_ids, fromType = &quot;ENTREZID&quot;, toType = &quot;SYMBOL&quot;, OrgDb = org.Hs.eg.db)

# Merge the converted gene symbols back into the enrichment result
ekegg@result$geneID &lt;- sapply(strsplit(ekegg@result$geneID, &quot;/&quot;), function(ids) {
  symbols &lt;- gene_symbols$SYMBOL[match(ids, gene_symbols$ENTREZID)]
  paste(symbols, collapse = &quot;/&quot;)
})
length(unique(ekegg@gene))</code></pre>
<pre><code>[1] 446</code></pre>
<pre class="r"><code># Plot the cnetplot with gene symbols
cnetplot(ekegg, circular = TRUE, colorEdge = TRUE, cex_label_gene=0.4, cex_label_category=1)</code></pre>
<p><img src="figure/part03_func_enrichment_clusterprofiler.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<ul>
<li><a
href="https://www.sciencedirect.com/science/article/pii/S2666675821000667">reference-clusterProfiler
4.0</a></li>
</ul>
<pre class="r"><code># Your gene list (symbols or IDs from ekegg)
all_genes &lt;- unique(unlist(strsplit(ekegg@result$geneID, split = &quot;/&quot;)))

# Define new genes (example)
new_genes &lt;- setdiff(all_genes, positive_genes)  # replace with your actual new genes

# Example: your gene groups vector (names are gene symbols or IDs)
gene_group &lt;- ifelse(all_genes %in% new_genes, &quot;new&quot;, &quot;old&quot;)
names(gene_group) &lt;- all_genes

# Create a vector of colors by gene group
group_colors &lt;- c(new = &quot;red&quot;, old = &quot;grey50&quot;, category = &quot;orange&quot;)


# Replace long description with line break version
ekegg@result$Description &lt;- gsub(
  &quot;ATP-dependent chromatin remodeling&quot;,
  &quot;ATP-dependent\nchromatin remodeling&quot;,
  ekegg@result$Description
)


# Generate cnetplot and assign it to a variable
p &lt;- cnetplot(ekegg,
              circular = TRUE,
              colorEdge = TRUE,
              cex_label_gene = 0.6,
              cex_label_category = 1)

# Assign &#39;node_group&#39; based on &#39;name&#39;
p$data &lt;- p$data %&gt;%
  mutate(node_group = case_when(
    name %in% new_genes ~ &quot;new&quot;,
    name %in% positive_genes ~ &quot;old&quot;,
    TRUE ~ &quot;category&quot;
  )) %&gt;%
  # Map group to color column used by plot
  mutate(color = group_colors[node_group])

# Now force ggplot to use your colors for the nodes
net_plot=p + 
  # Override color scale to identity so colors in &#39;color&#39; column are used directly
  scale_color_identity()+
  ggtitle(&quot;(D) Genes in top five KEGG pathways&quot;)+  # Replace with your desired title
  theme(plot.title = element_text(hjust = 0.5, size=25), 
        legend.position = &quot;right&quot;, 
        legend.title = element_blank(),                      # Remove legend title
    legend.text = element_text(size = 14)                # Bigger legend labels
    )
net_plot</code></pre>
<p><img src="figure/part03_func_enrichment_clusterprofiler.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Extract top 5 pathways
ekegg_top5 &lt;- ekegg
ekegg_top5@result &lt;- ekegg@result[1:5, ]

# Convert to tibble to use dplyr functions safely
ekegg_df &lt;- as_tibble(ekegg_top5@result)

# Extract gene–pathway relationships
gene_pathway_df &lt;- ekegg_df %&gt;%
  dplyr::select(Description, geneID) %&gt;%
  tidyr::separate_rows(geneID, sep = &quot;/&quot;) %&gt;%
  dplyr::rename(pathway = Description, gene = geneID) %&gt;%
  dplyr::distinct()


# Filter genes shown in the cnetplot
Genes_in_pathways &lt;- unique(p$data$name[p$data$node_group %in% c(&quot;new&quot;, &quot;old&quot;)])
gene_pathway_df &lt;- gene_pathway_df %&gt;%
  filter(gene %in% Genes_in_pathways)

# Add annotations
in_positive &lt;- ifelse(gene_pathway_df$gene %in% positive_genes, &quot;Yes&quot;, &quot;NO&quot;)
in_deep_motifs &lt;- ifelse(gene_pathway_df$gene %in% top_genes, &quot;Yes&quot;, &quot;NO&quot;)

names(in_positive) &lt;- gene_pathway_df$gene
names(in_deep_motifs) &lt;- gene_pathway_df$gene

annotation_df &lt;- gene_pathway_df %&gt;%
  mutate(
    IS_SFARI_gene = in_positive[gene],
    IS_MOTIFs_gene = in_deep_motifs[gene]
  )

annotation_df%&gt;%
datatable(extensions = &#39;Buttons&#39;,
            options = list(dom = &#39;Blfrtip&#39;,
                           buttons = c(&#39;copy&#39;, &#39;csv&#39;, &#39;excel&#39;, &#39;pdf&#39;, &#39;print&#39;),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,&quot;All&quot;))))</code></pre>
<div class="datatables html-widget html-fill-item" id="htmlwidget-0d7598595affc4f25ab8" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-0d7598595affc4f25ab8">{"x":{"filter":"none","vertical":false,"extensions":["Buttons"],"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79"],["Long-term potentiation","Long-term potentiation","Long-term potentiation","Long-term potentiation","Long-term potentiation","Long-term potentiation","Long-term potentiation","Long-term potentiation","Long-term potentiation","Long-term potentiation","Long-term potentiation","Long-term potentiation","Long-term potentiation","Spinocerebellar ataxia","Spinocerebellar ataxia","Spinocerebellar ataxia","Spinocerebellar ataxia","Spinocerebellar ataxia","Spinocerebellar ataxia","Spinocerebellar ataxia","Spinocerebellar ataxia","Spinocerebellar ataxia","Spinocerebellar ataxia","Spinocerebellar ataxia","Spinocerebellar ataxia","Spinocerebellar ataxia","Spinocerebellar ataxia","Spinocerebellar ataxia","Spinocerebellar ataxia","Spinocerebellar ataxia","Axon guidance","Axon guidance","Axon guidance","Axon guidance","Axon guidance","Axon guidance","Axon guidance","Axon guidance","Axon guidance","Axon guidance","Axon guidance","Axon guidance","Axon guidance","Axon guidance","Axon guidance","Axon guidance","Axon guidance","Axon guidance","Axon guidance","Glutamatergic synapse","Glutamatergic synapse","Glutamatergic synapse","Glutamatergic synapse","Glutamatergic synapse","Glutamatergic synapse","Glutamatergic synapse","Glutamatergic synapse","Glutamatergic synapse","Glutamatergic synapse","Glutamatergic synapse","Glutamatergic synapse","Glutamatergic synapse","Glutamatergic synapse","Glutamatergic synapse","ATP-dependent\nchromatin remodeling","ATP-dependent\nchromatin remodeling","ATP-dependent\nchromatin remodeling","ATP-dependent\nchromatin remodeling","ATP-dependent\nchromatin remodeling","ATP-dependent\nchromatin remodeling","ATP-dependent\nchromatin remodeling","ATP-dependent\nchromatin remodeling","ATP-dependent\nchromatin remodeling","ATP-dependent\nchromatin remodeling","ATP-dependent\nchromatin remodeling","ATP-dependent\nchromatin remodeling","ATP-dependent\nchromatin remodeling","ATP-dependent\nchromatin remodeling","ATP-dependent\nchromatin remodeling"],["GRIN2B","CREBBP","EP300","CACNA1C","GRIN2A","BRAF","GRIA2","CAMK2A","ADCY1","GRIN1","PPP1CC","CAMK2B","HRAS","GRIN2B","PIK3CA","PUM1","CACNA1A","ATXN2L","PSMD11","PSMD12","ADRM1","RELN","CIC","MTOR","GRIN2A","PSMD6","GRIA2","PSMA3","PUM2","GRIN1","PIK3CA","PTPN11","LRRC4","PAK5","ARHGEF12","SLIT1","SRGAP3","CAMK2A","PLXNA1","DPYSL2","PLXNB2","LRRC4C","GSK3B","NEO1","CAMK2B","GNAI1","PLXNB1","PLXNA2","HRAS","GRIN2B","CACNA1C","SLC17A7","SHANK3","CACNA1A","GRIN2A","GRIK3","LZTS3","DLG4","GRIA2","GRIK2","ADCY1","GRIN1","GNAI1","SHANK2","SMARCA4","ARID1A","SRCAP","GATAD2B","BPTF","ARID1B","BAZ1B","SMARCC2","CHD3","CECR2","SMARCE1","YY1","SMARCA2","INO80D","BAZ2B"],["Yes","Yes","Yes","Yes","Yes","Yes","Yes","NO","NO","Yes","NO","NO","Yes","Yes","NO","NO","Yes","NO","Yes","Yes","NO","Yes","Yes","Yes","Yes","Yes","Yes","NO","NO","Yes","NO","Yes","NO","NO","NO","NO","NO","NO","NO","Yes","NO","Yes","NO","NO","NO","Yes","NO","NO","Yes","Yes","Yes","NO","Yes","Yes","Yes","NO","NO","Yes","Yes","NO","NO","Yes","Yes","Yes","Yes","NO","Yes","NO","NO","Yes","NO","Yes","Yes","NO","NO","Yes","Yes","NO","Yes"],["Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>pathway<\/th>\n      <th>gene<\/th>\n      <th>IS_SFARI_gene<\/th>\n      <th>IS_MOTIFs_gene<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["copy","csv","excel","pdf","print"],"lengthMenu":[[10,25,50,-1],["10","25","50","All"]],"columnDefs":[{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"pathway","targets":1},{"name":"gene","targets":2},{"name":"IS_SFARI_gene","targets":3},{"name":"IS_MOTIFs_gene","targets":4}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code># Extract node data from the cnetplot object
node_data &lt;- p$data

# Get number of unique genes (exclude categories)
num_genes &lt;- length(unique(node_data$name[node_data$node_group %in% c(&quot;new&quot;, &quot;old&quot;)]))
num_genes</code></pre>
<pre><code>[1] 63</code></pre>
<pre class="r"><code># Get number of unique pathways (categories)
num_pathways &lt;- length(unique(node_data$name[node_data$node_group == &quot;category&quot;]))
num_pathways</code></pre>
<pre><code>[1] 5</code></pre>
<pre class="r"><code>Genes_in_pathways=node_data$name[node_data$node_group %in% c(&quot;new&quot;, &quot;old&quot;)]

in_positive &lt;- ifelse(Genes_in_pathways %in% positive_genes, &quot;Yes&quot;, &quot;NO&quot;)
in_deep_motifs=ifelse(Genes_in_pathways %in% top_genes, &quot;Yes&quot;, &quot;NO&quot;)

data.frame(Genes_in_pathways, IS_SFARI_gene=in_positive, IS_MOTIFs_gene=in_deep_motifs)%&gt;%
datatable(extensions = &#39;Buttons&#39;,
            options = list(dom = &#39;Blfrtip&#39;,
                           buttons = c(&#39;copy&#39;, &#39;csv&#39;, &#39;excel&#39;, &#39;pdf&#39;, &#39;print&#39;),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,&quot;All&quot;))))</code></pre>
<div class="datatables html-widget html-fill-item" id="htmlwidget-5bf888dc62471906a988" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-5bf888dc62471906a988">{"x":{"filter":"none","vertical":false,"extensions":["Buttons"],"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63"],["GRIN2B","CREBBP","EP300","CACNA1C","GRIN2A","BRAF","GRIA2","CAMK2A","ADCY1","GRIN1","PPP1CC","CAMK2B","HRAS","PIK3CA","PUM1","CACNA1A","ATXN2L","PSMD11","PSMD12","ADRM1","RELN","CIC","MTOR","PSMD6","PSMA3","PUM2","PTPN11","LRRC4","PAK5","ARHGEF12","SLIT1","SRGAP3","PLXNA1","DPYSL2","PLXNB2","LRRC4C","GSK3B","NEO1","GNAI1","PLXNB1","PLXNA2","SLC17A7","SHANK3","GRIK3","LZTS3","DLG4","GRIK2","SHANK2","SMARCA4","ARID1A","SRCAP","GATAD2B","BPTF","ARID1B","BAZ1B","SMARCC2","CHD3","CECR2","SMARCE1","YY1","SMARCA2","INO80D","BAZ2B"],["Yes","Yes","Yes","Yes","Yes","Yes","Yes","NO","NO","Yes","NO","NO","Yes","NO","NO","Yes","NO","Yes","Yes","NO","Yes","Yes","Yes","Yes","NO","NO","Yes","NO","NO","NO","NO","NO","NO","Yes","NO","Yes","NO","NO","Yes","NO","NO","NO","Yes","NO","NO","Yes","NO","Yes","Yes","NO","Yes","NO","NO","Yes","NO","Yes","Yes","NO","NO","Yes","Yes","NO","Yes"],["Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Genes_in_pathways<\/th>\n      <th>IS_SFARI_gene<\/th>\n      <th>IS_MOTIFs_gene<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["copy","csv","excel","pdf","print"],"lengthMenu":[[10,25,50,-1],["10","25","50","All"]],"columnDefs":[{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Genes_in_pathways","targets":1},{"name":"IS_SFARI_gene","targets":2},{"name":"IS_MOTIFs_gene","targets":3}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>library(gridExtra)
all &lt;- grid.arrange(kegg_comparison_plot,  net_plot, nrow=1) 
ggsave(all,filename = paste0(&quot;C:\\Users\\hans\\OneDrive - Marquette University\\AI_for_Autism\\Fig\\KEGG-pathway-Deep-MOTIFs.pdf&quot;),width = 18,height = 7,dpi = 800)</code></pre>
<pre class="r"><code>all &lt;- grid.arrange(ranking_change,  gene_set_enrichment,kegg_comparison_plot,  net_plot,  nrow=2) 
ggsave(all,filename = paste0(&quot;C:\\Users\\hans\\OneDrive - Marquette University\\AI_for_Autism\\Fig\\real_data_validation.pdf&quot;),width = 20,height = 16,dpi = 900)</code></pre>
<pre class="r"><code># show how many genes are displayed in the graph
top_n &lt;- 5  # change this number to your desired showCategory
top_terms &lt;- ekegg@result %&gt;%
  arrange(p.adjust) %&gt;%
  slice_head(n = top_n)

genes_shown &lt;- unique(unlist(strsplit(top_terms$geneID, &quot;/&quot;)))
length(genes_shown)  # Number of genes that will appear</code></pre>
<pre><code>[1] 63</code></pre>
<pre class="r"><code>novel_genes=intersect(genes_shown, new_genes)

length(novel_genes)</code></pre>
<pre><code>[1] 32</code></pre>
<pre class="r"><code>setdiff(novel_genes, SFARI_genes$gene.symbol)</code></pre>
<pre><code> [1] &quot;ADCY1&quot;    &quot;PPP1CC&quot;   &quot;PUM1&quot;     &quot;ATXN2L&quot;   &quot;ADRM1&quot;    &quot;PSMA3&quot;   
 [7] &quot;PUM2&quot;     &quot;PAK5&quot;     &quot;ARHGEF12&quot; &quot;SLIT1&quot;    &quot;PLXNA1&quot;   &quot;PLXNB2&quot;  
[13] &quot;GSK3B&quot;    &quot;PLXNA2&quot;   &quot;SLC17A7&quot;  &quot;LZTS3&quot;    &quot;ARID1A&quot;   &quot;GATAD2B&quot; 
[19] &quot;BPTF&quot;     &quot;BAZ1B&quot;    &quot;SMARCE1&quot;  &quot;INO80D&quot;  </code></pre>
<ul>
<li><p>63 genes were displayed in top 5 enriched pathways</p></li>
<li><p>32 are novel gene by Deep-MOTIFs, 22 of which are not in any
SFARI categories.</p></li>
</ul>
<pre class="r"><code>#library(dplyr)
#ekegg@result %&gt;%
#  arrange(p.adjust) %&gt;%
#  select(ID, Description, p.adjust) %&gt;%
#  head(10)  # Top 10 enriched terms


#list top 10 enriched terms 
ekegg_result_sorted &lt;- ekegg@result[order(ekegg@result$p.adjust), ]
head(ekegg_result_sorted[, c(&quot;ID&quot;, &quot;Description&quot;, &quot;p.adjust&quot;)], 20)</code></pre>
<pre><code>               ID                                       Description
hsa04720 hsa04720                            Long-term potentiation
hsa05017 hsa05017                            Spinocerebellar ataxia
hsa04360 hsa04360                                     Axon guidance
hsa04724 hsa04724                             Glutamatergic synapse
hsa03082 hsa03082               ATP-dependent\nchromatin remodeling
hsa04919 hsa04919                 Thyroid hormone signaling pathway
hsa00310 hsa00310                                Lysine degradation
hsa05225 hsa05225                          Hepatocellular carcinoma
hsa05033 hsa05033                                Nicotine addiction
hsa05213 hsa05213                                Endometrial cancer
hsa04721 hsa04721                            Synaptic vesicle cycle
hsa04728 hsa04728                              Dopaminergic synapse
hsa05022 hsa05022 Pathways of neurodegeneration - multiple diseases
hsa04934 hsa04934                                  Cushing syndrome
hsa04916 hsa04916                                     Melanogenesis
hsa04024 hsa04024                            cAMP signaling pathway
hsa05215 hsa05215                                   Prostate cancer
hsa05224 hsa05224                                     Breast cancer
hsa05016 hsa05016                                Huntington disease
hsa04310 hsa04310                             Wnt signaling pathway
             p.adjust
hsa04720 2.170928e-06
hsa05017 1.042721e-05
hsa04360 1.042721e-05
hsa04724 1.042721e-05
hsa03082 1.042721e-05
hsa04919 1.371055e-05
hsa00310 1.371055e-05
hsa05225 3.727612e-05
hsa05033 1.787085e-04
hsa05213 3.528842e-04
hsa04721 5.370403e-04
hsa04728 5.370403e-04
hsa05022 5.730050e-04
hsa04934 6.007590e-04
hsa04916 7.103547e-04
hsa04024 8.274028e-04
hsa05215 9.813542e-04
hsa05224 1.186882e-03
hsa05016 1.413170e-03
hsa04310 1.485627e-03</code></pre>
<pre class="r"><code>pdf(&quot;cnetplot_all56terms.pdf&quot;, width = 16, height = 12)
cnetplot(ekegg, showCategory = 56, circular = TRUE, colorEdge = TRUE)
dev.off()</code></pre>
</div>
</div>
</div>
<div id="novel-pathways-in-motifs" class="section level2">
<h2>novel pathways in MOTIFs</h2>
<pre class="r"><code>pvalue_threshold=0.05
gostres2_all &lt;- gost(query = list(&quot;Deep-MOTIFs genes&quot;=top_genes, &quot;SFARI genes&quot;=positive_genes), 
organism = &quot;hsapiens&quot;, significant = F, correction_method = &quot;bonferroni&quot;)</code></pre>
<pre class="r"><code>highlight_terms_motif_term_id=gostres2_all$result %&gt;% filter(query==&quot;Deep-MOTIFs genes&quot;)%&gt;% filter(p_value&lt;pvalue_threshold)  %&gt;% dplyr::select(term_id)%&gt;% pull()
highlight_terms_sfari_term_id=gostres2_all$result %&gt;% filter(query==&quot;SFARI genes&quot;)%&gt;% filter(p_value&gt;pvalue_threshold)%&gt;% dplyr::select(term_id)%&gt;% pull()
common_term_id=intersect(highlight_terms_motif_term_id, highlight_terms_sfari_term_id)


highlight_terms=gostres2_all$result %&gt;% filter (term_id %in% common_term_id) %&gt;% dplyr::select(term_id, term_name, query, p_value)</code></pre>
<pre class="r"><code>highlight_terms_wide=highlight_terms%&gt;%  # convert from long format to wide format 
  pivot_wider(
    names_from = query,
    values_from = p_value,
    names_prefix = &quot;pvalue_&quot;
  )
highlight_terms_wide=highlight_terms_wide %&gt;% drop_na()
dim(highlight_terms_wide)

highlight_terms_wide%&gt;% 
datatable(extensions = &#39;Buttons&#39;,
            options = list(dom = &#39;Blfrtip&#39;,
                           buttons = c(&#39;copy&#39;, &#39;csv&#39;, &#39;excel&#39;, &#39;pdf&#39;, &#39;print&#39;),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,&quot;All&quot;))))</code></pre>
<pre class="r"><code>highlight_terms_wide &lt;- highlight_terms_wide %&gt;%
  mutate(group = case_when(
    grepl(&quot;^GO:&quot;, term_id) ~ &quot;GO&quot;,
    grepl(&quot;^HPA:&quot;, term_id) ~ &quot;HPA&quot;,
    grepl(&quot;^HP:&quot;, term_id) ~ &quot;HP&quot;,
    grepl(&quot;^TF:&quot;, term_id) ~ &quot;TF&quot;,
    TRUE ~ &quot;Others&quot;
  ))

top10_labels &lt;- highlight_terms_wide %&gt;%
  mutate(min_p = pmin(`pvalue_Deep-MOTIFs genes`, `pvalue_SFARI genes`)) %&gt;%
  arrange(min_p) %&gt;%
  dplyr::slice(1:10)

library(ggrepel)

ggplot(highlight_terms_wide, aes(x = -log10(`pvalue_Deep-MOTIFs genes`), 
                                 y = -log10(`pvalue_SFARI genes`), 
                                 color = group)) +
  geom_point(alpha = 0.6, size = 2) +
  #geom_abline(slope = 1, intercept = 0, linetype = &quot;solid&quot;, color = &quot;black&quot;, size=1) +
  scale_color_manual(values = c(
    &quot;GO&quot; = &quot;goldenrod2&quot;, 
    &quot;HPA&quot; = &quot;purple4&quot;, 
    &quot;HP&quot; = &quot;990&quot;, 
    &quot;TF&quot; = &quot;blue&quot;,
    &quot;Others&quot; = &quot;grey70&quot;)) +
  geom_text_repel(
    data = top10_labels,
    aes(label = term_name),
    size = 3,
    max.overlaps = Inf
  ) +
  labs(
    title = &quot;&quot;,
    x = &quot;-log10(p value): Deep-MOTIFs&quot;,
    y = &quot;-log10(p value): SFARI&quot;,
    color = &quot;Group&quot;
  ) +
  theme(legend.title = element_blank())+
  theme(plot.title = element_text(hjust = 0.5, size = 12))</code></pre>
</div>
<div id="missed-pathways-in-motifs" class="section level2">
<h2>missed pathways in MOTIFs</h2>
<pre class="r"><code>pvalue_threshold=0.05
gostres2_all &lt;- gost(query = list(&quot;Deep-MOTIFs genes&quot;=top_genes, &quot;SFARI genes&quot;=positive_genes), 
organism = &quot;hsapiens&quot;, significant = F, correction_method = &quot;bonferroni&quot;)</code></pre>
<pre class="r"><code>highlight_terms_motif_term_id=gostres2_all$result %&gt;% filter(query==&quot;Deep-MOTIFs genes&quot;)%&gt;% filter(p_value&gt;pvalue_threshold)  %&gt;% dplyr::select(term_id)%&gt;% pull()
highlight_terms_sfari_term_id=gostres2_all$result %&gt;% filter(query==&quot;SFARI genes&quot;)%&gt;% filter(p_value&lt;pvalue_threshold)%&gt;% dplyr::select(term_id)%&gt;% pull()
common_term_id=intersect(highlight_terms_motif_term_id, highlight_terms_sfari_term_id)


highlight_terms=gostres2_all$result %&gt;% filter (term_id %in% common_term_id) %&gt;% dplyr::select(term_id, term_name, query, p_value)</code></pre>
<pre class="r"><code>highlight_terms_wide=highlight_terms%&gt;%  # convert from long format to wide format 
  pivot_wider(
    names_from = query,
    values_from = p_value,
    names_prefix = &quot;pvalue_&quot;
  )
highlight_terms_wide=highlight_terms_wide %&gt;% drop_na()
dim(highlight_terms_wide)

highlight_terms_wide%&gt;% 
datatable(extensions = &#39;Buttons&#39;,
            options = list(dom = &#39;Blfrtip&#39;,
                           buttons = c(&#39;copy&#39;, &#39;csv&#39;, &#39;excel&#39;, &#39;pdf&#39;, &#39;print&#39;),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,&quot;All&quot;))))</code></pre>
<pre class="r"><code>highlight_terms_wide &lt;- highlight_terms_wide %&gt;%
  mutate(group = case_when(
    grepl(&quot;^GO:&quot;, term_id) ~ &quot;GO&quot;,
    grepl(&quot;^HPA:&quot;, term_id) ~ &quot;HPA&quot;,
    grepl(&quot;^HP:&quot;, term_id) ~ &quot;HP&quot;,
    grepl(&quot;^TF:&quot;, term_id) ~ &quot;TF&quot;,
    TRUE ~ &quot;Others&quot;
  ))

top10_labels &lt;- highlight_terms_wide %&gt;%
  mutate(min_p = pmin(`pvalue_Deep-MOTIFs genes`, `pvalue_SFARI genes`)) %&gt;%
  arrange(min_p) %&gt;%
  dplyr::slice(1:10)

library(ggrepel)

ggplot(highlight_terms_wide, aes(x = -log10(`pvalue_Deep-MOTIFs genes`), 
                                 y = -log10(`pvalue_SFARI genes`), 
                                 color = group)) +
  geom_point(alpha = 0.6, size = 2) +
  #geom_abline(slope = 1, intercept = 0, linetype = &quot;solid&quot;, color = &quot;black&quot;, size=1) +
  scale_color_manual(values = c(
    &quot;GO&quot; = &quot;goldenrod2&quot;, 
    &quot;HPA&quot; = &quot;purple4&quot;, 
    &quot;HP&quot; = &quot;990&quot;, 
    &quot;TF&quot; = &quot;blue&quot;,
    &quot;Others&quot; = &quot;grey70&quot;)) +
  geom_text_repel(
    data = top10_labels,
    aes(label = term_name),
    size = 3,
    max.overlaps = Inf
  ) +
  labs(
    title = &quot;&quot;,
    x = &quot;-log10(p value): Deep-MOTIFs&quot;,
    y = &quot;-log10(p value): SFARI&quot;,
    color = &quot;Group&quot;
  ) +
  theme(legend.title = element_blank())+
  theme(plot.title = element_text(hjust = 0.5, size = 12))</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.0 (2024-04-24 ucrt)
Platform: x86_64-w64-mingw32/x64
Running under: Windows 10 x64 (build 19045)

Matrix products: default


locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] stats4    grid      stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] enrichplot_1.24.4      org.Hs.eg.db_3.19.1    AnnotationDbi_1.66.0  
 [4] IRanges_2.38.0         S4Vectors_0.42.0       Biobase_2.64.0        
 [7] BiocGenerics_0.50.0    clusterProfiler_4.12.6 rprojroot_2.0.4       
[10] VennDiagram_1.7.3      futile.logger_1.4.3    DT_0.33               
[13] lubridate_1.9.3        forcats_1.0.0          stringr_1.5.1         
[16] dplyr_1.1.4            purrr_1.0.2            readr_2.1.5           
[19] tidyr_1.3.1            tibble_3.2.1           ggplot2_3.5.1         
[22] tidyverse_2.0.0       

loaded via a namespace (and not attached):
  [1] RColorBrewer_1.1-3      rstudioapi_0.16.0       jsonlite_1.8.8         
  [4] magrittr_2.0.3          farver_2.1.2            rmarkdown_2.26         
  [7] fs_1.6.4                zlibbioc_1.50.0         vctrs_0.6.5            
 [10] memoise_2.0.1           ggtree_3.12.0           htmltools_0.5.8.1      
 [13] lambda.r_1.2.4          gridGraphics_0.5-1      sass_0.4.9             
 [16] bslib_0.7.0             htmlwidgets_1.6.4       plyr_1.8.9             
 [19] httr2_1.0.1             futile.options_1.0.1    cachem_1.0.8           
 [22] igraph_2.0.3            lifecycle_1.0.4         pkgconfig_2.0.3        
 [25] gson_0.1.0              Matrix_1.7-0            R6_2.5.1               
 [28] fastmap_1.1.1           GenomeInfoDbData_1.2.12 digest_0.6.35          
 [31] aplot_0.2.5             colorspace_2.1-0        patchwork_1.3.0        
 [34] crosstalk_1.2.1         RSQLite_2.3.6           labeling_0.4.3         
 [37] fansi_1.0.6             timechange_0.3.0        httr_1.4.7             
 [40] polyclip_1.10-7         compiler_4.4.0          bit64_4.0.5            
 [43] withr_3.0.0             BiocParallel_1.38.0     viridis_0.6.5          
 [46] DBI_1.2.3               highr_0.10              ggforce_0.4.2          
 [49] R.utils_2.12.3          MASS_7.3-60.2           rappdirs_0.3.3         
 [52] tools_4.4.0             scatterpie_0.2.4        ape_5.8-1              
 [55] httpuv_1.6.15           R.oo_1.27.0             glue_1.7.0             
 [58] nlme_3.1-164            GOSemSim_2.30.2         promises_1.3.0         
 [61] shadowtext_0.1.4        reshape2_1.4.4          fgsea_1.30.0           
 [64] generics_0.1.3          gtable_0.3.5            tzdb_0.4.0             
 [67] R.methodsS3_1.8.2       data.table_1.15.4       hms_1.1.3              
 [70] tidygraph_1.3.1         utf8_1.2.4              XVector_0.44.0         
 [73] ggrepel_0.9.5           pillar_1.9.0            yulab.utils_0.2.0      
 [76] later_1.3.2             splines_4.4.0           tweenr_2.0.3           
 [79] treeio_1.28.0           lattice_0.22-6          bit_4.0.5              
 [82] tidyselect_1.2.1        GO.db_3.19.1            Biostrings_2.72.0      
 [85] knitr_1.46              git2r_0.33.0            gridExtra_2.3          
 [88] xfun_0.43               graphlayouts_1.2.2      stringi_1.8.4          
 [91] UCSC.utils_1.0.0        lazyeval_0.2.2          ggfun_0.1.8            
 [94] workflowr_1.7.1         yaml_2.3.8              evaluate_0.23          
 [97] codetools_0.2-20        ggraph_2.2.1            qvalue_2.36.0          
[100] ggplotify_0.1.2         cli_3.6.2               munsell_0.5.1          
[103] jquerylib_0.1.4         Rcpp_1.0.12             GenomeInfoDb_1.40.0    
[106] png_0.1-8               parallel_4.4.0          blob_1.2.4             
[109] DOSE_3.30.5             tidytree_0.4.6          viridisLite_0.4.2      
[112] scales_1.3.0            crayon_1.5.2            rlang_1.1.3            
[115] cowplot_1.1.3           fastmatch_1.1-6         KEGGREST_1.44.0        
[118] formatR_1.14           </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
