---
title: "deep motif score distribution"
output:
  workflowr::wflow_html:
    code_folding: hide
    toc: true
date: "2025-02-11"
---



```{r, message=F, warning=F}
rm(list=ls())
set.seed(123)
library(tidyverse)
library(ggplot2)
library(DT)
library(rprojroot)
root <- rprojroot::find_rstudio_root_file()
```



```{r, message=F, warning=F}
#result_round1=as_tibble(read.csv(file.path(root, "../DiWangResults/ASD_prediction_results/ASD_RISK_PREDICTION_round1.csv"))) # old results
result_round1=as_tibble(read.csv(file.path(root, "../DiWangResults/results updated_20250303/ASD_RISK_PREDICTION round 1.csv")))
result_round1=result_round1[result_round1['gene'] != 0,]  # remove rows with gene=0

forecASD_results=as_tibble(read.csv(file.path(root, "../DiWangResults/forecASD_results.csv")))
forecASD_results=forecASD_results[forecASD_results['gene'] != 0,]  # remove rows with gene=0
```



## histogram of scores 



```{r, message=F, warning=F}
ggplot(result_round1, aes(x = result_round1$RISK_PROBABILITY_FUSION)) +
  geom_histogram(binwidth = 0.1, fill = "blue", color = "black", aes(y = ..count..)) +
  stat_bin(binwidth = 0.1, geom = "text", aes(label = ..count..), vjust = -0.5, color = "red") +
  labs(title = "Histogram of deep motif scores", x = "deep motif score", y = "")+
  geom_vline(xintercept = 0.5, color = "green", linetype = "dashed", size = 1)+
  theme(plot.title = element_text(hjust = 0.5, size=15))  #center the title 

```



```{r, message=F, warning=F}
ggplot(forecASD_results, aes(x = forecASD_results$risk.score)) +
  geom_histogram(binwidth = 0.1, fill = "blue", color = "black", aes(y = ..count..)) +
  stat_bin(binwidth = 0.1, geom = "text", aes(label = ..count..), vjust = -0.5, color = "red") +
  labs(title = "Histogram of forecASD scores", x = "forecASD score", y = "")+
  geom_vline(xintercept = 0.5, color = "green", linetype = "dashed", size = 1)+
  theme(plot.title = element_text(hjust = 0.5, size=15))  #center the title 

```



```{r, message=F, warning=F}

# Combine the data into a data frame
data <- data.frame(
  scores = c(result_round1$RISK_PROBABILITY_FUSION, forecASD_results$risk.score),
  group = factor(c(rep("deep-motifs",length(result_round1$RISK_PROBABILITY_FUSION)), rep("forecASD", length(forecASD_results$risk.score))))
)

# Load ggplot2 for plotting
library(ggplot2)

# Create the histogram with side-by-side bars
ggplot(data, aes(x = scores, fill = group)) +
  geom_histogram(position = "dodge", bins = 20, alpha = 0.7) +
  #geom_histogram(position = "dodge", bins = 20, color = "black") +
  #stat_bin(aes(label = ..count..), geom = "text", position = position_dodge(width = 0.9), vjust = -0.5) +
  labs(title = "Histogram of scores by forecASD and deep-motifs", x = "scores", y = "")+
  geom_vline(xintercept = 0.5, color = "green", linetype = "dashed", size = 1)+
  theme(plot.title = element_text(hjust = 0.5, size=15))  #center the title 
```



## implicated risk genes 


```{r, message=F, warning=F}
score_cutoff=0.5


SFARI_genes=as_tibble(read.csv(file.path(root, "../../../Dataset/SFARI_base/SFARI-Gene_genes_08-19-2024release_09-16-2024export.csv")))
positive_genes=SFARI_genes %>% filter (gene.score=="1") %>% dplyr::select(gene.symbol) %>% pull()


motifs_genes=result_round1 %>% filter(RISK_PROBABILITY_FUSION>score_cutoff) %>% dplyr::select(gene) %>% pull()



SFARI_score_for_motif_genes <- ifelse(
  motifs_genes %in% SFARI_genes$gene.symbol,
  SFARI_genes$gene.score[match(motifs_genes, SFARI_genes$gene.symbol)],
  NA
)

result_round1 %>% filter(RISK_PROBABILITY_FUSION>score_cutoff) %>% dplyr::select(gene, gene_gencodeV33, chromosome, RISK_PROBABILITY_FUSION)%>% mutate(SFARI_score=SFARI_score_for_motif_genes) %>% arrange(desc(RISK_PROBABILITY_FUSION)) %>% 
datatable(extensions = 'Buttons',
            options = list(dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'print'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))


SFARI_table=table(SFARI_score_for_motif_genes, useNA = "ifany")


# Define labels, replacing NA with a readable string
labels <- names(SFARI_table)
labels[is.na(labels)] <- "NA"

# Append counts to labels
labels <- paste(labels, "(", SFARI_table, ")", sep = "")

# Create a pie chart
pie(SFARI_table, labels = labels, main = "break-down of motifs genes by SFARI categories", col = rainbow(length(SFARI_table)))
```



* with score cutoff 0.5, 447 genes are selected. 




```{r, message=F, warning=F}
score_cutoff=0.5


SFARI_genes=as_tibble(read.csv(file.path(root, "../../../Dataset/SFARI_base/SFARI-Gene_genes_08-19-2024release_09-16-2024export.csv")))
positive_genes=SFARI_genes %>% filter (gene.score=="1") %>% dplyr::select(gene.symbol) %>% pull()


forecASD_genes= forecASD_results%>% filter(risk.score>score_cutoff) %>% dplyr::select(gene) %>% pull()



SFARI_score_for_forecASD_genes <- ifelse(
  forecASD_genes %in% SFARI_genes$gene.symbol,
  SFARI_genes$gene.score[match(forecASD_genes, SFARI_genes$gene.symbol)],
  NA
)

forecASD_results %>% filter(risk.score>score_cutoff) %>% dplyr::select(gene, gene_gencodeV33, chromosome, risk.score)%>% mutate(SFARI_score=SFARI_score_for_forecASD_genes) %>% arrange(desc(risk.score)) %>% 
datatable(extensions = 'Buttons',
            options = list(dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'print'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))


SFARI_table=table(SFARI_score_for_forecASD_genes, useNA = "ifany")


# Define labels, replacing NA with a readable string
labels <- names(SFARI_table)
labels[is.na(labels)] <- "NA"

# Append counts to labels
labels <- paste(labels, "(", SFARI_table, ")", sep = "")

# Create a pie chart
pie(SFARI_table, labels = labels, main = "break-down of forecASD genes by SFARI categories", col = rainbow(length(SFARI_table)))
```


* with cutoff 0.5, forecASD identified 471 risk genes. 


## deep-motifs new genes not in SFARI cate 1 


```{r, message=F, warning=F}
SFARI_genes=as_tibble(read.csv(file.path(root, "../../../Dataset/SFARI_base/SFARI-Gene_genes_08-19-2024release_09-16-2024export.csv")))

positive_genes=SFARI_genes %>% filter (gene.score=="1") %>% dplyr::select(gene.symbol) %>% pull()
positive_genes=intersect(positive_genes, result_round1$gene)

score_cutoff=0.5

motifs_new_genes=result_round1 %>% filter(RISK_PROBABILITY_FUSION>score_cutoff) %>% dplyr::select(gene)%>% filter (! gene %in% positive_genes) %>% pull()

length(result_round1 %>% filter(RISK_PROBABILITY_FUSION>score_cutoff) %>% dplyr::select(gene)%>% filter (gene %in% positive_genes) %>% pull())


SFARI_score_for_motif_new_genes <- ifelse(
  motifs_new_genes %in% SFARI_genes$gene.symbol,
  SFARI_genes$gene.score[match(motifs_new_genes, SFARI_genes$gene.symbol)],
  NA
)

sum(is.na(SFARI_score_for_motif_new_genes))
  
result_round1 %>% filter(RISK_PROBABILITY_FUSION>score_cutoff) %>% dplyr::select(gene, gene_gencodeV33, chromosome, RISK_PROBABILITY_FUSION)%>% filter (! gene %in% positive_genes)%>% mutate (SFARI_score=SFARI_score_for_motif_new_genes) %>%arrange(desc(RISK_PROBABILITY_FUSION)) %>% 
datatable(extensions = 'Buttons',
          caption = "new ASD genes (not in SFARI cate 1)", 
            options = list(dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'print'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))
```


* 233 genes are in SFARI category 1, but 208 are used and appear in final result. Among 208 genes, 206 are captured by motifs, being among 447 genes, and 2 genes ("ACTB", "PCCB") are missed.  

* among 241 new ASD genes (not in cate 1), 154 are not in SFARI list at all. 



## ranking of deep-motifs genes vs forecASD genes 



```{r, message=F, warning=F}
total_risk_genes=unique(c(forecASD_genes, motifs_genes))
result_round1_sorted=result_round1 %>% arrange(desc(RISK_PROBABILITY_FUSION))
forecASD_results_sorted=forecASD_results %>% arrange(desc(risk.score))

total_risk_genes_rank1 <- match(total_risk_genes, result_round1_sorted$gene)
total_risk_genes_rank2=match(total_risk_genes, forecASD_results_sorted$gene)


# Combine into a data frame
df <- data.frame(
  Gene = total_risk_genes,
  Rank1_forecASD = total_risk_genes_rank2,
  Rank2_deep_motifs = total_risk_genes_rank1
)

# Reshape data for plotting
df_long <- df %>%
  tidyr::pivot_longer(cols = c(Rank1_forecASD, Rank2_deep_motifs), names_to = "Ranking", values_to = "Rank")

# Plot the slope graph with gene labels on both sides
ggplot(df_long, aes(x = Ranking, y = Rank, group = Gene)) +
  geom_line(aes(color = Gene), size = 1) +  # Connecting lines
  geom_point(size = 1) +                    # Points at each ranking
  # Add gene labels on the left side for Rank1
  geom_text(data = df_long %>% filter(Ranking == "Rank1_forecASD"), 
            aes(label = Gene), 
            hjust = 1.2, size = 2) +
  # Add gene labels on the right side for Rank2
  geom_text(data = df_long %>% filter(Ranking == "Rank2_deep_motifs"), 
            aes(label = Gene), 
            hjust = -0.2, size = 2) +
  scale_y_reverse() +                       # Reverse y-axis for ranking
  labs(title = "", x = "", y = "Rank") +
  theme_minimal() +
  theme(legend.position = "none")            # Remove legend for clarity


# Perform the test
result <- wilcox.test(total_risk_genes_rank1, total_risk_genes_rank2, paired = TRUE, alternative = "two.sided")

# Display the results
print(result)

```





## genes overlapping among deep motifs, forecASD, and SFARI



```{r, message=F, warning=F}

x=list(
  MOTIFs_gene=motifs_genes,
  #Autism_gene=as_tibble(neuro_disorers) %>% filter(Category=="ASD genes") %>% dplyr::select(`Gene name`) %>% pull(),
  ForecASD_gene=forecASD_genes,
  SFARI_genes=positive_genes

)
library(VennDiagram)

display_venn <- function(x, ...){
  library(VennDiagram)
  grid.newpage()
  venn_object <- venn.diagram(x, filename = NULL, ...)
  grid.draw(venn_object)
}

#png("C:\\Users\\hans\\OneDrive - Marquette University\\SCZ-DAWN\\Figures\\Fig_overlapping_genes.png")

display_venn(
        x,
    #    category.names = c("MIRAGE FDR genes" , "Autism genes" , "DD/ID genes", "SCZ GWAS genes"),
    category.names = c("MOTIFs genes" , "ForecASD genes" , "SFARI genes"),
        # Circles
        lwd = 2,
        lty = 'blank',
     #   fill = c("#999999", "#E69F00", "#56B4E9", "#009E73"),
    fill = c("red", "blue", "green"),
        # Numbers
        cex = 1.5,
        fontface = "italic",
        # Set names
        cat.cex = 1.5,
        cat.fontface = "bold",
        cat.default.pos = "outer",
       # cat.dist = c(0.055, 0.055, 0.1, 0.1, 0.3 )
    cat.pos = c(-20, 20, 180),  # Adjust angles as needed
         cat.dist = c(0.055, 0.055, -0.15) # Adjust distances from the circles
)
```
