---
title: "deep motif score distribution"
output:
  workflowr::wflow_html:
    code_folding: hide
    toc: true
date: "2025-02-11"
---



```{r, message=F, warning=F}
rm(list=ls())
set.seed(123)
library(tidyverse)
library(ggplot2)
library(DT)
library(rprojroot)
root <- rprojroot::find_rstudio_root_file()
```



```{r, message=F, warning=F}
#result_round1=as_tibble(read.csv(file.path(root, "../DiWangResults/ASD_prediction_results/ASD_RISK_PREDICTION_round1.csv"))) # old results
result_round1=as_tibble(read.csv(file.path(root, "../DiWangResults/results updated_20250303/ASD_RISK_PREDICTION round 1.csv")))
result_round1=result_round1[result_round1['gene'] != 0,]  # remove rows with gene=0

forecASD_results=as_tibble(read.csv(file.path(root, "../DiWangResults/forecASD_results.csv")))
forecASD_results=forecASD_results[forecASD_results['gene'] != 0,]  # remove rows with gene=0
```



## histogram of scores 



```{r, message=F, warning=F}
hist_plot1=ggplot(result_round1, aes(x = result_round1$RISK_PROBABILITY_FUSION)) +
  geom_histogram(binwidth = 0.1, fill = "blue", color = "black", aes(y = ..count..)) +
  stat_bin(binwidth = 0.1, geom = "text", aes(label = ..count..), vjust = -0.5, color = "red") +
  #geom_density(color = "darkorange", size = .2) +
  labs(title = "Histogram of Deep MOTIFs scores", x = "Deep MOTIFs score", y = "Number of genes")+
  geom_vline(xintercept = 0.5, color = "green", linetype = "dashed", size = 1)+
  theme(plot.title = element_text(hjust = 0.5, size=15))  #center the title 
hist_plot1
```



```{r, message=F, warning=F, eval=F}
## find gene coordinates 
library(biomaRt)

# Connect to the Ensembl database
ensembl <- useMart("ensembl", dataset = "hsapiens_gene_ensembl")
gene_names=result_round1$gene

# Query gene coordinates using biomaRt
gene_coords <- getBM(
  attributes = c("hgnc_symbol", "chromosome_name", "start_position", "end_position", "strand"),
  filters = "hgnc_symbol",
  values = gene_names,
  mart = ensembl
)
colnames(gene_coords)[1]="gene"

result_round1_with_coordinate=result_round1 %>% right_join(gene_coords, by="gene")



motifs_manhattan=data.frame(
  chr=result_round1_with_coordinate$chromosome,
  BP=result_round1_with_coordinate$start_position,
  score=result_round1_with_coordinate$RISK_PROBABILITY_FUSION, 
  gene=result_round1_with_coordinate$gene
)
motifs_manhattan=motifs_manhattan %>% drop_na()
library(qqman)

# Plot the Manhattan plot
manhattan(
  motifs_manhattan,
  chr = "chr",           # Chromosome column
  bp = "BP",             # Base pair position column
  p = "score",           # Risk score column
  snp = "gene",          # Gene name (optional)
  ylim = c(0, 1),       # Adjust y-axis limit for better visualization
  main = "Manhattan Plot of Gene Risk Scores",
  xlab = "Chromosome",
  ylab = "Deep-MOTIFs scores",
  col = c("blue", "red"), # Alternating colors for chromosomes
  suggestiveline = 0.9,  # Add a suggestive line (optional)
  genomewideline = 0.5   # Add a genome-wide significance line (optional)
)
```


```{r, message=F, warning=F, eval=F}
library(ggrepel)
gwasResults=data.frame(Gene=result_round1_with_coordinate$gene, post_prob=result_round1_with_coordinate$RISK_PROBABILITY_FUSION, chr=result_round1_with_coordinate$chromosome, pos=result_round1_with_coordinate$start_position)
gwasResults=gwasResults %>% drop_na()

pp_threshold <- 0.5 

don <- gwasResults %>% 
  
  # Compute chromosome size
  group_by(chr) %>% 
  summarise(chr_len=max(pos)) %>% 
  
  # Calculate cumulative position of each chromosome
  mutate(tot=cumsum(as.numeric(chr_len))-as.numeric(chr_len)) %>%
  dplyr::select(-chr_len) %>%
  
  # Add this info to the initial dataset
  left_join(gwasResults, ., by=c("chr"="chr")) %>%
  
  # Add a cumulative position of each SNP
  arrange(chr, pos) %>%
  mutate( poscum=pos+tot) %>%
  mutate( is_annotate=ifelse(Gene %in% gwasResults$Gene[gwasResults$post_prob > 0.98], "yes", "no"))

# Then we need to prepare the X axis. Indeed we do not want to display the cumulative position of SNP in bp, but just show the chromosome name instead.
axisdf = don %>% group_by(chr) %>% summarize(center=( max(poscum) + min(poscum) ) / 2 )

manhplot <- ggplot(don, aes(x=poscum, y=post_prob)) +
  
  # Show all points
  geom_point( aes(color=as.factor(chr)), alpha=2, size=2) +
  scale_color_manual(values = rep(c("grey", "steelblue"), 22 )) +
  
  # custom X axis:
  scale_x_continuous( label = axisdf$chr, breaks= axisdf$center ) +
  #scale_y_continuous(expand = c(0, 0) ) +     # remove space between plot area and x axis
  
  ylab("Deep MOTIFS score") +
  #ylim(0,max(log10(don$BF)) + 2) +
  ylim(0,1.2) +
  xlab("Chromosome") +
  
  geom_hline(yintercept = pp_threshold, col = "red") +
  annotate("text", x = Inf, y = pp_threshold, label = sprintf("MOTIFS score = %s", pp_threshold), hjust = 1.1, vjust = -1, color = "red", size = 7) +
  
  # Custom the theme:
  theme_bw() +
  #ggtitle(i) + theme(plot.title = element_text(hjust = 0.5)) +
  theme( 
    legend.position="none",
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.title.x = element_text(size = 28),
    axis.text.x = element_text(size = 26, color = "black"),
    axis.title.y = element_text(size = 28),
    axis.text.y = element_text(size = 26, color = "black"),
  ) +
  geom_label_repel( data=subset(don, is_annotate=="yes"), aes(label=Gene), size=3)
manhplot
```










```{r, message=F, warning=F}
hist_plot2=ggplot(forecASD_results, aes(x = forecASD_results$risk.score)) +
  geom_histogram(binwidth = 0.1, fill = "blue", color = "black", aes(y = ..count..)) +
  stat_bin(binwidth = 0.1, geom = "text", aes(label = ..count..), vjust = -0.5, color = "red") +
  labs(title = "Histogram of forecASD scores", x = "forecASD score", y = "")+
  geom_vline(xintercept = 0.5, color = "green", linetype = "dashed", size = 1)+
  theme(plot.title = element_text(hjust = 0.5, size=15))  #center the title 
hist_plot2
```



```{r, message=F, warning=F}

# Combine the data into a data frame
data <- data.frame(
  scores = c(result_round1$RISK_PROBABILITY_FUSION, forecASD_results$risk.score),
  group = factor(c(rep("Deep-MOTIFs",length(result_round1$RISK_PROBABILITY_FUSION)), rep("forecASD", length(forecASD_results$risk.score))))
)


# Create the histogram with side-by-side bars
ggplot(data, aes(x = scores, fill = group)) +
  geom_histogram(position = "dodge", bins = 20, alpha = 0.7) +
  stat_bin(
    aes(label = ..count..),
    geom = "text",
    position = position_dodge(width = 0.1),
    vjust =-0.5,
    size = 2, 
    bins = 20
  ) +
  labs(title = "", x = "scores", y = "Number of genes")+
  geom_vline(xintercept = 0.5, color = "green", linetype = "dashed", size = 1)+
  theme(plot.title = element_text(hjust = 0.5, size=15))+  #center the title 
  theme(legend.position = "bottom")
```

```{r, message=F, warning=F}
library(patchwork)
# Combine plots side by side
hist_combined <- hist_plot1 + hist_plot2

# Or arrange vertically: p1 / p2
print(hist_combined)
```









## implicated risk genes 


```{r, message=F, warning=F}
score_cutoff=0.5


SFARI_genes=as_tibble(read.csv(file.path(root, "../../../Dataset/SFARI_base/SFARI-Gene_genes_08-19-2024release_09-16-2024export.csv")))
positive_genes=SFARI_genes %>% filter (gene.score=="1") %>% dplyr::select(gene.symbol) %>% pull()


motifs_genes=result_round1 %>% filter(RISK_PROBABILITY_FUSION>score_cutoff) %>% dplyr::select(gene) %>% pull()



SFARI_score_for_motif_genes <- ifelse(
  motifs_genes %in% SFARI_genes$gene.symbol,
  SFARI_genes$gene.score[match(motifs_genes, SFARI_genes$gene.symbol)],
  NA
)

result_round1 %>% filter(RISK_PROBABILITY_FUSION>score_cutoff) %>% dplyr::select(gene, gene_gencodeV33, chromosome, RISK_PROBABILITY_FUSION)%>% mutate(SFARI_score=SFARI_score_for_motif_genes) %>% arrange(desc(RISK_PROBABILITY_FUSION)) %>% 
datatable(extensions = 'Buttons',
            options = list(dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'print'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))


SFARI_table=table(SFARI_score_for_motif_genes, useNA = "ifany")

SFARI_table

```
 
 
 
 
```{r, message=F, warning=F}
 
 # Create data frame
df <- data.frame(
  score = c("1", "2", "3", "NA"),
  count = c(206, 77, 10, 154)
)

# Sort descending by count (important for label matching pie slices)
df <- df %>%
  arrange(desc(score)) %>%
  mutate(
    ypos = cumsum(count) - count / 2
  )

# Create pie chart with count labels
pie_fig1=ggplot(df, aes(x = "", y = count, fill = score)) +
  geom_col(width = 1) +
  coord_polar(theta = "y") +
  geom_text(aes(y = ypos, label = count), color = "black", size = 5) +
  labs(title = "Deep MOTIFs Genes (447)", fill = "SFARI score") +
  theme_void() +
  theme(plot.title = element_text(hjust = 0.5, size=15))+  #center the title
  theme(legend.position = "right")
 pie_fig1
```



* with score cutoff 0.5, 447 genes are selected. 




```{r, message=F, warning=F}
score_cutoff=0.5


SFARI_genes=as_tibble(read.csv(file.path(root, "../../../Dataset/SFARI_base/SFARI-Gene_genes_08-19-2024release_09-16-2024export.csv")))
positive_genes=SFARI_genes %>% filter (gene.score=="1") %>% dplyr::select(gene.symbol) %>% pull()


forecASD_genes= forecASD_results%>% filter(risk.score>score_cutoff) %>% dplyr::select(gene) %>% pull()



SFARI_score_for_forecASD_genes <- ifelse(
  forecASD_genes %in% SFARI_genes$gene.symbol,
  SFARI_genes$gene.score[match(forecASD_genes, SFARI_genes$gene.symbol)],
  NA
)

forecASD_results %>% filter(risk.score>score_cutoff) %>% dplyr::select(gene, gene_gencodeV33, chromosome, risk.score)%>% mutate(SFARI_score=SFARI_score_for_forecASD_genes) %>% arrange(desc(risk.score)) %>% 
datatable(extensions = 'Buttons',
            options = list(dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'print'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))


SFARI_table=table(SFARI_score_for_forecASD_genes, useNA = "ifany")

SFARI_table
```


* with cutoff 0.5, forecASD identified 471 risk genes. 



```{r, message=F, warning=F}
 
 # Create data frame
df <- data.frame(
  score = c("1", "2", "3", "NA"),
  count = c(176, 50, 9, 236)
)

# Sort descending by count (important for label matching pie slices)
df <- df %>%
  arrange(desc(score)) %>%
  mutate(
    ypos = cumsum(count) - count / 2
  )

# Create pie chart with count labels
pie_fig2=ggplot(df, aes(x = "", y = count, fill = score)) +
  geom_col(width = 1) +
  coord_polar(theta = "y") +
  geom_text(aes(y = ypos, label = count), color = "black", size = 5) +
  labs(title = "forecASD Genes (471)", fill = "SFARI score") +
  theme_void() +
  theme(plot.title = element_text(hjust = 0.5, size=15))+  #center the title
  theme(legend.position = "right")
pie_fig2 
```


```{r, message=F, warning=F}
library(patchwork)
pie_combine=pie_fig1+pie_fig2
print(pie_combine)
```



## deep-motifs new genes not in SFARI cate 1 


```{r, message=F, warning=F}
SFARI_genes=as_tibble(read.csv(file.path(root, "../../../Dataset/SFARI_base/SFARI-Gene_genes_08-19-2024release_09-16-2024export.csv")))

positive_genes=SFARI_genes %>% filter (gene.score=="1") %>% dplyr::select(gene.symbol) %>% pull()
positive_genes=intersect(positive_genes, result_round1$gene)

score_cutoff=0.5

motifs_new_genes=result_round1 %>% filter(RISK_PROBABILITY_FUSION>score_cutoff) %>% dplyr::select(gene)%>% filter (! gene %in% positive_genes) %>% pull()

length(result_round1 %>% filter(RISK_PROBABILITY_FUSION>score_cutoff) %>% dplyr::select(gene)%>% filter (gene %in% positive_genes) %>% pull())


SFARI_score_for_motif_new_genes <- ifelse(
  motifs_new_genes %in% SFARI_genes$gene.symbol,
  SFARI_genes$gene.score[match(motifs_new_genes, SFARI_genes$gene.symbol)],
  NA
)

sum(is.na(SFARI_score_for_motif_new_genes))
  
result_round1 %>% filter(RISK_PROBABILITY_FUSION>score_cutoff) %>% dplyr::select(gene, gene_gencodeV33, chromosome, RISK_PROBABILITY_FUSION)%>% filter (! gene %in% positive_genes)%>% mutate (SFARI_score=SFARI_score_for_motif_new_genes) %>%arrange(desc(RISK_PROBABILITY_FUSION)) %>% 
datatable(extensions = 'Buttons',
          caption = "new ASD genes (not in SFARI cate 1)", 
            options = list(dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'print'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))
```


* 233 genes are in SFARI category 1, but 208 are used and appear in final result. Among 208 genes, 206 are captured by motifs, being among 447 genes, and 2 genes ("ACTB", "PCCB") are missed.  

* among 241 new ASD genes (not in cate 1), 154 are not in SFARI list at all. 



## ranking of deep-motifs genes vs forecASD genes 



```{r, message=F, warning=F}
total_risk_genes=unique(c(forecASD_genes, motifs_genes))
result_round1_sorted=result_round1 %>% arrange(desc(RISK_PROBABILITY_FUSION))
forecASD_results_sorted=forecASD_results %>% arrange(desc(risk.score))

total_risk_genes_rank1 <- match(total_risk_genes, result_round1_sorted$gene)
total_risk_genes_rank2=match(total_risk_genes, forecASD_results_sorted$gene)


# Combine into a data frame
df <- data.frame(
  Gene = total_risk_genes,
  Rank1_forecASD = total_risk_genes_rank2,
  Rank2_Deep_MOTIFs = total_risk_genes_rank1
)

# Reshape data for plotting
df_long <- df %>%
  tidyr::pivot_longer(cols = c(Rank1_forecASD, Rank2_Deep_MOTIFs), names_to = "Ranking", values_to = "Rank")

# Join df_long with SFARI_genes on gene symbol
df_long=df_long %>% left_join(SFARI_genes %>% dplyr::select(gene.symbol, gene.score), 
            by = c("Gene" = "gene.symbol"))


# Convert gene.score to a factor
df_long$gene.score <- factor(df_long$gene.score, levels = c("NA", "1", "2", "3"))

gene.score_colors <- c(
  "NA" = "#1f77b4",  # blue
  "3" = "blue",  # orange
  "2" = "#2ca02c",  # green
  "1" = "#d62728"   # red
  # Add more as needed
)

# Plot the slope graph with gene labels on both sides
ggplot(df_long, aes(x = Ranking, y = Rank, group = Gene)) +
  geom_line(aes(color = gene.score), size = 1) +  # Connecting lines
  geom_point(size = 1) +     # Points at each ranking
  
  # Labels for left side
  geom_text(data = df_long %>% filter(Ranking == "Rank1_forecASD"),
            aes(label = Gene), 
            hjust = 1.2, size = 2) +
  
   #Labels for right side
  geom_text(data = df_long %>% filter(Ranking == "Rank2_Deep_MOTIFs"),
            aes(label = Gene), 
            hjust = -0.2, size = 2) +
  
  scale_color_manual(values = gene.score_colors, # 🎨 Custom colors
                     name = "SFARI gene score"
                     ) +  
  
  scale_y_reverse() +                       # Reverse y-axis for ranking
  labs(title = "", x = "", y = "Rank") +
  theme_minimal() +
  theme(legend.position = "right")            # Remove legend for clarity


# Perform the test
result <- wilcox.test(total_risk_genes_rank1, total_risk_genes_rank2, paired = TRUE, alternative = "two.sided")

# Display the results
print(result)

```





## genes overlapping among deep motifs, forecASD, and SFARI



```{r, message=F, warning=F}

x=list(
  MOTIFs_gene=motifs_genes,
  #Autism_gene=as_tibble(neuro_disorers) %>% filter(Category=="ASD genes") %>% dplyr::select(`Gene name`) %>% pull(),
  ForecASD_gene=forecASD_genes,
  SFARI_genes=positive_genes

)
library(VennDiagram)

display_venn <- function(x, ...){
  library(VennDiagram)
  grid.newpage()
  venn_object <- venn.diagram(x, filename = NULL, ...)
  grid.draw(venn_object)
}

#png("C:\\Users\\hans\\OneDrive - Marquette University\\SCZ-DAWN\\Figures\\Fig_overlapping_genes.png")

display_venn(
        x,
    category.names = c("Deep MOTIFs genes" , "forecASD genes" , "SFARI genes"),
        # Circles
        lwd = 2,
        lty = 'blank',
     #   fill = c("#999999", "#E69F00", "#56B4E9", "#009E73"),
    fill = c("red", "blue", "green"),
        # Numbers
        cex = 1.5,
        fontface = "italic",
        # Set names
        cat.cex = 1.5,
        cat.fontface = "bold",
        cat.default.pos = "outer",
       # cat.dist = c(0.055, 0.055, 0.1, 0.1, 0.3 )
    cat.pos = c(-20, 20, 180),  # Adjust angles as needed
         cat.dist = c(0.055, 0.055, -0.15) # Adjust distances from the circles
)
```
