---
title: "enrichment"
output:
  workflowr::wflow_html:
    code_folding: hide
    toc: true
date: "2025-02-12"
---



```{r, message=F, warning=F}
rm(list=ls())
set.seed(123)
library(tidyverse)
library(ggplot2)
library(DT)
library(VennDiagram)
library(rprojroot)
root <- rprojroot::find_rstudio_root_file()
```



```{r, message=F, warning=F}
#result_round1=as_tibble(read.csv(file.path(root, "../DiWangResults/ASD_prediction_results/ASD_RISK_PREDICTION_round1.csv"))) # old results 

result_round1=as_tibble(read.csv(file.path(root, "../DiWangResults/results updated_20250303/ASD_RISK_PREDICTION round 1.csv")))
result_round1=result_round1[result_round1['gene'] != 0,]  # remove rows with gene=0
```


## enrichment of risk genes with risk probability >0.5 with autism genes in Fu paper 


```{r, message=F, warning=F}
score_cutoff=0.5
top_genes=result_round1 %>% filter(RISK_PROBABILITY_FUSION>score_cutoff) %>% select(gene_gencodeV33) %>% pull

jack_fu=as_tibble(read.csv(file.path(root, "../../../Dataset/20231221_PMID_35982160_JackFu/Fu 2022 Suppl-41588_2022_1104_MOESM3_ESM.csv")))
jack_fu_185_asd_genes=jack_fu %>% filter(ASD185=="TRUE") %>% select(gene_gencodeV33) %>% pull()

length(setdiff(jack_fu_185_asd_genes, top_genes))

length(setdiff(jack_fu_185_asd_genes, result_round1$gene))

 #Create a Venn diagram
venn.plot <- draw.pairwise.venn(
  area1 = length(top_genes), 
  area2 = length(jack_fu_185_asd_genes), 
  cross.area = length(intersect(top_genes, jack_fu_185_asd_genes)), 
  category = c("top mitif genes", "185 ASD genes"),
  fill = c("blue", "red"),
  alpha = 0.5
)

# Display plot
grid.draw(venn.plot)

## fisher exact test 

fisher.test(matrix(c(length(result_round1$gene), length(top_genes), length(intersect(result_round1$gene, jack_fu_185_asd_genes)), length(intersect(top_genes, jack_fu_185_asd_genes))), nrow = 2))

```



## overlapping with other disorder genes 



```{r, message=F, warning=F, results=F}

multiplesheets <- function(fname) {
   
  # getting info about all excel sheets
  sheets <- readxl::excel_sheets(fname)
  tibble <- lapply(sheets, function(x) readxl::read_excel(fname, sheet = x))
  data_frame <- lapply(tibble, as.data.frame)
    
  # assigning names to data frames
  names(data_frame) <- sheets
    
  # print data frame
  print(data_frame)
}

SCHEMA_gene_results=multiplesheets("C:\\han\\Dataset\\SCHEMA\\41586_2022_4556_MOESM3_ESM.xlsx")
neuro_disorers=SCHEMA_gene_results$`Table S11 - Genes from genome-w`
DD_ID_gene=as_tibble(neuro_disorers) %>% filter(Category=="DD/ID genes") %>% dplyr::select(`Gene name`) %>% pull()
SCZ_GWAS_gene=as_tibble(neuro_disorers) %>% filter(Category=="Schizophrenia GWAS genes") %>% dplyr::select(`Gene name`) %>% pull()

x=list(
  MOTIFs_gene=top_genes,
  #Autism_gene=as_tibble(neuro_disorers) %>% filter(Category=="ASD genes") %>% dplyr::select(`Gene name`) %>% pull(),
  Autism_gene=jack_fu_185_asd_genes,
  DD_ID_gene=DD_ID_gene,
  SCZ_GWAS_gene=SCZ_GWAS_gene

)
library(VennDiagram)

display_venn <- function(x, ...){
  library(VennDiagram)
  grid.newpage()
  venn_object <- venn.diagram(x, filename = NULL, ...)
  grid.draw(venn_object)
}

#png("C:\\Users\\hans\\OneDrive - Marquette University\\SCZ-DAWN\\Figures\\Fig_overlapping_genes.png")

display_venn(
        x,
    #    category.names = c("MIRAGE FDR genes" , "Autism genes" , "DD/ID genes", "SCZ GWAS genes"),
    category.names = c("MOTIFs genes" , "Autism genes" , "DD/ID genes", "SCZ GWAS genes"),
        # Circles
        lwd = 2,
        lty = 'blank',
     #   fill = c("#999999", "#E69F00", "#56B4E9", "#009E73"),
    fill = c("red", "blue", "green", "yellow"),
        # Numbers
        cex = 1.5,
        fontface = "italic",
        # Set names
        cat.cex = 1.5,
        cat.fontface = "bold",
        cat.default.pos = "outer",
       # cat.dist = c(0.055, 0.055, 0.1, 0.1, 0.3 )
         cat.dist = c(0.055, 0.055, 0.1, 0.1 )
)
```



```{r, message=F, warning=F}
## overlap with DD/ID genes

fisher.test(matrix(c(length(result_round1$gene), length(top_genes), length(intersect(result_round1$gene, DD_ID_gene)), length(intersect(top_genes, DD_ID_gene))), nrow = 2))


## overlap with SCZ GWAS genes

fisher.test(matrix(c(length(result_round1$gene), length(top_genes), length(intersect(result_round1$gene, SCZ_GWAS_gene)), length(intersect(top_genes, SCZ_GWAS_gene))), nrow = 2))
```

