---
title: "deep motif score distribution"
output:
  workflowr::wflow_html:
    code_folding: hide
    toc: true
date: "2025-02-11"
---



```{r, message=F, warning=F}
rm(list=ls())
set.seed(123)
library(tidyverse)
library(ggplot2)
library(DT)
library(rprojroot)
root <- rprojroot::find_rstudio_root_file()
```



```{r, message=F, warning=F}
#result_round1=as_tibble(read.csv(file.path(root, "../DiWangResults/ASD_prediction_results/ASD_RISK_PREDICTION_round1.csv"))) # old results
result_round1=as_tibble(read.csv(file.path(root, "../DiWangResults/results updated_20250303/ASD_RISK_PREDICTION round 1.csv")))
result_round1=result_round1[result_round1['gene'] != 0,]  # remove rows with gene=0
```







```{r, message=F, warning=F}
ggplot(result_round1, aes(x = result_round1$RISK_PROBABILITY_FUSION)) +
  geom_histogram(binwidth = 0.1, fill = "blue", color = "black", aes(y = ..count..)) +
  stat_bin(binwidth = 0.1, geom = "text", aes(label = ..count..), vjust = -0.5, color = "red") +
  labs(title = "Histogram of deep motif scores", x = "deep motif score", y = "")+
  geom_vline(xintercept = 0.5, color = "green", linetype = "dashed", size = 1)+
  theme(plot.title = element_text(hjust = 0.5, size=15))  #center the title 

```





```{r, message=F, warning=F}
score_cutoff=0.5


SFARI_genes=as_tibble(read.csv(file.path(root, "../../../Dataset/SFARI_base/SFARI-Gene_genes_08-19-2024release_09-16-2024export.csv")))
positive_genes=SFARI_genes %>% filter (gene.score=="1") %>% select(gene.symbol) %>% pull()


motifs_genes=result_round1 %>% filter(RISK_PROBABILITY_FUSION>score_cutoff) %>% select(gene) %>% pull()



SFARI_score_for_motif_genes <- ifelse(
  motifs_genes %in% SFARI_genes$gene.symbol,
  SFARI_genes$gene.score[match(motifs_genes, SFARI_genes$gene.symbol)],
  NA
)

result_round1 %>% filter(RISK_PROBABILITY_FUSION>score_cutoff) %>% select(gene, gene_gencodeV33, chromosome, RISK_PROBABILITY_FUSION)%>% mutate(SFARI_score=SFARI_score_for_motif_genes) %>% arrange(desc(RISK_PROBABILITY_FUSION)) %>% 
datatable(extensions = 'Buttons',
            options = list(dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'print'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))


SFARI_table=table(SFARI_score_for_motif_genes, useNA = "ifany")


# Define labels, replacing NA with a readable string
labels <- names(SFARI_table)
labels[is.na(labels)] <- "NA"

# Append counts to labels
labels <- paste(labels, "(", SFARI_table, ")", sep = "")

# Create a pie chart
pie(SFARI_table, labels = labels, main = "break-down of motifs genes by SFARI categories", col = rainbow(length(SFARI_table)))
```



* with score cutoff 0.5, 447 genes are selected. 



## new genes not in SFARI cate 1 


```{r, message=F, warning=F}
SFARI_genes=as_tibble(read.csv(file.path(root, "../../../Dataset/SFARI_base/SFARI-Gene_genes_08-19-2024release_09-16-2024export.csv")))
positive_genes=SFARI_genes %>% filter (gene.score=="1") %>% select(gene.symbol) %>% pull()

score_cutoff=0.5

motifs_new_genes=result_round1 %>% filter(RISK_PROBABILITY_FUSION>score_cutoff) %>% select(gene)%>% filter (! gene %in% positive_genes) %>% pull()

length(result_round1 %>% filter(RISK_PROBABILITY_FUSION>score_cutoff) %>% select(gene)%>% filter (gene %in% positive_genes) %>% pull())


SFARI_score_for_motif_new_genes <- ifelse(
  motifs_new_genes %in% SFARI_genes$gene.symbol,
  SFARI_genes$gene.score[match(motifs_new_genes, SFARI_genes$gene.symbol)],
  NA
)

sum(is.na(SFARI_score_for_motif_new_genes))
  
result_round1 %>% filter(RISK_PROBABILITY_FUSION>score_cutoff) %>% select(gene, gene_gencodeV33, chromosome, RISK_PROBABILITY_FUSION)%>% filter (! gene %in% positive_genes)%>% mutate (SFARI_score=SFARI_score_for_motif_new_genes) %>%arrange(desc(RISK_PROBABILITY_FUSION)) %>% 
datatable(extensions = 'Buttons',
          caption = "new ASD genes (not in SFARI cate 1)", 
            options = list(dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'print'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))
```


* 233 genes are in SFARI category 1, but 208 are used and appear in final result. Among 208 genes, 206 are captured by motifs, being among 447 genes, and 2 genes ("ACTB", "PCCB") are missed.  

* among 241 new ASD genes (not in cate 1), 154 are not in SFARI list at all. 